// @aws-sdk/client-pinpoint/dist/es/models/index.js
import { isa as __isa } from "@aws-sdk/smithy-client";
export var ADMChannelRequest;
(function (ADMChannelRequest) {
    ADMChannelRequest.isa = function (o) {
        return __isa(o, "ADMChannelRequest");
    };
})(ADMChannelRequest || (ADMChannelRequest = {}));
export var ADMChannelResponse;
(function (ADMChannelResponse) {
    ADMChannelResponse.isa = function (o) {
        return __isa(o, "ADMChannelResponse");
    };
})(ADMChannelResponse || (ADMChannelResponse = {}));
export var ADMMessage;
(function (ADMMessage) {
    ADMMessage.isa = function (o) { return __isa(o, "ADMMessage"); };
})(ADMMessage || (ADMMessage = {}));
export var APNSChannelRequest;
(function (APNSChannelRequest) {
    APNSChannelRequest.isa = function (o) {
        return __isa(o, "APNSChannelRequest");
    };
})(APNSChannelRequest || (APNSChannelRequest = {}));
export var APNSChannelResponse;
(function (APNSChannelResponse) {
    APNSChannelResponse.isa = function (o) {
        return __isa(o, "APNSChannelResponse");
    };
})(APNSChannelResponse || (APNSChannelResponse = {}));
export var APNSMessage;
(function (APNSMessage) {
    APNSMessage.isa = function (o) { return __isa(o, "APNSMessage"); };
})(APNSMessage || (APNSMessage = {}));
export var APNSPushNotificationTemplate;
(function (APNSPushNotificationTemplate) {
    APNSPushNotificationTemplate.isa = function (o) {
        return __isa(o, "APNSPushNotificationTemplate");
    };
})(APNSPushNotificationTemplate || (APNSPushNotificationTemplate = {}));
export var APNSSandboxChannelRequest;
(function (APNSSandboxChannelRequest) {
    APNSSandboxChannelRequest.isa = function (o) {
        return __isa(o, "APNSSandboxChannelRequest");
    };
})(APNSSandboxChannelRequest || (APNSSandboxChannelRequest = {}));
export var APNSSandboxChannelResponse;
(function (APNSSandboxChannelResponse) {
    APNSSandboxChannelResponse.isa = function (o) {
        return __isa(o, "APNSSandboxChannelResponse");
    };
})(APNSSandboxChannelResponse || (APNSSandboxChannelResponse = {}));
export var APNSVoipChannelRequest;
(function (APNSVoipChannelRequest) {
    APNSVoipChannelRequest.isa = function (o) {
        return __isa(o, "APNSVoipChannelRequest");
    };
})(APNSVoipChannelRequest || (APNSVoipChannelRequest = {}));
export var APNSVoipChannelResponse;
(function (APNSVoipChannelResponse) {
    APNSVoipChannelResponse.isa = function (o) {
        return __isa(o, "APNSVoipChannelResponse");
    };
})(APNSVoipChannelResponse || (APNSVoipChannelResponse = {}));
export var APNSVoipSandboxChannelRequest;
(function (APNSVoipSandboxChannelRequest) {
    APNSVoipSandboxChannelRequest.isa = function (o) {
        return __isa(o, "APNSVoipSandboxChannelRequest");
    };
})(APNSVoipSandboxChannelRequest || (APNSVoipSandboxChannelRequest = {}));
export var APNSVoipSandboxChannelResponse;
(function (APNSVoipSandboxChannelResponse) {
    APNSVoipSandboxChannelResponse.isa = function (o) {
        return __isa(o, "APNSVoipSandboxChannelResponse");
    };
})(APNSVoipSandboxChannelResponse || (APNSVoipSandboxChannelResponse = {}));
export var Action;
(function (Action) {
    Action["DEEP_LINK"] = "DEEP_LINK";
    Action["OPEN_APP"] = "OPEN_APP";
    Action["URL"] = "URL";
})(Action || (Action = {}));
export var ActivitiesResponse;
(function (ActivitiesResponse) {
    ActivitiesResponse.isa = function (o) {
        return __isa(o, "ActivitiesResponse");
    };
})(ActivitiesResponse || (ActivitiesResponse = {}));
export var Activity;
(function (Activity) {
    Activity.isa = function (o) { return __isa(o, "Activity"); };
})(Activity || (Activity = {}));
export var ActivityResponse;
(function (ActivityResponse) {
    ActivityResponse.isa = function (o) {
        return __isa(o, "ActivityResponse");
    };
})(ActivityResponse || (ActivityResponse = {}));
export var AddressConfiguration;
(function (AddressConfiguration) {
    AddressConfiguration.isa = function (o) {
        return __isa(o, "AddressConfiguration");
    };
})(AddressConfiguration || (AddressConfiguration = {}));
export var AndroidPushNotificationTemplate;
(function (AndroidPushNotificationTemplate) {
    AndroidPushNotificationTemplate.isa = function (o) {
        return __isa(o, "AndroidPushNotificationTemplate");
    };
})(AndroidPushNotificationTemplate || (AndroidPushNotificationTemplate = {}));
export var ApplicationDateRangeKpiResponse;
(function (ApplicationDateRangeKpiResponse) {
    ApplicationDateRangeKpiResponse.isa = function (o) {
        return __isa(o, "ApplicationDateRangeKpiResponse");
    };
})(ApplicationDateRangeKpiResponse || (ApplicationDateRangeKpiResponse = {}));
export var ApplicationResponse;
(function (ApplicationResponse) {
    ApplicationResponse.isa = function (o) {
        return __isa(o, "ApplicationResponse");
    };
})(ApplicationResponse || (ApplicationResponse = {}));
export var ApplicationSettingsResource;
(function (ApplicationSettingsResource) {
    ApplicationSettingsResource.isa = function (o) {
        return __isa(o, "ApplicationSettingsResource");
    };
})(ApplicationSettingsResource || (ApplicationSettingsResource = {}));
export var ApplicationsResponse;
(function (ApplicationsResponse) {
    ApplicationsResponse.isa = function (o) {
        return __isa(o, "ApplicationsResponse");
    };
})(ApplicationsResponse || (ApplicationsResponse = {}));
export var AttributeDimension;
(function (AttributeDimension) {
    AttributeDimension.isa = function (o) {
        return __isa(o, "AttributeDimension");
    };
})(AttributeDimension || (AttributeDimension = {}));
export var AttributeType;
(function (AttributeType) {
    AttributeType["EXCLUSIVE"] = "EXCLUSIVE";
    AttributeType["INCLUSIVE"] = "INCLUSIVE";
})(AttributeType || (AttributeType = {}));
export var AttributesResource;
(function (AttributesResource) {
    AttributesResource.isa = function (o) {
        return __isa(o, "AttributesResource");
    };
})(AttributesResource || (AttributesResource = {}));
export var BadRequestException;
(function (BadRequestException) {
    BadRequestException.isa = function (o) {
        return __isa(o, "BadRequestException");
    };
})(BadRequestException || (BadRequestException = {}));
export var BaiduChannelRequest;
(function (BaiduChannelRequest) {
    BaiduChannelRequest.isa = function (o) {
        return __isa(o, "BaiduChannelRequest");
    };
})(BaiduChannelRequest || (BaiduChannelRequest = {}));
export var BaiduChannelResponse;
(function (BaiduChannelResponse) {
    BaiduChannelResponse.isa = function (o) {
        return __isa(o, "BaiduChannelResponse");
    };
})(BaiduChannelResponse || (BaiduChannelResponse = {}));
export var BaiduMessage;
(function (BaiduMessage) {
    BaiduMessage.isa = function (o) { return __isa(o, "BaiduMessage"); };
})(BaiduMessage || (BaiduMessage = {}));
export var BaseKpiResult;
(function (BaseKpiResult) {
    BaseKpiResult.isa = function (o) { return __isa(o, "BaseKpiResult"); };
})(BaseKpiResult || (BaseKpiResult = {}));
export var CampaignDateRangeKpiResponse;
(function (CampaignDateRangeKpiResponse) {
    CampaignDateRangeKpiResponse.isa = function (o) {
        return __isa(o, "CampaignDateRangeKpiResponse");
    };
})(CampaignDateRangeKpiResponse || (CampaignDateRangeKpiResponse = {}));
export var CampaignEmailMessage;
(function (CampaignEmailMessage) {
    CampaignEmailMessage.isa = function (o) {
        return __isa(o, "CampaignEmailMessage");
    };
})(CampaignEmailMessage || (CampaignEmailMessage = {}));
export var CampaignEventFilter;
(function (CampaignEventFilter) {
    CampaignEventFilter.isa = function (o) {
        return __isa(o, "CampaignEventFilter");
    };
})(CampaignEventFilter || (CampaignEventFilter = {}));
export var CampaignHook;
(function (CampaignHook) {
    CampaignHook.isa = function (o) { return __isa(o, "CampaignHook"); };
})(CampaignHook || (CampaignHook = {}));
export var CampaignLimits;
(function (CampaignLimits) {
    CampaignLimits.isa = function (o) {
        return __isa(o, "CampaignLimits");
    };
})(CampaignLimits || (CampaignLimits = {}));
export var CampaignResponse;
(function (CampaignResponse) {
    CampaignResponse.isa = function (o) {
        return __isa(o, "CampaignResponse");
    };
})(CampaignResponse || (CampaignResponse = {}));
export var CampaignSmsMessage;
(function (CampaignSmsMessage) {
    CampaignSmsMessage.isa = function (o) {
        return __isa(o, "CampaignSmsMessage");
    };
})(CampaignSmsMessage || (CampaignSmsMessage = {}));
export var CampaignState;
(function (CampaignState) {
    CampaignState.isa = function (o) { return __isa(o, "CampaignState"); };
})(CampaignState || (CampaignState = {}));
export var CampaignStatus;
(function (CampaignStatus) {
    CampaignStatus["COMPLETED"] = "COMPLETED";
    CampaignStatus["DELETED"] = "DELETED";
    CampaignStatus["EXECUTING"] = "EXECUTING";
    CampaignStatus["PAUSED"] = "PAUSED";
    CampaignStatus["PENDING_NEXT_RUN"] = "PENDING_NEXT_RUN";
    CampaignStatus["SCHEDULED"] = "SCHEDULED";
})(CampaignStatus || (CampaignStatus = {}));
export var CampaignsResponse;
(function (CampaignsResponse) {
    CampaignsResponse.isa = function (o) {
        return __isa(o, "CampaignsResponse");
    };
})(CampaignsResponse || (CampaignsResponse = {}));
export var ChannelResponse;
(function (ChannelResponse) {
    ChannelResponse.isa = function (o) {
        return __isa(o, "ChannelResponse");
    };
})(ChannelResponse || (ChannelResponse = {}));
export var ChannelType;
(function (ChannelType) {
    ChannelType["ADM"] = "ADM";
    ChannelType["APNS"] = "APNS";
    ChannelType["APNS_SANDBOX"] = "APNS_SANDBOX";
    ChannelType["APNS_VOIP"] = "APNS_VOIP";
    ChannelType["APNS_VOIP_SANDBOX"] = "APNS_VOIP_SANDBOX";
    ChannelType["BAIDU"] = "BAIDU";
    ChannelType["CUSTOM"] = "CUSTOM";
    ChannelType["EMAIL"] = "EMAIL";
    ChannelType["GCM"] = "GCM";
    ChannelType["SMS"] = "SMS";
    ChannelType["VOICE"] = "VOICE";
})(ChannelType || (ChannelType = {}));
export var ChannelsResponse;
(function (ChannelsResponse) {
    ChannelsResponse.isa = function (o) {
        return __isa(o, "ChannelsResponse");
    };
})(ChannelsResponse || (ChannelsResponse = {}));
export var Condition;
(function (Condition) {
    Condition.isa = function (o) { return __isa(o, "Condition"); };
})(Condition || (Condition = {}));
export var ConditionalSplitActivity;
(function (ConditionalSplitActivity) {
    ConditionalSplitActivity.isa = function (o) {
        return __isa(o, "ConditionalSplitActivity");
    };
})(ConditionalSplitActivity || (ConditionalSplitActivity = {}));
export var CreateAppRequest;
(function (CreateAppRequest) {
    CreateAppRequest.isa = function (o) {
        return __isa(o, "CreateAppRequest");
    };
})(CreateAppRequest || (CreateAppRequest = {}));
export var CreateAppResponse;
(function (CreateAppResponse) {
    CreateAppResponse.isa = function (o) {
        return __isa(o, "CreateAppResponse");
    };
})(CreateAppResponse || (CreateAppResponse = {}));
export var CreateApplicationRequest;
(function (CreateApplicationRequest) {
    CreateApplicationRequest.isa = function (o) {
        return __isa(o, "CreateApplicationRequest");
    };
})(CreateApplicationRequest || (CreateApplicationRequest = {}));
export var CreateCampaignRequest;
(function (CreateCampaignRequest) {
    CreateCampaignRequest.isa = function (o) {
        return __isa(o, "CreateCampaignRequest");
    };
})(CreateCampaignRequest || (CreateCampaignRequest = {}));
export var CreateCampaignResponse;
(function (CreateCampaignResponse) {
    CreateCampaignResponse.isa = function (o) {
        return __isa(o, "CreateCampaignResponse");
    };
})(CreateCampaignResponse || (CreateCampaignResponse = {}));
export var CreateEmailTemplateRequest;
(function (CreateEmailTemplateRequest) {
    CreateEmailTemplateRequest.isa = function (o) {
        return __isa(o, "CreateEmailTemplateRequest");
    };
})(CreateEmailTemplateRequest || (CreateEmailTemplateRequest = {}));
export var CreateEmailTemplateResponse;
(function (CreateEmailTemplateResponse) {
    CreateEmailTemplateResponse.isa = function (o) {
        return __isa(o, "CreateEmailTemplateResponse");
    };
})(CreateEmailTemplateResponse || (CreateEmailTemplateResponse = {}));
export var CreateExportJobRequest;
(function (CreateExportJobRequest) {
    CreateExportJobRequest.isa = function (o) {
        return __isa(o, "CreateExportJobRequest");
    };
})(CreateExportJobRequest || (CreateExportJobRequest = {}));
export var CreateExportJobResponse;
(function (CreateExportJobResponse) {
    CreateExportJobResponse.isa = function (o) {
        return __isa(o, "CreateExportJobResponse");
    };
})(CreateExportJobResponse || (CreateExportJobResponse = {}));
export var CreateImportJobRequest;
(function (CreateImportJobRequest) {
    CreateImportJobRequest.isa = function (o) {
        return __isa(o, "CreateImportJobRequest");
    };
})(CreateImportJobRequest || (CreateImportJobRequest = {}));
export var CreateImportJobResponse;
(function (CreateImportJobResponse) {
    CreateImportJobResponse.isa = function (o) {
        return __isa(o, "CreateImportJobResponse");
    };
})(CreateImportJobResponse || (CreateImportJobResponse = {}));
export var CreateJourneyRequest;
(function (CreateJourneyRequest) {
    CreateJourneyRequest.isa = function (o) {
        return __isa(o, "CreateJourneyRequest");
    };
})(CreateJourneyRequest || (CreateJourneyRequest = {}));
export var CreateJourneyResponse;
(function (CreateJourneyResponse) {
    CreateJourneyResponse.isa = function (o) {
        return __isa(o, "CreateJourneyResponse");
    };
})(CreateJourneyResponse || (CreateJourneyResponse = {}));
export var CreatePushTemplateRequest;
(function (CreatePushTemplateRequest) {
    CreatePushTemplateRequest.isa = function (o) {
        return __isa(o, "CreatePushTemplateRequest");
    };
})(CreatePushTemplateRequest || (CreatePushTemplateRequest = {}));
export var CreatePushTemplateResponse;
(function (CreatePushTemplateResponse) {
    CreatePushTemplateResponse.isa = function (o) {
        return __isa(o, "CreatePushTemplateResponse");
    };
})(CreatePushTemplateResponse || (CreatePushTemplateResponse = {}));
export var CreateSegmentRequest;
(function (CreateSegmentRequest) {
    CreateSegmentRequest.isa = function (o) {
        return __isa(o, "CreateSegmentRequest");
    };
})(CreateSegmentRequest || (CreateSegmentRequest = {}));
export var CreateSegmentResponse;
(function (CreateSegmentResponse) {
    CreateSegmentResponse.isa = function (o) {
        return __isa(o, "CreateSegmentResponse");
    };
})(CreateSegmentResponse || (CreateSegmentResponse = {}));
export var CreateSmsTemplateRequest;
(function (CreateSmsTemplateRequest) {
    CreateSmsTemplateRequest.isa = function (o) {
        return __isa(o, "CreateSmsTemplateRequest");
    };
})(CreateSmsTemplateRequest || (CreateSmsTemplateRequest = {}));
export var CreateSmsTemplateResponse;
(function (CreateSmsTemplateResponse) {
    CreateSmsTemplateResponse.isa = function (o) {
        return __isa(o, "CreateSmsTemplateResponse");
    };
})(CreateSmsTemplateResponse || (CreateSmsTemplateResponse = {}));
export var CreateTemplateMessageBody;
(function (CreateTemplateMessageBody) {
    CreateTemplateMessageBody.isa = function (o) {
        return __isa(o, "CreateTemplateMessageBody");
    };
})(CreateTemplateMessageBody || (CreateTemplateMessageBody = {}));
export var CreateVoiceTemplateRequest;
(function (CreateVoiceTemplateRequest) {
    CreateVoiceTemplateRequest.isa = function (o) {
        return __isa(o, "CreateVoiceTemplateRequest");
    };
})(CreateVoiceTemplateRequest || (CreateVoiceTemplateRequest = {}));
export var CreateVoiceTemplateResponse;
(function (CreateVoiceTemplateResponse) {
    CreateVoiceTemplateResponse.isa = function (o) {
        return __isa(o, "CreateVoiceTemplateResponse");
    };
})(CreateVoiceTemplateResponse || (CreateVoiceTemplateResponse = {}));
export var DefaultMessage;
(function (DefaultMessage) {
    DefaultMessage.isa = function (o) {
        return __isa(o, "DefaultMessage");
    };
})(DefaultMessage || (DefaultMessage = {}));
export var DefaultPushNotificationMessage;
(function (DefaultPushNotificationMessage) {
    DefaultPushNotificationMessage.isa = function (o) {
        return __isa(o, "DefaultPushNotificationMessage");
    };
})(DefaultPushNotificationMessage || (DefaultPushNotificationMessage = {}));
export var DefaultPushNotificationTemplate;
(function (DefaultPushNotificationTemplate) {
    DefaultPushNotificationTemplate.isa = function (o) {
        return __isa(o, "DefaultPushNotificationTemplate");
    };
})(DefaultPushNotificationTemplate || (DefaultPushNotificationTemplate = {}));
export var DeleteAdmChannelRequest;
(function (DeleteAdmChannelRequest) {
    DeleteAdmChannelRequest.isa = function (o) {
        return __isa(o, "DeleteAdmChannelRequest");
    };
})(DeleteAdmChannelRequest || (DeleteAdmChannelRequest = {}));
export var DeleteAdmChannelResponse;
(function (DeleteAdmChannelResponse) {
    DeleteAdmChannelResponse.isa = function (o) {
        return __isa(o, "DeleteAdmChannelResponse");
    };
})(DeleteAdmChannelResponse || (DeleteAdmChannelResponse = {}));
export var DeleteApnsChannelRequest;
(function (DeleteApnsChannelRequest) {
    DeleteApnsChannelRequest.isa = function (o) {
        return __isa(o, "DeleteApnsChannelRequest");
    };
})(DeleteApnsChannelRequest || (DeleteApnsChannelRequest = {}));
export var DeleteApnsChannelResponse;
(function (DeleteApnsChannelResponse) {
    DeleteApnsChannelResponse.isa = function (o) {
        return __isa(o, "DeleteApnsChannelResponse");
    };
})(DeleteApnsChannelResponse || (DeleteApnsChannelResponse = {}));
export var DeleteApnsSandboxChannelRequest;
(function (DeleteApnsSandboxChannelRequest) {
    DeleteApnsSandboxChannelRequest.isa = function (o) {
        return __isa(o, "DeleteApnsSandboxChannelRequest");
    };
})(DeleteApnsSandboxChannelRequest || (DeleteApnsSandboxChannelRequest = {}));
export var DeleteApnsSandboxChannelResponse;
(function (DeleteApnsSandboxChannelResponse) {
    DeleteApnsSandboxChannelResponse.isa = function (o) {
        return __isa(o, "DeleteApnsSandboxChannelResponse");
    };
})(DeleteApnsSandboxChannelResponse || (DeleteApnsSandboxChannelResponse = {}));
export var DeleteApnsVoipChannelRequest;
(function (DeleteApnsVoipChannelRequest) {
    DeleteApnsVoipChannelRequest.isa = function (o) {
        return __isa(o, "DeleteApnsVoipChannelRequest");
    };
})(DeleteApnsVoipChannelRequest || (DeleteApnsVoipChannelRequest = {}));
export var DeleteApnsVoipChannelResponse;
(function (DeleteApnsVoipChannelResponse) {
    DeleteApnsVoipChannelResponse.isa = function (o) {
        return __isa(o, "DeleteApnsVoipChannelResponse");
    };
})(DeleteApnsVoipChannelResponse || (DeleteApnsVoipChannelResponse = {}));
export var DeleteApnsVoipSandboxChannelRequest;
(function (DeleteApnsVoipSandboxChannelRequest) {
    DeleteApnsVoipSandboxChannelRequest.isa = function (o) {
        return __isa(o, "DeleteApnsVoipSandboxChannelRequest");
    };
})(DeleteApnsVoipSandboxChannelRequest || (DeleteApnsVoipSandboxChannelRequest = {}));
export var DeleteApnsVoipSandboxChannelResponse;
(function (DeleteApnsVoipSandboxChannelResponse) {
    DeleteApnsVoipSandboxChannelResponse.isa = function (o) {
        return __isa(o, "DeleteApnsVoipSandboxChannelResponse");
    };
})(DeleteApnsVoipSandboxChannelResponse || (DeleteApnsVoipSandboxChannelResponse = {}));
export var DeleteAppRequest;
(function (DeleteAppRequest) {
    DeleteAppRequest.isa = function (o) {
        return __isa(o, "DeleteAppRequest");
    };
})(DeleteAppRequest || (DeleteAppRequest = {}));
export var DeleteAppResponse;
(function (DeleteAppResponse) {
    DeleteAppResponse.isa = function (o) {
        return __isa(o, "DeleteAppResponse");
    };
})(DeleteAppResponse || (DeleteAppResponse = {}));
export var DeleteBaiduChannelRequest;
(function (DeleteBaiduChannelRequest) {
    DeleteBaiduChannelRequest.isa = function (o) {
        return __isa(o, "DeleteBaiduChannelRequest");
    };
})(DeleteBaiduChannelRequest || (DeleteBaiduChannelRequest = {}));
export var DeleteBaiduChannelResponse;
(function (DeleteBaiduChannelResponse) {
    DeleteBaiduChannelResponse.isa = function (o) {
        return __isa(o, "DeleteBaiduChannelResponse");
    };
})(DeleteBaiduChannelResponse || (DeleteBaiduChannelResponse = {}));
export var DeleteCampaignRequest;
(function (DeleteCampaignRequest) {
    DeleteCampaignRequest.isa = function (o) {
        return __isa(o, "DeleteCampaignRequest");
    };
})(DeleteCampaignRequest || (DeleteCampaignRequest = {}));
export var DeleteCampaignResponse;
(function (DeleteCampaignResponse) {
    DeleteCampaignResponse.isa = function (o) {
        return __isa(o, "DeleteCampaignResponse");
    };
})(DeleteCampaignResponse || (DeleteCampaignResponse = {}));
export var DeleteEmailChannelRequest;
(function (DeleteEmailChannelRequest) {
    DeleteEmailChannelRequest.isa = function (o) {
        return __isa(o, "DeleteEmailChannelRequest");
    };
})(DeleteEmailChannelRequest || (DeleteEmailChannelRequest = {}));
export var DeleteEmailChannelResponse;
(function (DeleteEmailChannelResponse) {
    DeleteEmailChannelResponse.isa = function (o) {
        return __isa(o, "DeleteEmailChannelResponse");
    };
})(DeleteEmailChannelResponse || (DeleteEmailChannelResponse = {}));
export var DeleteEmailTemplateRequest;
(function (DeleteEmailTemplateRequest) {
    DeleteEmailTemplateRequest.isa = function (o) {
        return __isa(o, "DeleteEmailTemplateRequest");
    };
})(DeleteEmailTemplateRequest || (DeleteEmailTemplateRequest = {}));
export var DeleteEmailTemplateResponse;
(function (DeleteEmailTemplateResponse) {
    DeleteEmailTemplateResponse.isa = function (o) {
        return __isa(o, "DeleteEmailTemplateResponse");
    };
})(DeleteEmailTemplateResponse || (DeleteEmailTemplateResponse = {}));
export var DeleteEndpointRequest;
(function (DeleteEndpointRequest) {
    DeleteEndpointRequest.isa = function (o) {
        return __isa(o, "DeleteEndpointRequest");
    };
})(DeleteEndpointRequest || (DeleteEndpointRequest = {}));
export var DeleteEndpointResponse;
(function (DeleteEndpointResponse) {
    DeleteEndpointResponse.isa = function (o) {
        return __isa(o, "DeleteEndpointResponse");
    };
})(DeleteEndpointResponse || (DeleteEndpointResponse = {}));
export var DeleteEventStreamRequest;
(function (DeleteEventStreamRequest) {
    DeleteEventStreamRequest.isa = function (o) {
        return __isa(o, "DeleteEventStreamRequest");
    };
})(DeleteEventStreamRequest || (DeleteEventStreamRequest = {}));
export var DeleteEventStreamResponse;
(function (DeleteEventStreamResponse) {
    DeleteEventStreamResponse.isa = function (o) {
        return __isa(o, "DeleteEventStreamResponse");
    };
})(DeleteEventStreamResponse || (DeleteEventStreamResponse = {}));
export var DeleteGcmChannelRequest;
(function (DeleteGcmChannelRequest) {
    DeleteGcmChannelRequest.isa = function (o) {
        return __isa(o, "DeleteGcmChannelRequest");
    };
})(DeleteGcmChannelRequest || (DeleteGcmChannelRequest = {}));
export var DeleteGcmChannelResponse;
(function (DeleteGcmChannelResponse) {
    DeleteGcmChannelResponse.isa = function (o) {
        return __isa(o, "DeleteGcmChannelResponse");
    };
})(DeleteGcmChannelResponse || (DeleteGcmChannelResponse = {}));
export var DeleteJourneyRequest;
(function (DeleteJourneyRequest) {
    DeleteJourneyRequest.isa = function (o) {
        return __isa(o, "DeleteJourneyRequest");
    };
})(DeleteJourneyRequest || (DeleteJourneyRequest = {}));
export var DeleteJourneyResponse;
(function (DeleteJourneyResponse) {
    DeleteJourneyResponse.isa = function (o) {
        return __isa(o, "DeleteJourneyResponse");
    };
})(DeleteJourneyResponse || (DeleteJourneyResponse = {}));
export var DeletePushTemplateRequest;
(function (DeletePushTemplateRequest) {
    DeletePushTemplateRequest.isa = function (o) {
        return __isa(o, "DeletePushTemplateRequest");
    };
})(DeletePushTemplateRequest || (DeletePushTemplateRequest = {}));
export var DeletePushTemplateResponse;
(function (DeletePushTemplateResponse) {
    DeletePushTemplateResponse.isa = function (o) {
        return __isa(o, "DeletePushTemplateResponse");
    };
})(DeletePushTemplateResponse || (DeletePushTemplateResponse = {}));
export var DeleteSegmentRequest;
(function (DeleteSegmentRequest) {
    DeleteSegmentRequest.isa = function (o) {
        return __isa(o, "DeleteSegmentRequest");
    };
})(DeleteSegmentRequest || (DeleteSegmentRequest = {}));
export var DeleteSegmentResponse;
(function (DeleteSegmentResponse) {
    DeleteSegmentResponse.isa = function (o) {
        return __isa(o, "DeleteSegmentResponse");
    };
})(DeleteSegmentResponse || (DeleteSegmentResponse = {}));
export var DeleteSmsChannelRequest;
(function (DeleteSmsChannelRequest) {
    DeleteSmsChannelRequest.isa = function (o) {
        return __isa(o, "DeleteSmsChannelRequest");
    };
})(DeleteSmsChannelRequest || (DeleteSmsChannelRequest = {}));
export var DeleteSmsChannelResponse;
(function (DeleteSmsChannelResponse) {
    DeleteSmsChannelResponse.isa = function (o) {
        return __isa(o, "DeleteSmsChannelResponse");
    };
})(DeleteSmsChannelResponse || (DeleteSmsChannelResponse = {}));
export var DeleteSmsTemplateRequest;
(function (DeleteSmsTemplateRequest) {
    DeleteSmsTemplateRequest.isa = function (o) {
        return __isa(o, "DeleteSmsTemplateRequest");
    };
})(DeleteSmsTemplateRequest || (DeleteSmsTemplateRequest = {}));
export var DeleteSmsTemplateResponse;
(function (DeleteSmsTemplateResponse) {
    DeleteSmsTemplateResponse.isa = function (o) {
        return __isa(o, "DeleteSmsTemplateResponse");
    };
})(DeleteSmsTemplateResponse || (DeleteSmsTemplateResponse = {}));
export var DeleteUserEndpointsRequest;
(function (DeleteUserEndpointsRequest) {
    DeleteUserEndpointsRequest.isa = function (o) {
        return __isa(o, "DeleteUserEndpointsRequest");
    };
})(DeleteUserEndpointsRequest || (DeleteUserEndpointsRequest = {}));
export var DeleteUserEndpointsResponse;
(function (DeleteUserEndpointsResponse) {
    DeleteUserEndpointsResponse.isa = function (o) {
        return __isa(o, "DeleteUserEndpointsResponse");
    };
})(DeleteUserEndpointsResponse || (DeleteUserEndpointsResponse = {}));
export var DeleteVoiceChannelRequest;
(function (DeleteVoiceChannelRequest) {
    DeleteVoiceChannelRequest.isa = function (o) {
        return __isa(o, "DeleteVoiceChannelRequest");
    };
})(DeleteVoiceChannelRequest || (DeleteVoiceChannelRequest = {}));
export var DeleteVoiceChannelResponse;
(function (DeleteVoiceChannelResponse) {
    DeleteVoiceChannelResponse.isa = function (o) {
        return __isa(o, "DeleteVoiceChannelResponse");
    };
})(DeleteVoiceChannelResponse || (DeleteVoiceChannelResponse = {}));
export var DeleteVoiceTemplateRequest;
(function (DeleteVoiceTemplateRequest) {
    DeleteVoiceTemplateRequest.isa = function (o) {
        return __isa(o, "DeleteVoiceTemplateRequest");
    };
})(DeleteVoiceTemplateRequest || (DeleteVoiceTemplateRequest = {}));
export var DeleteVoiceTemplateResponse;
(function (DeleteVoiceTemplateResponse) {
    DeleteVoiceTemplateResponse.isa = function (o) {
        return __isa(o, "DeleteVoiceTemplateResponse");
    };
})(DeleteVoiceTemplateResponse || (DeleteVoiceTemplateResponse = {}));
export var DeliveryStatus;
(function (DeliveryStatus) {
    DeliveryStatus["DUPLICATE"] = "DUPLICATE";
    DeliveryStatus["OPT_OUT"] = "OPT_OUT";
    DeliveryStatus["PERMANENT_FAILURE"] = "PERMANENT_FAILURE";
    DeliveryStatus["SUCCESSFUL"] = "SUCCESSFUL";
    DeliveryStatus["TEMPORARY_FAILURE"] = "TEMPORARY_FAILURE";
    DeliveryStatus["THROTTLED"] = "THROTTLED";
    DeliveryStatus["UNKNOWN_FAILURE"] = "UNKNOWN_FAILURE";
})(DeliveryStatus || (DeliveryStatus = {}));
export var DimensionType;
(function (DimensionType) {
    DimensionType["EXCLUSIVE"] = "EXCLUSIVE";
    DimensionType["INCLUSIVE"] = "INCLUSIVE";
})(DimensionType || (DimensionType = {}));
export var DirectMessageConfiguration;
(function (DirectMessageConfiguration) {
    DirectMessageConfiguration.isa = function (o) {
        return __isa(o, "DirectMessageConfiguration");
    };
})(DirectMessageConfiguration || (DirectMessageConfiguration = {}));
export var Duration;
(function (Duration) {
    Duration["DAY_14"] = "DAY_14";
    Duration["DAY_30"] = "DAY_30";
    Duration["DAY_7"] = "DAY_7";
    Duration["HR_24"] = "HR_24";
})(Duration || (Duration = {}));
export var EmailChannelRequest;
(function (EmailChannelRequest) {
    EmailChannelRequest.isa = function (o) {
        return __isa(o, "EmailChannelRequest");
    };
})(EmailChannelRequest || (EmailChannelRequest = {}));
export var EmailChannelResponse;
(function (EmailChannelResponse) {
    EmailChannelResponse.isa = function (o) {
        return __isa(o, "EmailChannelResponse");
    };
})(EmailChannelResponse || (EmailChannelResponse = {}));
export var EmailMessage;
(function (EmailMessage) {
    EmailMessage.isa = function (o) { return __isa(o, "EmailMessage"); };
})(EmailMessage || (EmailMessage = {}));
export var EmailMessageActivity;
(function (EmailMessageActivity) {
    EmailMessageActivity.isa = function (o) {
        return __isa(o, "EmailMessageActivity");
    };
})(EmailMessageActivity || (EmailMessageActivity = {}));
export var EmailTemplateRequest;
(function (EmailTemplateRequest) {
    EmailTemplateRequest.isa = function (o) {
        return __isa(o, "EmailTemplateRequest");
    };
})(EmailTemplateRequest || (EmailTemplateRequest = {}));
export var EmailTemplateResponse;
(function (EmailTemplateResponse) {
    EmailTemplateResponse.isa = function (o) {
        return __isa(o, "EmailTemplateResponse");
    };
})(EmailTemplateResponse || (EmailTemplateResponse = {}));
export var EndpointBatchItem;
(function (EndpointBatchItem) {
    EndpointBatchItem.isa = function (o) {
        return __isa(o, "EndpointBatchItem");
    };
})(EndpointBatchItem || (EndpointBatchItem = {}));
export var EndpointBatchRequest;
(function (EndpointBatchRequest) {
    EndpointBatchRequest.isa = function (o) {
        return __isa(o, "EndpointBatchRequest");
    };
})(EndpointBatchRequest || (EndpointBatchRequest = {}));
export var EndpointDemographic;
(function (EndpointDemographic) {
    EndpointDemographic.isa = function (o) {
        return __isa(o, "EndpointDemographic");
    };
})(EndpointDemographic || (EndpointDemographic = {}));
export var EndpointItemResponse;
(function (EndpointItemResponse) {
    EndpointItemResponse.isa = function (o) {
        return __isa(o, "EndpointItemResponse");
    };
})(EndpointItemResponse || (EndpointItemResponse = {}));
export var EndpointLocation;
(function (EndpointLocation) {
    EndpointLocation.isa = function (o) {
        return __isa(o, "EndpointLocation");
    };
})(EndpointLocation || (EndpointLocation = {}));
export var EndpointMessageResult;
(function (EndpointMessageResult) {
    EndpointMessageResult.isa = function (o) {
        return __isa(o, "EndpointMessageResult");
    };
})(EndpointMessageResult || (EndpointMessageResult = {}));
export var EndpointRequest;
(function (EndpointRequest) {
    EndpointRequest.isa = function (o) {
        return __isa(o, "EndpointRequest");
    };
})(EndpointRequest || (EndpointRequest = {}));
export var EndpointResponse;
(function (EndpointResponse) {
    EndpointResponse.isa = function (o) {
        return __isa(o, "EndpointResponse");
    };
})(EndpointResponse || (EndpointResponse = {}));
export var EndpointSendConfiguration;
(function (EndpointSendConfiguration) {
    EndpointSendConfiguration.isa = function (o) {
        return __isa(o, "EndpointSendConfiguration");
    };
})(EndpointSendConfiguration || (EndpointSendConfiguration = {}));
export var EndpointUser;
(function (EndpointUser) {
    EndpointUser.isa = function (o) { return __isa(o, "EndpointUser"); };
})(EndpointUser || (EndpointUser = {}));
export var EndpointsResponse;
(function (EndpointsResponse) {
    EndpointsResponse.isa = function (o) {
        return __isa(o, "EndpointsResponse");
    };
})(EndpointsResponse || (EndpointsResponse = {}));
export var Event;
(function (Event) {
    Event.isa = function (o) { return __isa(o, "Event"); };
})(Event || (Event = {}));
export var EventCondition;
(function (EventCondition) {
    EventCondition.isa = function (o) {
        return __isa(o, "EventCondition");
    };
})(EventCondition || (EventCondition = {}));
export var EventDimensions;
(function (EventDimensions) {
    EventDimensions.isa = function (o) {
        return __isa(o, "EventDimensions");
    };
})(EventDimensions || (EventDimensions = {}));
export var EventItemResponse;
(function (EventItemResponse) {
    EventItemResponse.isa = function (o) {
        return __isa(o, "EventItemResponse");
    };
})(EventItemResponse || (EventItemResponse = {}));
export var EventStream;
(function (EventStream) {
    EventStream.isa = function (o) { return __isa(o, "EventStream"); };
})(EventStream || (EventStream = {}));
export var EventsBatch;
(function (EventsBatch) {
    EventsBatch.isa = function (o) { return __isa(o, "EventsBatch"); };
})(EventsBatch || (EventsBatch = {}));
export var EventsRequest;
(function (EventsRequest) {
    EventsRequest.isa = function (o) { return __isa(o, "EventsRequest"); };
})(EventsRequest || (EventsRequest = {}));
export var EventsResponse;
(function (EventsResponse) {
    EventsResponse.isa = function (o) {
        return __isa(o, "EventsResponse");
    };
})(EventsResponse || (EventsResponse = {}));
export var ExportJobRequest;
(function (ExportJobRequest) {
    ExportJobRequest.isa = function (o) {
        return __isa(o, "ExportJobRequest");
    };
})(ExportJobRequest || (ExportJobRequest = {}));
export var ExportJobResource;
(function (ExportJobResource) {
    ExportJobResource.isa = function (o) {
        return __isa(o, "ExportJobResource");
    };
})(ExportJobResource || (ExportJobResource = {}));
export var ExportJobResponse;
(function (ExportJobResponse) {
    ExportJobResponse.isa = function (o) {
        return __isa(o, "ExportJobResponse");
    };
})(ExportJobResponse || (ExportJobResponse = {}));
export var ExportJobsResponse;
(function (ExportJobsResponse) {
    ExportJobsResponse.isa = function (o) {
        return __isa(o, "ExportJobsResponse");
    };
})(ExportJobsResponse || (ExportJobsResponse = {}));
export var FilterType;
(function (FilterType) {
    FilterType["ENDPOINT"] = "ENDPOINT";
    FilterType["SYSTEM"] = "SYSTEM";
})(FilterType || (FilterType = {}));
export var ForbiddenException;
(function (ForbiddenException) {
    ForbiddenException.isa = function (o) {
        return __isa(o, "ForbiddenException");
    };
})(ForbiddenException || (ForbiddenException = {}));
export var Format;
(function (Format) {
    Format["CSV"] = "CSV";
    Format["JSON"] = "JSON";
})(Format || (Format = {}));
export var Frequency;
(function (Frequency) {
    Frequency["DAILY"] = "DAILY";
    Frequency["EVENT"] = "EVENT";
    Frequency["HOURLY"] = "HOURLY";
    Frequency["MONTHLY"] = "MONTHLY";
    Frequency["ONCE"] = "ONCE";
    Frequency["WEEKLY"] = "WEEKLY";
})(Frequency || (Frequency = {}));
export var GCMChannelRequest;
(function (GCMChannelRequest) {
    GCMChannelRequest.isa = function (o) {
        return __isa(o, "GCMChannelRequest");
    };
})(GCMChannelRequest || (GCMChannelRequest = {}));
export var GCMChannelResponse;
(function (GCMChannelResponse) {
    GCMChannelResponse.isa = function (o) {
        return __isa(o, "GCMChannelResponse");
    };
})(GCMChannelResponse || (GCMChannelResponse = {}));
export var GCMMessage;
(function (GCMMessage) {
    GCMMessage.isa = function (o) { return __isa(o, "GCMMessage"); };
})(GCMMessage || (GCMMessage = {}));
export var GPSCoordinates;
(function (GPSCoordinates) {
    GPSCoordinates.isa = function (o) {
        return __isa(o, "GPSCoordinates");
    };
})(GPSCoordinates || (GPSCoordinates = {}));
export var GPSPointDimension;
(function (GPSPointDimension) {
    GPSPointDimension.isa = function (o) {
        return __isa(o, "GPSPointDimension");
    };
})(GPSPointDimension || (GPSPointDimension = {}));
export var GetAdmChannelRequest;
(function (GetAdmChannelRequest) {
    GetAdmChannelRequest.isa = function (o) {
        return __isa(o, "GetAdmChannelRequest");
    };
})(GetAdmChannelRequest || (GetAdmChannelRequest = {}));
export var GetAdmChannelResponse;
(function (GetAdmChannelResponse) {
    GetAdmChannelResponse.isa = function (o) {
        return __isa(o, "GetAdmChannelResponse");
    };
})(GetAdmChannelResponse || (GetAdmChannelResponse = {}));
export var GetApnsChannelRequest;
(function (GetApnsChannelRequest) {
    GetApnsChannelRequest.isa = function (o) {
        return __isa(o, "GetApnsChannelRequest");
    };
})(GetApnsChannelRequest || (GetApnsChannelRequest = {}));
export var GetApnsChannelResponse;
(function (GetApnsChannelResponse) {
    GetApnsChannelResponse.isa = function (o) {
        return __isa(o, "GetApnsChannelResponse");
    };
})(GetApnsChannelResponse || (GetApnsChannelResponse = {}));
export var GetApnsSandboxChannelRequest;
(function (GetApnsSandboxChannelRequest) {
    GetApnsSandboxChannelRequest.isa = function (o) {
        return __isa(o, "GetApnsSandboxChannelRequest");
    };
})(GetApnsSandboxChannelRequest || (GetApnsSandboxChannelRequest = {}));
export var GetApnsSandboxChannelResponse;
(function (GetApnsSandboxChannelResponse) {
    GetApnsSandboxChannelResponse.isa = function (o) {
        return __isa(o, "GetApnsSandboxChannelResponse");
    };
})(GetApnsSandboxChannelResponse || (GetApnsSandboxChannelResponse = {}));
export var GetApnsVoipChannelRequest;
(function (GetApnsVoipChannelRequest) {
    GetApnsVoipChannelRequest.isa = function (o) {
        return __isa(o, "GetApnsVoipChannelRequest");
    };
})(GetApnsVoipChannelRequest || (GetApnsVoipChannelRequest = {}));
export var GetApnsVoipChannelResponse;
(function (GetApnsVoipChannelResponse) {
    GetApnsVoipChannelResponse.isa = function (o) {
        return __isa(o, "GetApnsVoipChannelResponse");
    };
})(GetApnsVoipChannelResponse || (GetApnsVoipChannelResponse = {}));
export var GetApnsVoipSandboxChannelRequest;
(function (GetApnsVoipSandboxChannelRequest) {
    GetApnsVoipSandboxChannelRequest.isa = function (o) {
        return __isa(o, "GetApnsVoipSandboxChannelRequest");
    };
})(GetApnsVoipSandboxChannelRequest || (GetApnsVoipSandboxChannelRequest = {}));
export var GetApnsVoipSandboxChannelResponse;
(function (GetApnsVoipSandboxChannelResponse) {
    GetApnsVoipSandboxChannelResponse.isa = function (o) {
        return __isa(o, "GetApnsVoipSandboxChannelResponse");
    };
})(GetApnsVoipSandboxChannelResponse || (GetApnsVoipSandboxChannelResponse = {}));
export var GetAppRequest;
(function (GetAppRequest) {
    GetAppRequest.isa = function (o) { return __isa(o, "GetAppRequest"); };
})(GetAppRequest || (GetAppRequest = {}));
export var GetAppResponse;
(function (GetAppResponse) {
    GetAppResponse.isa = function (o) {
        return __isa(o, "GetAppResponse");
    };
})(GetAppResponse || (GetAppResponse = {}));
export var GetApplicationDateRangeKpiRequest;
(function (GetApplicationDateRangeKpiRequest) {
    GetApplicationDateRangeKpiRequest.isa = function (o) {
        return __isa(o, "GetApplicationDateRangeKpiRequest");
    };
})(GetApplicationDateRangeKpiRequest || (GetApplicationDateRangeKpiRequest = {}));
export var GetApplicationDateRangeKpiResponse;
(function (GetApplicationDateRangeKpiResponse) {
    GetApplicationDateRangeKpiResponse.isa = function (o) {
        return __isa(o, "GetApplicationDateRangeKpiResponse");
    };
})(GetApplicationDateRangeKpiResponse || (GetApplicationDateRangeKpiResponse = {}));
export var GetApplicationSettingsRequest;
(function (GetApplicationSettingsRequest) {
    GetApplicationSettingsRequest.isa = function (o) {
        return __isa(o, "GetApplicationSettingsRequest");
    };
})(GetApplicationSettingsRequest || (GetApplicationSettingsRequest = {}));
export var GetApplicationSettingsResponse;
(function (GetApplicationSettingsResponse) {
    GetApplicationSettingsResponse.isa = function (o) {
        return __isa(o, "GetApplicationSettingsResponse");
    };
})(GetApplicationSettingsResponse || (GetApplicationSettingsResponse = {}));
export var GetAppsRequest;
(function (GetAppsRequest) {
    GetAppsRequest.isa = function (o) {
        return __isa(o, "GetAppsRequest");
    };
})(GetAppsRequest || (GetAppsRequest = {}));
export var GetAppsResponse;
(function (GetAppsResponse) {
    GetAppsResponse.isa = function (o) {
        return __isa(o, "GetAppsResponse");
    };
})(GetAppsResponse || (GetAppsResponse = {}));
export var GetBaiduChannelRequest;
(function (GetBaiduChannelRequest) {
    GetBaiduChannelRequest.isa = function (o) {
        return __isa(o, "GetBaiduChannelRequest");
    };
})(GetBaiduChannelRequest || (GetBaiduChannelRequest = {}));
export var GetBaiduChannelResponse;
(function (GetBaiduChannelResponse) {
    GetBaiduChannelResponse.isa = function (o) {
        return __isa(o, "GetBaiduChannelResponse");
    };
})(GetBaiduChannelResponse || (GetBaiduChannelResponse = {}));
export var GetCampaignActivitiesRequest;
(function (GetCampaignActivitiesRequest) {
    GetCampaignActivitiesRequest.isa = function (o) {
        return __isa(o, "GetCampaignActivitiesRequest");
    };
})(GetCampaignActivitiesRequest || (GetCampaignActivitiesRequest = {}));
export var GetCampaignActivitiesResponse;
(function (GetCampaignActivitiesResponse) {
    GetCampaignActivitiesResponse.isa = function (o) {
        return __isa(o, "GetCampaignActivitiesResponse");
    };
})(GetCampaignActivitiesResponse || (GetCampaignActivitiesResponse = {}));
export var GetCampaignDateRangeKpiRequest;
(function (GetCampaignDateRangeKpiRequest) {
    GetCampaignDateRangeKpiRequest.isa = function (o) {
        return __isa(o, "GetCampaignDateRangeKpiRequest");
    };
})(GetCampaignDateRangeKpiRequest || (GetCampaignDateRangeKpiRequest = {}));
export var GetCampaignDateRangeKpiResponse;
(function (GetCampaignDateRangeKpiResponse) {
    GetCampaignDateRangeKpiResponse.isa = function (o) {
        return __isa(o, "GetCampaignDateRangeKpiResponse");
    };
})(GetCampaignDateRangeKpiResponse || (GetCampaignDateRangeKpiResponse = {}));
export var GetCampaignRequest;
(function (GetCampaignRequest) {
    GetCampaignRequest.isa = function (o) {
        return __isa(o, "GetCampaignRequest");
    };
})(GetCampaignRequest || (GetCampaignRequest = {}));
export var GetCampaignResponse;
(function (GetCampaignResponse) {
    GetCampaignResponse.isa = function (o) {
        return __isa(o, "GetCampaignResponse");
    };
})(GetCampaignResponse || (GetCampaignResponse = {}));
export var GetCampaignVersionRequest;
(function (GetCampaignVersionRequest) {
    GetCampaignVersionRequest.isa = function (o) {
        return __isa(o, "GetCampaignVersionRequest");
    };
})(GetCampaignVersionRequest || (GetCampaignVersionRequest = {}));
export var GetCampaignVersionResponse;
(function (GetCampaignVersionResponse) {
    GetCampaignVersionResponse.isa = function (o) {
        return __isa(o, "GetCampaignVersionResponse");
    };
})(GetCampaignVersionResponse || (GetCampaignVersionResponse = {}));
export var GetCampaignVersionsRequest;
(function (GetCampaignVersionsRequest) {
    GetCampaignVersionsRequest.isa = function (o) {
        return __isa(o, "GetCampaignVersionsRequest");
    };
})(GetCampaignVersionsRequest || (GetCampaignVersionsRequest = {}));
export var GetCampaignVersionsResponse;
(function (GetCampaignVersionsResponse) {
    GetCampaignVersionsResponse.isa = function (o) {
        return __isa(o, "GetCampaignVersionsResponse");
    };
})(GetCampaignVersionsResponse || (GetCampaignVersionsResponse = {}));
export var GetCampaignsRequest;
(function (GetCampaignsRequest) {
    GetCampaignsRequest.isa = function (o) {
        return __isa(o, "GetCampaignsRequest");
    };
})(GetCampaignsRequest || (GetCampaignsRequest = {}));
export var GetCampaignsResponse;
(function (GetCampaignsResponse) {
    GetCampaignsResponse.isa = function (o) {
        return __isa(o, "GetCampaignsResponse");
    };
})(GetCampaignsResponse || (GetCampaignsResponse = {}));
export var GetChannelsRequest;
(function (GetChannelsRequest) {
    GetChannelsRequest.isa = function (o) {
        return __isa(o, "GetChannelsRequest");
    };
})(GetChannelsRequest || (GetChannelsRequest = {}));
export var GetChannelsResponse;
(function (GetChannelsResponse) {
    GetChannelsResponse.isa = function (o) {
        return __isa(o, "GetChannelsResponse");
    };
})(GetChannelsResponse || (GetChannelsResponse = {}));
export var GetEmailChannelRequest;
(function (GetEmailChannelRequest) {
    GetEmailChannelRequest.isa = function (o) {
        return __isa(o, "GetEmailChannelRequest");
    };
})(GetEmailChannelRequest || (GetEmailChannelRequest = {}));
export var GetEmailChannelResponse;
(function (GetEmailChannelResponse) {
    GetEmailChannelResponse.isa = function (o) {
        return __isa(o, "GetEmailChannelResponse");
    };
})(GetEmailChannelResponse || (GetEmailChannelResponse = {}));
export var GetEmailTemplateRequest;
(function (GetEmailTemplateRequest) {
    GetEmailTemplateRequest.isa = function (o) {
        return __isa(o, "GetEmailTemplateRequest");
    };
})(GetEmailTemplateRequest || (GetEmailTemplateRequest = {}));
export var GetEmailTemplateResponse;
(function (GetEmailTemplateResponse) {
    GetEmailTemplateResponse.isa = function (o) {
        return __isa(o, "GetEmailTemplateResponse");
    };
})(GetEmailTemplateResponse || (GetEmailTemplateResponse = {}));
export var GetEndpointRequest;
(function (GetEndpointRequest) {
    GetEndpointRequest.isa = function (o) {
        return __isa(o, "GetEndpointRequest");
    };
})(GetEndpointRequest || (GetEndpointRequest = {}));
export var GetEndpointResponse;
(function (GetEndpointResponse) {
    GetEndpointResponse.isa = function (o) {
        return __isa(o, "GetEndpointResponse");
    };
})(GetEndpointResponse || (GetEndpointResponse = {}));
export var GetEventStreamRequest;
(function (GetEventStreamRequest) {
    GetEventStreamRequest.isa = function (o) {
        return __isa(o, "GetEventStreamRequest");
    };
})(GetEventStreamRequest || (GetEventStreamRequest = {}));
export var GetEventStreamResponse;
(function (GetEventStreamResponse) {
    GetEventStreamResponse.isa = function (o) {
        return __isa(o, "GetEventStreamResponse");
    };
})(GetEventStreamResponse || (GetEventStreamResponse = {}));
export var GetExportJobRequest;
(function (GetExportJobRequest) {
    GetExportJobRequest.isa = function (o) {
        return __isa(o, "GetExportJobRequest");
    };
})(GetExportJobRequest || (GetExportJobRequest = {}));
export var GetExportJobResponse;
(function (GetExportJobResponse) {
    GetExportJobResponse.isa = function (o) {
        return __isa(o, "GetExportJobResponse");
    };
})(GetExportJobResponse || (GetExportJobResponse = {}));
export var GetExportJobsRequest;
(function (GetExportJobsRequest) {
    GetExportJobsRequest.isa = function (o) {
        return __isa(o, "GetExportJobsRequest");
    };
})(GetExportJobsRequest || (GetExportJobsRequest = {}));
export var GetExportJobsResponse;
(function (GetExportJobsResponse) {
    GetExportJobsResponse.isa = function (o) {
        return __isa(o, "GetExportJobsResponse");
    };
})(GetExportJobsResponse || (GetExportJobsResponse = {}));
export var GetGcmChannelRequest;
(function (GetGcmChannelRequest) {
    GetGcmChannelRequest.isa = function (o) {
        return __isa(o, "GetGcmChannelRequest");
    };
})(GetGcmChannelRequest || (GetGcmChannelRequest = {}));
export var GetGcmChannelResponse;
(function (GetGcmChannelResponse) {
    GetGcmChannelResponse.isa = function (o) {
        return __isa(o, "GetGcmChannelResponse");
    };
})(GetGcmChannelResponse || (GetGcmChannelResponse = {}));
export var GetImportJobRequest;
(function (GetImportJobRequest) {
    GetImportJobRequest.isa = function (o) {
        return __isa(o, "GetImportJobRequest");
    };
})(GetImportJobRequest || (GetImportJobRequest = {}));
export var GetImportJobResponse;
(function (GetImportJobResponse) {
    GetImportJobResponse.isa = function (o) {
        return __isa(o, "GetImportJobResponse");
    };
})(GetImportJobResponse || (GetImportJobResponse = {}));
export var GetImportJobsRequest;
(function (GetImportJobsRequest) {
    GetImportJobsRequest.isa = function (o) {
        return __isa(o, "GetImportJobsRequest");
    };
})(GetImportJobsRequest || (GetImportJobsRequest = {}));
export var GetImportJobsResponse;
(function (GetImportJobsResponse) {
    GetImportJobsResponse.isa = function (o) {
        return __isa(o, "GetImportJobsResponse");
    };
})(GetImportJobsResponse || (GetImportJobsResponse = {}));
export var GetJourneyDateRangeKpiRequest;
(function (GetJourneyDateRangeKpiRequest) {
    GetJourneyDateRangeKpiRequest.isa = function (o) {
        return __isa(o, "GetJourneyDateRangeKpiRequest");
    };
})(GetJourneyDateRangeKpiRequest || (GetJourneyDateRangeKpiRequest = {}));
export var GetJourneyDateRangeKpiResponse;
(function (GetJourneyDateRangeKpiResponse) {
    GetJourneyDateRangeKpiResponse.isa = function (o) {
        return __isa(o, "GetJourneyDateRangeKpiResponse");
    };
})(GetJourneyDateRangeKpiResponse || (GetJourneyDateRangeKpiResponse = {}));
export var GetJourneyExecutionActivityMetricsRequest;
(function (GetJourneyExecutionActivityMetricsRequest) {
    GetJourneyExecutionActivityMetricsRequest.isa = function (o) {
        return __isa(o, "GetJourneyExecutionActivityMetricsRequest");
    };
})(GetJourneyExecutionActivityMetricsRequest || (GetJourneyExecutionActivityMetricsRequest = {}));
export var GetJourneyExecutionActivityMetricsResponse;
(function (GetJourneyExecutionActivityMetricsResponse) {
    GetJourneyExecutionActivityMetricsResponse.isa = function (o) {
        return __isa(o, "GetJourneyExecutionActivityMetricsResponse");
    };
})(GetJourneyExecutionActivityMetricsResponse || (GetJourneyExecutionActivityMetricsResponse = {}));
export var GetJourneyExecutionMetricsRequest;
(function (GetJourneyExecutionMetricsRequest) {
    GetJourneyExecutionMetricsRequest.isa = function (o) {
        return __isa(o, "GetJourneyExecutionMetricsRequest");
    };
})(GetJourneyExecutionMetricsRequest || (GetJourneyExecutionMetricsRequest = {}));
export var GetJourneyExecutionMetricsResponse;
(function (GetJourneyExecutionMetricsResponse) {
    GetJourneyExecutionMetricsResponse.isa = function (o) {
        return __isa(o, "GetJourneyExecutionMetricsResponse");
    };
})(GetJourneyExecutionMetricsResponse || (GetJourneyExecutionMetricsResponse = {}));
export var GetJourneyRequest;
(function (GetJourneyRequest) {
    GetJourneyRequest.isa = function (o) {
        return __isa(o, "GetJourneyRequest");
    };
})(GetJourneyRequest || (GetJourneyRequest = {}));
export var GetJourneyResponse;
(function (GetJourneyResponse) {
    GetJourneyResponse.isa = function (o) {
        return __isa(o, "GetJourneyResponse");
    };
})(GetJourneyResponse || (GetJourneyResponse = {}));
export var GetPushTemplateRequest;
(function (GetPushTemplateRequest) {
    GetPushTemplateRequest.isa = function (o) {
        return __isa(o, "GetPushTemplateRequest");
    };
})(GetPushTemplateRequest || (GetPushTemplateRequest = {}));
export var GetPushTemplateResponse;
(function (GetPushTemplateResponse) {
    GetPushTemplateResponse.isa = function (o) {
        return __isa(o, "GetPushTemplateResponse");
    };
})(GetPushTemplateResponse || (GetPushTemplateResponse = {}));
export var GetSegmentExportJobsRequest;
(function (GetSegmentExportJobsRequest) {
    GetSegmentExportJobsRequest.isa = function (o) {
        return __isa(o, "GetSegmentExportJobsRequest");
    };
})(GetSegmentExportJobsRequest || (GetSegmentExportJobsRequest = {}));
export var GetSegmentExportJobsResponse;
(function (GetSegmentExportJobsResponse) {
    GetSegmentExportJobsResponse.isa = function (o) {
        return __isa(o, "GetSegmentExportJobsResponse");
    };
})(GetSegmentExportJobsResponse || (GetSegmentExportJobsResponse = {}));
export var GetSegmentImportJobsRequest;
(function (GetSegmentImportJobsRequest) {
    GetSegmentImportJobsRequest.isa = function (o) {
        return __isa(o, "GetSegmentImportJobsRequest");
    };
})(GetSegmentImportJobsRequest || (GetSegmentImportJobsRequest = {}));
export var GetSegmentImportJobsResponse;
(function (GetSegmentImportJobsResponse) {
    GetSegmentImportJobsResponse.isa = function (o) {
        return __isa(o, "GetSegmentImportJobsResponse");
    };
})(GetSegmentImportJobsResponse || (GetSegmentImportJobsResponse = {}));
export var GetSegmentRequest;
(function (GetSegmentRequest) {
    GetSegmentRequest.isa = function (o) {
        return __isa(o, "GetSegmentRequest");
    };
})(GetSegmentRequest || (GetSegmentRequest = {}));
export var GetSegmentResponse;
(function (GetSegmentResponse) {
    GetSegmentResponse.isa = function (o) {
        return __isa(o, "GetSegmentResponse");
    };
})(GetSegmentResponse || (GetSegmentResponse = {}));
export var GetSegmentVersionRequest;
(function (GetSegmentVersionRequest) {
    GetSegmentVersionRequest.isa = function (o) {
        return __isa(o, "GetSegmentVersionRequest");
    };
})(GetSegmentVersionRequest || (GetSegmentVersionRequest = {}));
export var GetSegmentVersionResponse;
(function (GetSegmentVersionResponse) {
    GetSegmentVersionResponse.isa = function (o) {
        return __isa(o, "GetSegmentVersionResponse");
    };
})(GetSegmentVersionResponse || (GetSegmentVersionResponse = {}));
export var GetSegmentVersionsRequest;
(function (GetSegmentVersionsRequest) {
    GetSegmentVersionsRequest.isa = function (o) {
        return __isa(o, "GetSegmentVersionsRequest");
    };
})(GetSegmentVersionsRequest || (GetSegmentVersionsRequest = {}));
export var GetSegmentVersionsResponse;
(function (GetSegmentVersionsResponse) {
    GetSegmentVersionsResponse.isa = function (o) {
        return __isa(o, "GetSegmentVersionsResponse");
    };
})(GetSegmentVersionsResponse || (GetSegmentVersionsResponse = {}));
export var GetSegmentsRequest;
(function (GetSegmentsRequest) {
    GetSegmentsRequest.isa = function (o) {
        return __isa(o, "GetSegmentsRequest");
    };
})(GetSegmentsRequest || (GetSegmentsRequest = {}));
export var GetSegmentsResponse;
(function (GetSegmentsResponse) {
    GetSegmentsResponse.isa = function (o) {
        return __isa(o, "GetSegmentsResponse");
    };
})(GetSegmentsResponse || (GetSegmentsResponse = {}));
export var GetSmsChannelRequest;
(function (GetSmsChannelRequest) {
    GetSmsChannelRequest.isa = function (o) {
        return __isa(o, "GetSmsChannelRequest");
    };
})(GetSmsChannelRequest || (GetSmsChannelRequest = {}));
export var GetSmsChannelResponse;
(function (GetSmsChannelResponse) {
    GetSmsChannelResponse.isa = function (o) {
        return __isa(o, "GetSmsChannelResponse");
    };
})(GetSmsChannelResponse || (GetSmsChannelResponse = {}));
export var GetSmsTemplateRequest;
(function (GetSmsTemplateRequest) {
    GetSmsTemplateRequest.isa = function (o) {
        return __isa(o, "GetSmsTemplateRequest");
    };
})(GetSmsTemplateRequest || (GetSmsTemplateRequest = {}));
export var GetSmsTemplateResponse;
(function (GetSmsTemplateResponse) {
    GetSmsTemplateResponse.isa = function (o) {
        return __isa(o, "GetSmsTemplateResponse");
    };
})(GetSmsTemplateResponse || (GetSmsTemplateResponse = {}));
export var GetUserEndpointsRequest;
(function (GetUserEndpointsRequest) {
    GetUserEndpointsRequest.isa = function (o) {
        return __isa(o, "GetUserEndpointsRequest");
    };
})(GetUserEndpointsRequest || (GetUserEndpointsRequest = {}));
export var GetUserEndpointsResponse;
(function (GetUserEndpointsResponse) {
    GetUserEndpointsResponse.isa = function (o) {
        return __isa(o, "GetUserEndpointsResponse");
    };
})(GetUserEndpointsResponse || (GetUserEndpointsResponse = {}));
export var GetVoiceChannelRequest;
(function (GetVoiceChannelRequest) {
    GetVoiceChannelRequest.isa = function (o) {
        return __isa(o, "GetVoiceChannelRequest");
    };
})(GetVoiceChannelRequest || (GetVoiceChannelRequest = {}));
export var GetVoiceChannelResponse;
(function (GetVoiceChannelResponse) {
    GetVoiceChannelResponse.isa = function (o) {
        return __isa(o, "GetVoiceChannelResponse");
    };
})(GetVoiceChannelResponse || (GetVoiceChannelResponse = {}));
export var GetVoiceTemplateRequest;
(function (GetVoiceTemplateRequest) {
    GetVoiceTemplateRequest.isa = function (o) {
        return __isa(o, "GetVoiceTemplateRequest");
    };
})(GetVoiceTemplateRequest || (GetVoiceTemplateRequest = {}));
export var GetVoiceTemplateResponse;
(function (GetVoiceTemplateResponse) {
    GetVoiceTemplateResponse.isa = function (o) {
        return __isa(o, "GetVoiceTemplateResponse");
    };
})(GetVoiceTemplateResponse || (GetVoiceTemplateResponse = {}));
export var HoldoutActivity;
(function (HoldoutActivity) {
    HoldoutActivity.isa = function (o) {
        return __isa(o, "HoldoutActivity");
    };
})(HoldoutActivity || (HoldoutActivity = {}));
export var ImportJobRequest;
(function (ImportJobRequest) {
    ImportJobRequest.isa = function (o) {
        return __isa(o, "ImportJobRequest");
    };
})(ImportJobRequest || (ImportJobRequest = {}));
export var ImportJobResource;
(function (ImportJobResource) {
    ImportJobResource.isa = function (o) {
        return __isa(o, "ImportJobResource");
    };
})(ImportJobResource || (ImportJobResource = {}));
export var ImportJobResponse;
(function (ImportJobResponse) {
    ImportJobResponse.isa = function (o) {
        return __isa(o, "ImportJobResponse");
    };
})(ImportJobResponse || (ImportJobResponse = {}));
export var ImportJobsResponse;
(function (ImportJobsResponse) {
    ImportJobsResponse.isa = function (o) {
        return __isa(o, "ImportJobsResponse");
    };
})(ImportJobsResponse || (ImportJobsResponse = {}));
export var Include;
(function (Include) {
    Include["ALL"] = "ALL";
    Include["ANY"] = "ANY";
    Include["NONE"] = "NONE";
})(Include || (Include = {}));
export var InternalServerErrorException;
(function (InternalServerErrorException) {
    InternalServerErrorException.isa = function (o) {
        return __isa(o, "InternalServerErrorException");
    };
})(InternalServerErrorException || (InternalServerErrorException = {}));
export var ItemResponse;
(function (ItemResponse) {
    ItemResponse.isa = function (o) { return __isa(o, "ItemResponse"); };
})(ItemResponse || (ItemResponse = {}));
export var JobStatus;
(function (JobStatus) {
    JobStatus["COMPLETED"] = "COMPLETED";
    JobStatus["COMPLETING"] = "COMPLETING";
    JobStatus["CREATED"] = "CREATED";
    JobStatus["FAILED"] = "FAILED";
    JobStatus["FAILING"] = "FAILING";
    JobStatus["INITIALIZING"] = "INITIALIZING";
    JobStatus["PENDING_JOB"] = "PENDING_JOB";
    JobStatus["PREPARING_FOR_INITIALIZATION"] = "PREPARING_FOR_INITIALIZATION";
    JobStatus["PROCESSING"] = "PROCESSING";
})(JobStatus || (JobStatus = {}));
export var JourneyDateRangeKpiResponse;
(function (JourneyDateRangeKpiResponse) {
    JourneyDateRangeKpiResponse.isa = function (o) {
        return __isa(o, "JourneyDateRangeKpiResponse");
    };
})(JourneyDateRangeKpiResponse || (JourneyDateRangeKpiResponse = {}));
export var JourneyEmailMessage;
(function (JourneyEmailMessage) {
    JourneyEmailMessage.isa = function (o) {
        return __isa(o, "JourneyEmailMessage");
    };
})(JourneyEmailMessage || (JourneyEmailMessage = {}));
export var JourneyExecutionActivityMetricsResponse;
(function (JourneyExecutionActivityMetricsResponse) {
    JourneyExecutionActivityMetricsResponse.isa = function (o) {
        return __isa(o, "JourneyExecutionActivityMetricsResponse");
    };
})(JourneyExecutionActivityMetricsResponse || (JourneyExecutionActivityMetricsResponse = {}));
export var JourneyExecutionMetricsResponse;
(function (JourneyExecutionMetricsResponse) {
    JourneyExecutionMetricsResponse.isa = function (o) {
        return __isa(o, "JourneyExecutionMetricsResponse");
    };
})(JourneyExecutionMetricsResponse || (JourneyExecutionMetricsResponse = {}));
export var JourneyLimits;
(function (JourneyLimits) {
    JourneyLimits.isa = function (o) { return __isa(o, "JourneyLimits"); };
})(JourneyLimits || (JourneyLimits = {}));
export var JourneyResponse;
(function (JourneyResponse) {
    JourneyResponse.isa = function (o) {
        return __isa(o, "JourneyResponse");
    };
})(JourneyResponse || (JourneyResponse = {}));
export var JourneySchedule;
(function (JourneySchedule) {
    JourneySchedule.isa = function (o) {
        return __isa(o, "JourneySchedule");
    };
})(JourneySchedule || (JourneySchedule = {}));
export var JourneyStateRequest;
(function (JourneyStateRequest) {
    JourneyStateRequest.isa = function (o) {
        return __isa(o, "JourneyStateRequest");
    };
})(JourneyStateRequest || (JourneyStateRequest = {}));
export var JourneysResponse;
(function (JourneysResponse) {
    JourneysResponse.isa = function (o) {
        return __isa(o, "JourneysResponse");
    };
})(JourneysResponse || (JourneysResponse = {}));
export var ListJourneysRequest;
(function (ListJourneysRequest) {
    ListJourneysRequest.isa = function (o) {
        return __isa(o, "ListJourneysRequest");
    };
})(ListJourneysRequest || (ListJourneysRequest = {}));
export var ListJourneysResponse;
(function (ListJourneysResponse) {
    ListJourneysResponse.isa = function (o) {
        return __isa(o, "ListJourneysResponse");
    };
})(ListJourneysResponse || (ListJourneysResponse = {}));
export var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.isa = function (o) {
        return __isa(o, "ListTagsForResourceRequest");
    };
})(ListTagsForResourceRequest || (ListTagsForResourceRequest = {}));
export var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.isa = function (o) {
        return __isa(o, "ListTagsForResourceResponse");
    };
})(ListTagsForResourceResponse || (ListTagsForResourceResponse = {}));
export var ListTemplateVersionsRequest;
(function (ListTemplateVersionsRequest) {
    ListTemplateVersionsRequest.isa = function (o) {
        return __isa(o, "ListTemplateVersionsRequest");
    };
})(ListTemplateVersionsRequest || (ListTemplateVersionsRequest = {}));
export var ListTemplateVersionsResponse;
(function (ListTemplateVersionsResponse) {
    ListTemplateVersionsResponse.isa = function (o) {
        return __isa(o, "ListTemplateVersionsResponse");
    };
})(ListTemplateVersionsResponse || (ListTemplateVersionsResponse = {}));
export var ListTemplatesRequest;
(function (ListTemplatesRequest) {
    ListTemplatesRequest.isa = function (o) {
        return __isa(o, "ListTemplatesRequest");
    };
})(ListTemplatesRequest || (ListTemplatesRequest = {}));
export var ListTemplatesResponse;
(function (ListTemplatesResponse) {
    ListTemplatesResponse.isa = function (o) {
        return __isa(o, "ListTemplatesResponse");
    };
})(ListTemplatesResponse || (ListTemplatesResponse = {}));
export var Message;
(function (Message) {
    Message.isa = function (o) { return __isa(o, "Message"); };
})(Message || (Message = {}));
export var MessageBody;
(function (MessageBody) {
    MessageBody.isa = function (o) { return __isa(o, "MessageBody"); };
})(MessageBody || (MessageBody = {}));
export var MessageConfiguration;
(function (MessageConfiguration) {
    MessageConfiguration.isa = function (o) {
        return __isa(o, "MessageConfiguration");
    };
})(MessageConfiguration || (MessageConfiguration = {}));
export var MessageRequest;
(function (MessageRequest) {
    MessageRequest.isa = function (o) {
        return __isa(o, "MessageRequest");
    };
})(MessageRequest || (MessageRequest = {}));
export var MessageResponse;
(function (MessageResponse) {
    MessageResponse.isa = function (o) {
        return __isa(o, "MessageResponse");
    };
})(MessageResponse || (MessageResponse = {}));
export var MessageResult;
(function (MessageResult) {
    MessageResult.isa = function (o) { return __isa(o, "MessageResult"); };
})(MessageResult || (MessageResult = {}));
export var MessageType;
(function (MessageType) {
    MessageType["PROMOTIONAL"] = "PROMOTIONAL";
    MessageType["TRANSACTIONAL"] = "TRANSACTIONAL";
})(MessageType || (MessageType = {}));
export var MethodNotAllowedException;
(function (MethodNotAllowedException) {
    MethodNotAllowedException.isa = function (o) {
        return __isa(o, "MethodNotAllowedException");
    };
})(MethodNotAllowedException || (MethodNotAllowedException = {}));
export var MetricDimension;
(function (MetricDimension) {
    MetricDimension.isa = function (o) {
        return __isa(o, "MetricDimension");
    };
})(MetricDimension || (MetricDimension = {}));
export var Mode;
(function (Mode) {
    Mode["DELIVERY"] = "DELIVERY";
    Mode["FILTER"] = "FILTER";
})(Mode || (Mode = {}));
export var MultiConditionalBranch;
(function (MultiConditionalBranch) {
    MultiConditionalBranch.isa = function (o) {
        return __isa(o, "MultiConditionalBranch");
    };
})(MultiConditionalBranch || (MultiConditionalBranch = {}));
export var MultiConditionalSplitActivity;
(function (MultiConditionalSplitActivity) {
    MultiConditionalSplitActivity.isa = function (o) {
        return __isa(o, "MultiConditionalSplitActivity");
    };
})(MultiConditionalSplitActivity || (MultiConditionalSplitActivity = {}));
export var NotFoundException;
(function (NotFoundException) {
    NotFoundException.isa = function (o) {
        return __isa(o, "NotFoundException");
    };
})(NotFoundException || (NotFoundException = {}));
export var NumberValidateRequest;
(function (NumberValidateRequest) {
    NumberValidateRequest.isa = function (o) {
        return __isa(o, "NumberValidateRequest");
    };
})(NumberValidateRequest || (NumberValidateRequest = {}));
export var NumberValidateResponse;
(function (NumberValidateResponse) {
    NumberValidateResponse.isa = function (o) {
        return __isa(o, "NumberValidateResponse");
    };
})(NumberValidateResponse || (NumberValidateResponse = {}));
export var Operator;
(function (Operator) {
    Operator["ALL"] = "ALL";
    Operator["ANY"] = "ANY";
})(Operator || (Operator = {}));
export var PhoneNumberValidateRequest;
(function (PhoneNumberValidateRequest) {
    PhoneNumberValidateRequest.isa = function (o) {
        return __isa(o, "PhoneNumberValidateRequest");
    };
})(PhoneNumberValidateRequest || (PhoneNumberValidateRequest = {}));
export var PhoneNumberValidateResponse;
(function (PhoneNumberValidateResponse) {
    PhoneNumberValidateResponse.isa = function (o) {
        return __isa(o, "PhoneNumberValidateResponse");
    };
})(PhoneNumberValidateResponse || (PhoneNumberValidateResponse = {}));
export var PublicEndpoint;
(function (PublicEndpoint) {
    PublicEndpoint.isa = function (o) {
        return __isa(o, "PublicEndpoint");
    };
})(PublicEndpoint || (PublicEndpoint = {}));
export var PushNotificationTemplateRequest;
(function (PushNotificationTemplateRequest) {
    PushNotificationTemplateRequest.isa = function (o) {
        return __isa(o, "PushNotificationTemplateRequest");
    };
})(PushNotificationTemplateRequest || (PushNotificationTemplateRequest = {}));
export var PushNotificationTemplateResponse;
(function (PushNotificationTemplateResponse) {
    PushNotificationTemplateResponse.isa = function (o) {
        return __isa(o, "PushNotificationTemplateResponse");
    };
})(PushNotificationTemplateResponse || (PushNotificationTemplateResponse = {}));
export var PutEventStreamRequest;
(function (PutEventStreamRequest) {
    PutEventStreamRequest.isa = function (o) {
        return __isa(o, "PutEventStreamRequest");
    };
})(PutEventStreamRequest || (PutEventStreamRequest = {}));
export var PutEventStreamResponse;
(function (PutEventStreamResponse) {
    PutEventStreamResponse.isa = function (o) {
        return __isa(o, "PutEventStreamResponse");
    };
})(PutEventStreamResponse || (PutEventStreamResponse = {}));
export var PutEventsRequest;
(function (PutEventsRequest) {
    PutEventsRequest.isa = function (o) {
        return __isa(o, "PutEventsRequest");
    };
})(PutEventsRequest || (PutEventsRequest = {}));
export var PutEventsResponse;
(function (PutEventsResponse) {
    PutEventsResponse.isa = function (o) {
        return __isa(o, "PutEventsResponse");
    };
})(PutEventsResponse || (PutEventsResponse = {}));
export var QuietTime;
(function (QuietTime) {
    QuietTime.isa = function (o) { return __isa(o, "QuietTime"); };
})(QuietTime || (QuietTime = {}));
export var RandomSplitActivity;
(function (RandomSplitActivity) {
    RandomSplitActivity.isa = function (o) {
        return __isa(o, "RandomSplitActivity");
    };
})(RandomSplitActivity || (RandomSplitActivity = {}));
export var RandomSplitEntry;
(function (RandomSplitEntry) {
    RandomSplitEntry.isa = function (o) {
        return __isa(o, "RandomSplitEntry");
    };
})(RandomSplitEntry || (RandomSplitEntry = {}));
export var RawEmail;
(function (RawEmail) {
    RawEmail.isa = function (o) { return __isa(o, "RawEmail"); };
})(RawEmail || (RawEmail = {}));
export var RecencyDimension;
(function (RecencyDimension) {
    RecencyDimension.isa = function (o) {
        return __isa(o, "RecencyDimension");
    };
})(RecencyDimension || (RecencyDimension = {}));
export var RecencyType;
(function (RecencyType) {
    RecencyType["ACTIVE"] = "ACTIVE";
    RecencyType["INACTIVE"] = "INACTIVE";
})(RecencyType || (RecencyType = {}));
export var RemoveAttributesRequest;
(function (RemoveAttributesRequest) {
    RemoveAttributesRequest.isa = function (o) {
        return __isa(o, "RemoveAttributesRequest");
    };
})(RemoveAttributesRequest || (RemoveAttributesRequest = {}));
export var RemoveAttributesResponse;
(function (RemoveAttributesResponse) {
    RemoveAttributesResponse.isa = function (o) {
        return __isa(o, "RemoveAttributesResponse");
    };
})(RemoveAttributesResponse || (RemoveAttributesResponse = {}));
export var ResultRow;
(function (ResultRow) {
    ResultRow.isa = function (o) { return __isa(o, "ResultRow"); };
})(ResultRow || (ResultRow = {}));
export var ResultRowValue;
(function (ResultRowValue) {
    ResultRowValue.isa = function (o) {
        return __isa(o, "ResultRowValue");
    };
})(ResultRowValue || (ResultRowValue = {}));
export var SMSChannelRequest;
(function (SMSChannelRequest) {
    SMSChannelRequest.isa = function (o) {
        return __isa(o, "SMSChannelRequest");
    };
})(SMSChannelRequest || (SMSChannelRequest = {}));
export var SMSChannelResponse;
(function (SMSChannelResponse) {
    SMSChannelResponse.isa = function (o) {
        return __isa(o, "SMSChannelResponse");
    };
})(SMSChannelResponse || (SMSChannelResponse = {}));
export var SMSMessage;
(function (SMSMessage) {
    SMSMessage.isa = function (o) { return __isa(o, "SMSMessage"); };
})(SMSMessage || (SMSMessage = {}));
export var SMSTemplateRequest;
(function (SMSTemplateRequest) {
    SMSTemplateRequest.isa = function (o) {
        return __isa(o, "SMSTemplateRequest");
    };
})(SMSTemplateRequest || (SMSTemplateRequest = {}));
export var SMSTemplateResponse;
(function (SMSTemplateResponse) {
    SMSTemplateResponse.isa = function (o) {
        return __isa(o, "SMSTemplateResponse");
    };
})(SMSTemplateResponse || (SMSTemplateResponse = {}));
export var Schedule;
(function (Schedule) {
    Schedule.isa = function (o) { return __isa(o, "Schedule"); };
})(Schedule || (Schedule = {}));
export var SegmentBehaviors;
(function (SegmentBehaviors) {
    SegmentBehaviors.isa = function (o) {
        return __isa(o, "SegmentBehaviors");
    };
})(SegmentBehaviors || (SegmentBehaviors = {}));
export var SegmentCondition;
(function (SegmentCondition) {
    SegmentCondition.isa = function (o) {
        return __isa(o, "SegmentCondition");
    };
})(SegmentCondition || (SegmentCondition = {}));
export var SegmentDemographics;
(function (SegmentDemographics) {
    SegmentDemographics.isa = function (o) {
        return __isa(o, "SegmentDemographics");
    };
})(SegmentDemographics || (SegmentDemographics = {}));
export var SegmentDimensions;
(function (SegmentDimensions) {
    SegmentDimensions.isa = function (o) {
        return __isa(o, "SegmentDimensions");
    };
})(SegmentDimensions || (SegmentDimensions = {}));
export var SegmentGroup;
(function (SegmentGroup) {
    SegmentGroup.isa = function (o) { return __isa(o, "SegmentGroup"); };
})(SegmentGroup || (SegmentGroup = {}));
export var SegmentGroupList;
(function (SegmentGroupList) {
    SegmentGroupList.isa = function (o) {
        return __isa(o, "SegmentGroupList");
    };
})(SegmentGroupList || (SegmentGroupList = {}));
export var SegmentImportResource;
(function (SegmentImportResource) {
    SegmentImportResource.isa = function (o) {
        return __isa(o, "SegmentImportResource");
    };
})(SegmentImportResource || (SegmentImportResource = {}));
export var SegmentLocation;
(function (SegmentLocation) {
    SegmentLocation.isa = function (o) {
        return __isa(o, "SegmentLocation");
    };
})(SegmentLocation || (SegmentLocation = {}));
export var SegmentReference;
(function (SegmentReference) {
    SegmentReference.isa = function (o) {
        return __isa(o, "SegmentReference");
    };
})(SegmentReference || (SegmentReference = {}));
export var SegmentResponse;
(function (SegmentResponse) {
    SegmentResponse.isa = function (o) {
        return __isa(o, "SegmentResponse");
    };
})(SegmentResponse || (SegmentResponse = {}));
export var SegmentType;
(function (SegmentType) {
    SegmentType["DIMENSIONAL"] = "DIMENSIONAL";
    SegmentType["IMPORT"] = "IMPORT";
})(SegmentType || (SegmentType = {}));
export var SegmentsResponse;
(function (SegmentsResponse) {
    SegmentsResponse.isa = function (o) {
        return __isa(o, "SegmentsResponse");
    };
})(SegmentsResponse || (SegmentsResponse = {}));
export var SendMessagesRequest;
(function (SendMessagesRequest) {
    SendMessagesRequest.isa = function (o) {
        return __isa(o, "SendMessagesRequest");
    };
})(SendMessagesRequest || (SendMessagesRequest = {}));
export var SendMessagesResponse;
(function (SendMessagesResponse) {
    SendMessagesResponse.isa = function (o) {
        return __isa(o, "SendMessagesResponse");
    };
})(SendMessagesResponse || (SendMessagesResponse = {}));
export var SendUsersMessageRequest;
(function (SendUsersMessageRequest) {
    SendUsersMessageRequest.isa = function (o) {
        return __isa(o, "SendUsersMessageRequest");
    };
})(SendUsersMessageRequest || (SendUsersMessageRequest = {}));
export var SendUsersMessageResponse;
(function (SendUsersMessageResponse) {
    SendUsersMessageResponse.isa = function (o) {
        return __isa(o, "SendUsersMessageResponse");
    };
})(SendUsersMessageResponse || (SendUsersMessageResponse = {}));
export var SendUsersMessagesRequest;
(function (SendUsersMessagesRequest) {
    SendUsersMessagesRequest.isa = function (o) {
        return __isa(o, "SendUsersMessagesRequest");
    };
})(SendUsersMessagesRequest || (SendUsersMessagesRequest = {}));
export var SendUsersMessagesResponse;
(function (SendUsersMessagesResponse) {
    SendUsersMessagesResponse.isa = function (o) {
        return __isa(o, "SendUsersMessagesResponse");
    };
})(SendUsersMessagesResponse || (SendUsersMessagesResponse = {}));
export var Session;
(function (Session) {
    Session.isa = function (o) { return __isa(o, "Session"); };
})(Session || (Session = {}));
export var SetDimension;
(function (SetDimension) {
    SetDimension.isa = function (o) { return __isa(o, "SetDimension"); };
})(SetDimension || (SetDimension = {}));
export var SimpleCondition;
(function (SimpleCondition) {
    SimpleCondition.isa = function (o) {
        return __isa(o, "SimpleCondition");
    };
})(SimpleCondition || (SimpleCondition = {}));
export var SimpleEmail;
(function (SimpleEmail) {
    SimpleEmail.isa = function (o) { return __isa(o, "SimpleEmail"); };
})(SimpleEmail || (SimpleEmail = {}));
export var SimpleEmailPart;
(function (SimpleEmailPart) {
    SimpleEmailPart.isa = function (o) {
        return __isa(o, "SimpleEmailPart");
    };
})(SimpleEmailPart || (SimpleEmailPart = {}));
export var SourceType;
(function (SourceType) {
    SourceType["ALL"] = "ALL";
    SourceType["ANY"] = "ANY";
    SourceType["NONE"] = "NONE";
})(SourceType || (SourceType = {}));
export var StartCondition;
(function (StartCondition) {
    StartCondition.isa = function (o) {
        return __isa(o, "StartCondition");
    };
})(StartCondition || (StartCondition = {}));
export var State;
(function (State) {
    State["ACTIVE"] = "ACTIVE";
    State["CANCELLED"] = "CANCELLED";
    State["CLOSED"] = "CLOSED";
    State["COMPLETED"] = "COMPLETED";
    State["DRAFT"] = "DRAFT";
})(State || (State = {}));
export var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.isa = function (o) {
        return __isa(o, "TagResourceRequest");
    };
})(TagResourceRequest || (TagResourceRequest = {}));
export var TagsModel;
(function (TagsModel) {
    TagsModel.isa = function (o) { return __isa(o, "TagsModel"); };
})(TagsModel || (TagsModel = {}));
export var Template;
(function (Template) {
    Template.isa = function (o) { return __isa(o, "Template"); };
})(Template || (Template = {}));
export var TemplateActiveVersionRequest;
(function (TemplateActiveVersionRequest) {
    TemplateActiveVersionRequest.isa = function (o) {
        return __isa(o, "TemplateActiveVersionRequest");
    };
})(TemplateActiveVersionRequest || (TemplateActiveVersionRequest = {}));
export var TemplateConfiguration;
(function (TemplateConfiguration) {
    TemplateConfiguration.isa = function (o) {
        return __isa(o, "TemplateConfiguration");
    };
})(TemplateConfiguration || (TemplateConfiguration = {}));
export var TemplateResponse;
(function (TemplateResponse) {
    TemplateResponse.isa = function (o) {
        return __isa(o, "TemplateResponse");
    };
})(TemplateResponse || (TemplateResponse = {}));
export var TemplateType;
(function (TemplateType) {
    TemplateType["EMAIL"] = "EMAIL";
    TemplateType["PUSH"] = "PUSH";
    TemplateType["SMS"] = "SMS";
    TemplateType["VOICE"] = "VOICE";
})(TemplateType || (TemplateType = {}));
export var TemplateVersionResponse;
(function (TemplateVersionResponse) {
    TemplateVersionResponse.isa = function (o) {
        return __isa(o, "TemplateVersionResponse");
    };
})(TemplateVersionResponse || (TemplateVersionResponse = {}));
export var TemplateVersionsResponse;
(function (TemplateVersionsResponse) {
    TemplateVersionsResponse.isa = function (o) {
        return __isa(o, "TemplateVersionsResponse");
    };
})(TemplateVersionsResponse || (TemplateVersionsResponse = {}));
export var TemplatesResponse;
(function (TemplatesResponse) {
    TemplatesResponse.isa = function (o) {
        return __isa(o, "TemplatesResponse");
    };
})(TemplatesResponse || (TemplatesResponse = {}));
export var TooManyRequestsException;
(function (TooManyRequestsException) {
    TooManyRequestsException.isa = function (o) {
        return __isa(o, "TooManyRequestsException");
    };
})(TooManyRequestsException || (TooManyRequestsException = {}));
export var TreatmentResource;
(function (TreatmentResource) {
    TreatmentResource.isa = function (o) {
        return __isa(o, "TreatmentResource");
    };
})(TreatmentResource || (TreatmentResource = {}));
export var Type;
(function (Type) {
    Type["ALL"] = "ALL";
    Type["ANY"] = "ANY";
    Type["NONE"] = "NONE";
})(Type || (Type = {}));
export var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.isa = function (o) {
        return __isa(o, "UntagResourceRequest");
    };
})(UntagResourceRequest || (UntagResourceRequest = {}));
export var UpdateAdmChannelRequest;
(function (UpdateAdmChannelRequest) {
    UpdateAdmChannelRequest.isa = function (o) {
        return __isa(o, "UpdateAdmChannelRequest");
    };
})(UpdateAdmChannelRequest || (UpdateAdmChannelRequest = {}));
export var UpdateAdmChannelResponse;
(function (UpdateAdmChannelResponse) {
    UpdateAdmChannelResponse.isa = function (o) {
        return __isa(o, "UpdateAdmChannelResponse");
    };
})(UpdateAdmChannelResponse || (UpdateAdmChannelResponse = {}));
export var UpdateApnsChannelRequest;
(function (UpdateApnsChannelRequest) {
    UpdateApnsChannelRequest.isa = function (o) {
        return __isa(o, "UpdateApnsChannelRequest");
    };
})(UpdateApnsChannelRequest || (UpdateApnsChannelRequest = {}));
export var UpdateApnsChannelResponse;
(function (UpdateApnsChannelResponse) {
    UpdateApnsChannelResponse.isa = function (o) {
        return __isa(o, "UpdateApnsChannelResponse");
    };
})(UpdateApnsChannelResponse || (UpdateApnsChannelResponse = {}));
export var UpdateApnsSandboxChannelRequest;
(function (UpdateApnsSandboxChannelRequest) {
    UpdateApnsSandboxChannelRequest.isa = function (o) {
        return __isa(o, "UpdateApnsSandboxChannelRequest");
    };
})(UpdateApnsSandboxChannelRequest || (UpdateApnsSandboxChannelRequest = {}));
export var UpdateApnsSandboxChannelResponse;
(function (UpdateApnsSandboxChannelResponse) {
    UpdateApnsSandboxChannelResponse.isa = function (o) {
        return __isa(o, "UpdateApnsSandboxChannelResponse");
    };
})(UpdateApnsSandboxChannelResponse || (UpdateApnsSandboxChannelResponse = {}));
export var UpdateApnsVoipChannelRequest;
(function (UpdateApnsVoipChannelRequest) {
    UpdateApnsVoipChannelRequest.isa = function (o) {
        return __isa(o, "UpdateApnsVoipChannelRequest");
    };
})(UpdateApnsVoipChannelRequest || (UpdateApnsVoipChannelRequest = {}));
export var UpdateApnsVoipChannelResponse;
(function (UpdateApnsVoipChannelResponse) {
    UpdateApnsVoipChannelResponse.isa = function (o) {
        return __isa(o, "UpdateApnsVoipChannelResponse");
    };
})(UpdateApnsVoipChannelResponse || (UpdateApnsVoipChannelResponse = {}));
export var UpdateApnsVoipSandboxChannelRequest;
(function (UpdateApnsVoipSandboxChannelRequest) {
    UpdateApnsVoipSandboxChannelRequest.isa = function (o) {
        return __isa(o, "UpdateApnsVoipSandboxChannelRequest");
    };
})(UpdateApnsVoipSandboxChannelRequest || (UpdateApnsVoipSandboxChannelRequest = {}));
export var UpdateApnsVoipSandboxChannelResponse;
(function (UpdateApnsVoipSandboxChannelResponse) {
    UpdateApnsVoipSandboxChannelResponse.isa = function (o) {
        return __isa(o, "UpdateApnsVoipSandboxChannelResponse");
    };
})(UpdateApnsVoipSandboxChannelResponse || (UpdateApnsVoipSandboxChannelResponse = {}));
export var UpdateApplicationSettingsRequest;
(function (UpdateApplicationSettingsRequest) {
    UpdateApplicationSettingsRequest.isa = function (o) {
        return __isa(o, "UpdateApplicationSettingsRequest");
    };
})(UpdateApplicationSettingsRequest || (UpdateApplicationSettingsRequest = {}));
export var UpdateApplicationSettingsResponse;
(function (UpdateApplicationSettingsResponse) {
    UpdateApplicationSettingsResponse.isa = function (o) {
        return __isa(o, "UpdateApplicationSettingsResponse");
    };
})(UpdateApplicationSettingsResponse || (UpdateApplicationSettingsResponse = {}));
export var UpdateAttributesRequest;
(function (UpdateAttributesRequest) {
    UpdateAttributesRequest.isa = function (o) {
        return __isa(o, "UpdateAttributesRequest");
    };
})(UpdateAttributesRequest || (UpdateAttributesRequest = {}));
export var UpdateBaiduChannelRequest;
(function (UpdateBaiduChannelRequest) {
    UpdateBaiduChannelRequest.isa = function (o) {
        return __isa(o, "UpdateBaiduChannelRequest");
    };
})(UpdateBaiduChannelRequest || (UpdateBaiduChannelRequest = {}));
export var UpdateBaiduChannelResponse;
(function (UpdateBaiduChannelResponse) {
    UpdateBaiduChannelResponse.isa = function (o) {
        return __isa(o, "UpdateBaiduChannelResponse");
    };
})(UpdateBaiduChannelResponse || (UpdateBaiduChannelResponse = {}));
export var UpdateCampaignRequest;
(function (UpdateCampaignRequest) {
    UpdateCampaignRequest.isa = function (o) {
        return __isa(o, "UpdateCampaignRequest");
    };
})(UpdateCampaignRequest || (UpdateCampaignRequest = {}));
export var UpdateCampaignResponse;
(function (UpdateCampaignResponse) {
    UpdateCampaignResponse.isa = function (o) {
        return __isa(o, "UpdateCampaignResponse");
    };
})(UpdateCampaignResponse || (UpdateCampaignResponse = {}));
export var UpdateEmailChannelRequest;
(function (UpdateEmailChannelRequest) {
    UpdateEmailChannelRequest.isa = function (o) {
        return __isa(o, "UpdateEmailChannelRequest");
    };
})(UpdateEmailChannelRequest || (UpdateEmailChannelRequest = {}));
export var UpdateEmailChannelResponse;
(function (UpdateEmailChannelResponse) {
    UpdateEmailChannelResponse.isa = function (o) {
        return __isa(o, "UpdateEmailChannelResponse");
    };
})(UpdateEmailChannelResponse || (UpdateEmailChannelResponse = {}));
export var UpdateEmailTemplateRequest;
(function (UpdateEmailTemplateRequest) {
    UpdateEmailTemplateRequest.isa = function (o) {
        return __isa(o, "UpdateEmailTemplateRequest");
    };
})(UpdateEmailTemplateRequest || (UpdateEmailTemplateRequest = {}));
export var UpdateEmailTemplateResponse;
(function (UpdateEmailTemplateResponse) {
    UpdateEmailTemplateResponse.isa = function (o) {
        return __isa(o, "UpdateEmailTemplateResponse");
    };
})(UpdateEmailTemplateResponse || (UpdateEmailTemplateResponse = {}));
export var UpdateEndpointRequest;
(function (UpdateEndpointRequest) {
    UpdateEndpointRequest.isa = function (o) {
        return __isa(o, "UpdateEndpointRequest");
    };
})(UpdateEndpointRequest || (UpdateEndpointRequest = {}));
export var UpdateEndpointResponse;
(function (UpdateEndpointResponse) {
    UpdateEndpointResponse.isa = function (o) {
        return __isa(o, "UpdateEndpointResponse");
    };
})(UpdateEndpointResponse || (UpdateEndpointResponse = {}));
export var UpdateEndpointsBatchRequest;
(function (UpdateEndpointsBatchRequest) {
    UpdateEndpointsBatchRequest.isa = function (o) {
        return __isa(o, "UpdateEndpointsBatchRequest");
    };
})(UpdateEndpointsBatchRequest || (UpdateEndpointsBatchRequest = {}));
export var UpdateEndpointsBatchResponse;
(function (UpdateEndpointsBatchResponse) {
    UpdateEndpointsBatchResponse.isa = function (o) {
        return __isa(o, "UpdateEndpointsBatchResponse");
    };
})(UpdateEndpointsBatchResponse || (UpdateEndpointsBatchResponse = {}));
export var UpdateGcmChannelRequest;
(function (UpdateGcmChannelRequest) {
    UpdateGcmChannelRequest.isa = function (o) {
        return __isa(o, "UpdateGcmChannelRequest");
    };
})(UpdateGcmChannelRequest || (UpdateGcmChannelRequest = {}));
export var UpdateGcmChannelResponse;
(function (UpdateGcmChannelResponse) {
    UpdateGcmChannelResponse.isa = function (o) {
        return __isa(o, "UpdateGcmChannelResponse");
    };
})(UpdateGcmChannelResponse || (UpdateGcmChannelResponse = {}));
export var UpdateJourneyRequest;
(function (UpdateJourneyRequest) {
    UpdateJourneyRequest.isa = function (o) {
        return __isa(o, "UpdateJourneyRequest");
    };
})(UpdateJourneyRequest || (UpdateJourneyRequest = {}));
export var UpdateJourneyResponse;
(function (UpdateJourneyResponse) {
    UpdateJourneyResponse.isa = function (o) {
        return __isa(o, "UpdateJourneyResponse");
    };
})(UpdateJourneyResponse || (UpdateJourneyResponse = {}));
export var UpdateJourneyStateRequest;
(function (UpdateJourneyStateRequest) {
    UpdateJourneyStateRequest.isa = function (o) {
        return __isa(o, "UpdateJourneyStateRequest");
    };
})(UpdateJourneyStateRequest || (UpdateJourneyStateRequest = {}));
export var UpdateJourneyStateResponse;
(function (UpdateJourneyStateResponse) {
    UpdateJourneyStateResponse.isa = function (o) {
        return __isa(o, "UpdateJourneyStateResponse");
    };
})(UpdateJourneyStateResponse || (UpdateJourneyStateResponse = {}));
export var UpdatePushTemplateRequest;
(function (UpdatePushTemplateRequest) {
    UpdatePushTemplateRequest.isa = function (o) {
        return __isa(o, "UpdatePushTemplateRequest");
    };
})(UpdatePushTemplateRequest || (UpdatePushTemplateRequest = {}));
export var UpdatePushTemplateResponse;
(function (UpdatePushTemplateResponse) {
    UpdatePushTemplateResponse.isa = function (o) {
        return __isa(o, "UpdatePushTemplateResponse");
    };
})(UpdatePushTemplateResponse || (UpdatePushTemplateResponse = {}));
export var UpdateSegmentRequest;
(function (UpdateSegmentRequest) {
    UpdateSegmentRequest.isa = function (o) {
        return __isa(o, "UpdateSegmentRequest");
    };
})(UpdateSegmentRequest || (UpdateSegmentRequest = {}));
export var UpdateSegmentResponse;
(function (UpdateSegmentResponse) {
    UpdateSegmentResponse.isa = function (o) {
        return __isa(o, "UpdateSegmentResponse");
    };
})(UpdateSegmentResponse || (UpdateSegmentResponse = {}));
export var UpdateSmsChannelRequest;
(function (UpdateSmsChannelRequest) {
    UpdateSmsChannelRequest.isa = function (o) {
        return __isa(o, "UpdateSmsChannelRequest");
    };
})(UpdateSmsChannelRequest || (UpdateSmsChannelRequest = {}));
export var UpdateSmsChannelResponse;
(function (UpdateSmsChannelResponse) {
    UpdateSmsChannelResponse.isa = function (o) {
        return __isa(o, "UpdateSmsChannelResponse");
    };
})(UpdateSmsChannelResponse || (UpdateSmsChannelResponse = {}));
export var UpdateSmsTemplateRequest;
(function (UpdateSmsTemplateRequest) {
    UpdateSmsTemplateRequest.isa = function (o) {
        return __isa(o, "UpdateSmsTemplateRequest");
    };
})(UpdateSmsTemplateRequest || (UpdateSmsTemplateRequest = {}));
export var UpdateSmsTemplateResponse;
(function (UpdateSmsTemplateResponse) {
    UpdateSmsTemplateResponse.isa = function (o) {
        return __isa(o, "UpdateSmsTemplateResponse");
    };
})(UpdateSmsTemplateResponse || (UpdateSmsTemplateResponse = {}));
export var UpdateTemplateActiveVersionRequest;
(function (UpdateTemplateActiveVersionRequest) {
    UpdateTemplateActiveVersionRequest.isa = function (o) {
        return __isa(o, "UpdateTemplateActiveVersionRequest");
    };
})(UpdateTemplateActiveVersionRequest || (UpdateTemplateActiveVersionRequest = {}));
export var UpdateTemplateActiveVersionResponse;
(function (UpdateTemplateActiveVersionResponse) {
    UpdateTemplateActiveVersionResponse.isa = function (o) {
        return __isa(o, "UpdateTemplateActiveVersionResponse");
    };
})(UpdateTemplateActiveVersionResponse || (UpdateTemplateActiveVersionResponse = {}));
export var UpdateVoiceChannelRequest;
(function (UpdateVoiceChannelRequest) {
    UpdateVoiceChannelRequest.isa = function (o) {
        return __isa(o, "UpdateVoiceChannelRequest");
    };
})(UpdateVoiceChannelRequest || (UpdateVoiceChannelRequest = {}));
export var UpdateVoiceChannelResponse;
(function (UpdateVoiceChannelResponse) {
    UpdateVoiceChannelResponse.isa = function (o) {
        return __isa(o, "UpdateVoiceChannelResponse");
    };
})(UpdateVoiceChannelResponse || (UpdateVoiceChannelResponse = {}));
export var UpdateVoiceTemplateRequest;
(function (UpdateVoiceTemplateRequest) {
    UpdateVoiceTemplateRequest.isa = function (o) {
        return __isa(o, "UpdateVoiceTemplateRequest");
    };
})(UpdateVoiceTemplateRequest || (UpdateVoiceTemplateRequest = {}));
export var UpdateVoiceTemplateResponse;
(function (UpdateVoiceTemplateResponse) {
    UpdateVoiceTemplateResponse.isa = function (o) {
        return __isa(o, "UpdateVoiceTemplateResponse");
    };
})(UpdateVoiceTemplateResponse || (UpdateVoiceTemplateResponse = {}));
export var VoiceChannelRequest;
(function (VoiceChannelRequest) {
    VoiceChannelRequest.isa = function (o) {
        return __isa(o, "VoiceChannelRequest");
    };
})(VoiceChannelRequest || (VoiceChannelRequest = {}));
export var VoiceChannelResponse;
(function (VoiceChannelResponse) {
    VoiceChannelResponse.isa = function (o) {
        return __isa(o, "VoiceChannelResponse");
    };
})(VoiceChannelResponse || (VoiceChannelResponse = {}));
export var VoiceMessage;
(function (VoiceMessage) {
    VoiceMessage.isa = function (o) { return __isa(o, "VoiceMessage"); };
})(VoiceMessage || (VoiceMessage = {}));
export var VoiceTemplateRequest;
(function (VoiceTemplateRequest) {
    VoiceTemplateRequest.isa = function (o) {
        return __isa(o, "VoiceTemplateRequest");
    };
})(VoiceTemplateRequest || (VoiceTemplateRequest = {}));
export var VoiceTemplateResponse;
(function (VoiceTemplateResponse) {
    VoiceTemplateResponse.isa = function (o) {
        return __isa(o, "VoiceTemplateResponse");
    };
})(VoiceTemplateResponse || (VoiceTemplateResponse = {}));
export var WaitActivity;
(function (WaitActivity) {
    WaitActivity.isa = function (o) { return __isa(o, "WaitActivity"); };
})(WaitActivity || (WaitActivity = {}));
export var WaitTime;
(function (WaitTime) {
    WaitTime.isa = function (o) { return __isa(o, "WaitTime"); };
})(WaitTime || (WaitTime = {}));
export var WriteApplicationSettingsRequest;
(function (WriteApplicationSettingsRequest) {
    WriteApplicationSettingsRequest.isa = function (o) {
        return __isa(o, "WriteApplicationSettingsRequest");
    };
})(WriteApplicationSettingsRequest || (WriteApplicationSettingsRequest = {}));
export var WriteCampaignRequest;
(function (WriteCampaignRequest) {
    WriteCampaignRequest.isa = function (o) {
        return __isa(o, "WriteCampaignRequest");
    };
})(WriteCampaignRequest || (WriteCampaignRequest = {}));
export var WriteEventStream;
(function (WriteEventStream) {
    WriteEventStream.isa = function (o) {
        return __isa(o, "WriteEventStream");
    };
})(WriteEventStream || (WriteEventStream = {}));
export var WriteJourneyRequest;
(function (WriteJourneyRequest) {
    WriteJourneyRequest.isa = function (o) {
        return __isa(o, "WriteJourneyRequest");
    };
})(WriteJourneyRequest || (WriteJourneyRequest = {}));
export var WriteSegmentRequest;
(function (WriteSegmentRequest) {
    WriteSegmentRequest.isa = function (o) {
        return __isa(o, "WriteSegmentRequest");
    };
})(WriteSegmentRequest || (WriteSegmentRequest = {}));
export var WriteTreatmentResource;
(function (WriteTreatmentResource) {
    WriteTreatmentResource.isa = function (o) {
        return __isa(o, "WriteTreatmentResource");
    };
})(WriteTreatmentResource || (WriteTreatmentResource = {}));
