// @aws-sdk/client-kinesis/dist/cjs/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const smithy_client_1 = require("@aws-sdk/smithy-client");
var AddTagsToStreamInput;
(function (AddTagsToStreamInput) {
    AddTagsToStreamInput.isa = (o) => smithy_client_1.isa(o, "AddTagsToStreamInput");
})(AddTagsToStreamInput = exports.AddTagsToStreamInput || (exports.AddTagsToStreamInput = {}));
var Consumer;
(function (Consumer) {
    Consumer.isa = (o) => smithy_client_1.isa(o, "Consumer");
})(Consumer = exports.Consumer || (exports.Consumer = {}));
var ConsumerDescription;
(function (ConsumerDescription) {
    ConsumerDescription.isa = (o) => smithy_client_1.isa(o, "ConsumerDescription");
})(ConsumerDescription = exports.ConsumerDescription || (exports.ConsumerDescription = {}));
var ConsumerStatus;
(function (ConsumerStatus) {
    ConsumerStatus["ACTIVE"] = "ACTIVE";
    ConsumerStatus["CREATING"] = "CREATING";
    ConsumerStatus["DELETING"] = "DELETING";
})(ConsumerStatus = exports.ConsumerStatus || (exports.ConsumerStatus = {}));
var CreateStreamInput;
(function (CreateStreamInput) {
    CreateStreamInput.isa = (o) => smithy_client_1.isa(o, "CreateStreamInput");
})(CreateStreamInput = exports.CreateStreamInput || (exports.CreateStreamInput = {}));
var DecreaseStreamRetentionPeriodInput;
(function (DecreaseStreamRetentionPeriodInput) {
    DecreaseStreamRetentionPeriodInput.isa = (o) => smithy_client_1.isa(o, "DecreaseStreamRetentionPeriodInput");
})(DecreaseStreamRetentionPeriodInput = exports.DecreaseStreamRetentionPeriodInput || (exports.DecreaseStreamRetentionPeriodInput = {}));
var DeleteStreamInput;
(function (DeleteStreamInput) {
    DeleteStreamInput.isa = (o) => smithy_client_1.isa(o, "DeleteStreamInput");
})(DeleteStreamInput = exports.DeleteStreamInput || (exports.DeleteStreamInput = {}));
var DeregisterStreamConsumerInput;
(function (DeregisterStreamConsumerInput) {
    DeregisterStreamConsumerInput.isa = (o) => smithy_client_1.isa(o, "DeregisterStreamConsumerInput");
})(DeregisterStreamConsumerInput = exports.DeregisterStreamConsumerInput || (exports.DeregisterStreamConsumerInput = {}));
var DescribeLimitsInput;
(function (DescribeLimitsInput) {
    DescribeLimitsInput.isa = (o) => smithy_client_1.isa(o, "DescribeLimitsInput");
})(DescribeLimitsInput = exports.DescribeLimitsInput || (exports.DescribeLimitsInput = {}));
var DescribeLimitsOutput;
(function (DescribeLimitsOutput) {
    DescribeLimitsOutput.isa = (o) => smithy_client_1.isa(o, "DescribeLimitsOutput");
})(DescribeLimitsOutput = exports.DescribeLimitsOutput || (exports.DescribeLimitsOutput = {}));
var DescribeStreamConsumerInput;
(function (DescribeStreamConsumerInput) {
    DescribeStreamConsumerInput.isa = (o) => smithy_client_1.isa(o, "DescribeStreamConsumerInput");
})(DescribeStreamConsumerInput = exports.DescribeStreamConsumerInput || (exports.DescribeStreamConsumerInput = {}));
var DescribeStreamConsumerOutput;
(function (DescribeStreamConsumerOutput) {
    DescribeStreamConsumerOutput.isa = (o) => smithy_client_1.isa(o, "DescribeStreamConsumerOutput");
})(DescribeStreamConsumerOutput = exports.DescribeStreamConsumerOutput || (exports.DescribeStreamConsumerOutput = {}));
var DescribeStreamInput;
(function (DescribeStreamInput) {
    DescribeStreamInput.isa = (o) => smithy_client_1.isa(o, "DescribeStreamInput");
})(DescribeStreamInput = exports.DescribeStreamInput || (exports.DescribeStreamInput = {}));
var DescribeStreamOutput;
(function (DescribeStreamOutput) {
    DescribeStreamOutput.isa = (o) => smithy_client_1.isa(o, "DescribeStreamOutput");
})(DescribeStreamOutput = exports.DescribeStreamOutput || (exports.DescribeStreamOutput = {}));
var DescribeStreamSummaryInput;
(function (DescribeStreamSummaryInput) {
    DescribeStreamSummaryInput.isa = (o) => smithy_client_1.isa(o, "DescribeStreamSummaryInput");
})(DescribeStreamSummaryInput = exports.DescribeStreamSummaryInput || (exports.DescribeStreamSummaryInput = {}));
var DescribeStreamSummaryOutput;
(function (DescribeStreamSummaryOutput) {
    DescribeStreamSummaryOutput.isa = (o) => smithy_client_1.isa(o, "DescribeStreamSummaryOutput");
})(DescribeStreamSummaryOutput = exports.DescribeStreamSummaryOutput || (exports.DescribeStreamSummaryOutput = {}));
var DisableEnhancedMonitoringInput;
(function (DisableEnhancedMonitoringInput) {
    DisableEnhancedMonitoringInput.isa = (o) => smithy_client_1.isa(o, "DisableEnhancedMonitoringInput");
})(DisableEnhancedMonitoringInput = exports.DisableEnhancedMonitoringInput || (exports.DisableEnhancedMonitoringInput = {}));
var EnableEnhancedMonitoringInput;
(function (EnableEnhancedMonitoringInput) {
    EnableEnhancedMonitoringInput.isa = (o) => smithy_client_1.isa(o, "EnableEnhancedMonitoringInput");
})(EnableEnhancedMonitoringInput = exports.EnableEnhancedMonitoringInput || (exports.EnableEnhancedMonitoringInput = {}));
var EncryptionType;
(function (EncryptionType) {
    EncryptionType["KMS"] = "KMS";
    EncryptionType["NONE"] = "NONE";
})(EncryptionType = exports.EncryptionType || (exports.EncryptionType = {}));
var EnhancedMetrics;
(function (EnhancedMetrics) {
    EnhancedMetrics.isa = (o) => smithy_client_1.isa(o, "EnhancedMetrics");
})(EnhancedMetrics = exports.EnhancedMetrics || (exports.EnhancedMetrics = {}));
var EnhancedMonitoringOutput;
(function (EnhancedMonitoringOutput) {
    EnhancedMonitoringOutput.isa = (o) => smithy_client_1.isa(o, "EnhancedMonitoringOutput");
})(EnhancedMonitoringOutput = exports.EnhancedMonitoringOutput || (exports.EnhancedMonitoringOutput = {}));
var ExpiredIteratorException;
(function (ExpiredIteratorException) {
    ExpiredIteratorException.isa = (o) => smithy_client_1.isa(o, "ExpiredIteratorException");
})(ExpiredIteratorException = exports.ExpiredIteratorException || (exports.ExpiredIteratorException = {}));
var ExpiredNextTokenException;
(function (ExpiredNextTokenException) {
    ExpiredNextTokenException.isa = (o) => smithy_client_1.isa(o, "ExpiredNextTokenException");
})(ExpiredNextTokenException = exports.ExpiredNextTokenException || (exports.ExpiredNextTokenException = {}));
var GetRecordsInput;
(function (GetRecordsInput) {
    GetRecordsInput.isa = (o) => smithy_client_1.isa(o, "GetRecordsInput");
})(GetRecordsInput = exports.GetRecordsInput || (exports.GetRecordsInput = {}));
var GetRecordsOutput;
(function (GetRecordsOutput) {
    GetRecordsOutput.isa = (o) => smithy_client_1.isa(o, "GetRecordsOutput");
})(GetRecordsOutput = exports.GetRecordsOutput || (exports.GetRecordsOutput = {}));
var GetShardIteratorInput;
(function (GetShardIteratorInput) {
    GetShardIteratorInput.isa = (o) => smithy_client_1.isa(o, "GetShardIteratorInput");
})(GetShardIteratorInput = exports.GetShardIteratorInput || (exports.GetShardIteratorInput = {}));
var GetShardIteratorOutput;
(function (GetShardIteratorOutput) {
    GetShardIteratorOutput.isa = (o) => smithy_client_1.isa(o, "GetShardIteratorOutput");
})(GetShardIteratorOutput = exports.GetShardIteratorOutput || (exports.GetShardIteratorOutput = {}));
var HashKeyRange;
(function (HashKeyRange) {
    HashKeyRange.isa = (o) => smithy_client_1.isa(o, "HashKeyRange");
})(HashKeyRange = exports.HashKeyRange || (exports.HashKeyRange = {}));
var IncreaseStreamRetentionPeriodInput;
(function (IncreaseStreamRetentionPeriodInput) {
    IncreaseStreamRetentionPeriodInput.isa = (o) => smithy_client_1.isa(o, "IncreaseStreamRetentionPeriodInput");
})(IncreaseStreamRetentionPeriodInput = exports.IncreaseStreamRetentionPeriodInput || (exports.IncreaseStreamRetentionPeriodInput = {}));
var InternalFailureException;
(function (InternalFailureException) {
    InternalFailureException.isa = (o) => smithy_client_1.isa(o, "InternalFailureException");
})(InternalFailureException = exports.InternalFailureException || (exports.InternalFailureException = {}));
var InvalidArgumentException;
(function (InvalidArgumentException) {
    InvalidArgumentException.isa = (o) => smithy_client_1.isa(o, "InvalidArgumentException");
})(InvalidArgumentException = exports.InvalidArgumentException || (exports.InvalidArgumentException = {}));
var KMSAccessDeniedException;
(function (KMSAccessDeniedException) {
    KMSAccessDeniedException.isa = (o) => smithy_client_1.isa(o, "KMSAccessDeniedException");
})(KMSAccessDeniedException = exports.KMSAccessDeniedException || (exports.KMSAccessDeniedException = {}));
var KMSDisabledException;
(function (KMSDisabledException) {
    KMSDisabledException.isa = (o) => smithy_client_1.isa(o, "KMSDisabledException");
})(KMSDisabledException = exports.KMSDisabledException || (exports.KMSDisabledException = {}));
var KMSInvalidStateException;
(function (KMSInvalidStateException) {
    KMSInvalidStateException.isa = (o) => smithy_client_1.isa(o, "KMSInvalidStateException");
})(KMSInvalidStateException = exports.KMSInvalidStateException || (exports.KMSInvalidStateException = {}));
var KMSNotFoundException;
(function (KMSNotFoundException) {
    KMSNotFoundException.isa = (o) => smithy_client_1.isa(o, "KMSNotFoundException");
})(KMSNotFoundException = exports.KMSNotFoundException || (exports.KMSNotFoundException = {}));
var KMSOptInRequired;
(function (KMSOptInRequired) {
    KMSOptInRequired.isa = (o) => smithy_client_1.isa(o, "KMSOptInRequired");
})(KMSOptInRequired = exports.KMSOptInRequired || (exports.KMSOptInRequired = {}));
var KMSThrottlingException;
(function (KMSThrottlingException) {
    KMSThrottlingException.isa = (o) => smithy_client_1.isa(o, "KMSThrottlingException");
})(KMSThrottlingException = exports.KMSThrottlingException || (exports.KMSThrottlingException = {}));
var LimitExceededException;
(function (LimitExceededException) {
    LimitExceededException.isa = (o) => smithy_client_1.isa(o, "LimitExceededException");
})(LimitExceededException = exports.LimitExceededException || (exports.LimitExceededException = {}));
var ListShardsInput;
(function (ListShardsInput) {
    ListShardsInput.isa = (o) => smithy_client_1.isa(o, "ListShardsInput");
})(ListShardsInput = exports.ListShardsInput || (exports.ListShardsInput = {}));
var ListShardsOutput;
(function (ListShardsOutput) {
    ListShardsOutput.isa = (o) => smithy_client_1.isa(o, "ListShardsOutput");
})(ListShardsOutput = exports.ListShardsOutput || (exports.ListShardsOutput = {}));
var ListStreamConsumersInput;
(function (ListStreamConsumersInput) {
    ListStreamConsumersInput.isa = (o) => smithy_client_1.isa(o, "ListStreamConsumersInput");
})(ListStreamConsumersInput = exports.ListStreamConsumersInput || (exports.ListStreamConsumersInput = {}));
var ListStreamConsumersOutput;
(function (ListStreamConsumersOutput) {
    ListStreamConsumersOutput.isa = (o) => smithy_client_1.isa(o, "ListStreamConsumersOutput");
})(ListStreamConsumersOutput = exports.ListStreamConsumersOutput || (exports.ListStreamConsumersOutput = {}));
var ListStreamsInput;
(function (ListStreamsInput) {
    ListStreamsInput.isa = (o) => smithy_client_1.isa(o, "ListStreamsInput");
})(ListStreamsInput = exports.ListStreamsInput || (exports.ListStreamsInput = {}));
var ListStreamsOutput;
(function (ListStreamsOutput) {
    ListStreamsOutput.isa = (o) => smithy_client_1.isa(o, "ListStreamsOutput");
})(ListStreamsOutput = exports.ListStreamsOutput || (exports.ListStreamsOutput = {}));
var ListTagsForStreamInput;
(function (ListTagsForStreamInput) {
    ListTagsForStreamInput.isa = (o) => smithy_client_1.isa(o, "ListTagsForStreamInput");
})(ListTagsForStreamInput = exports.ListTagsForStreamInput || (exports.ListTagsForStreamInput = {}));
var ListTagsForStreamOutput;
(function (ListTagsForStreamOutput) {
    ListTagsForStreamOutput.isa = (o) => smithy_client_1.isa(o, "ListTagsForStreamOutput");
})(ListTagsForStreamOutput = exports.ListTagsForStreamOutput || (exports.ListTagsForStreamOutput = {}));
var MergeShardsInput;
(function (MergeShardsInput) {
    MergeShardsInput.isa = (o) => smithy_client_1.isa(o, "MergeShardsInput");
})(MergeShardsInput = exports.MergeShardsInput || (exports.MergeShardsInput = {}));
var MetricsName;
(function (MetricsName) {
    MetricsName["ALL"] = "ALL";
    MetricsName["INCOMING_BYTES"] = "IncomingBytes";
    MetricsName["INCOMING_RECORDS"] = "IncomingRecords";
    MetricsName["ITERATOR_AGE_MILLISECONDS"] = "IteratorAgeMilliseconds";
    MetricsName["OUTGOING_BYTES"] = "OutgoingBytes";
    MetricsName["OUTGOING_RECORDS"] = "OutgoingRecords";
    MetricsName["READ_PROVISIONED_THROUGHPUT_EXCEEDED"] = "ReadProvisionedThroughputExceeded";
    MetricsName["WRITE_PROVISIONED_THROUGHPUT_EXCEEDED"] = "WriteProvisionedThroughputExceeded";
})(MetricsName = exports.MetricsName || (exports.MetricsName = {}));
var ProvisionedThroughputExceededException;
(function (ProvisionedThroughputExceededException) {
    ProvisionedThroughputExceededException.isa = (o) => smithy_client_1.isa(o, "ProvisionedThroughputExceededException");
})(ProvisionedThroughputExceededException = exports.ProvisionedThroughputExceededException || (exports.ProvisionedThroughputExceededException = {}));
var PutRecordInput;
(function (PutRecordInput) {
    PutRecordInput.isa = (o) => smithy_client_1.isa(o, "PutRecordInput");
})(PutRecordInput = exports.PutRecordInput || (exports.PutRecordInput = {}));
var PutRecordOutput;
(function (PutRecordOutput) {
    PutRecordOutput.isa = (o) => smithy_client_1.isa(o, "PutRecordOutput");
})(PutRecordOutput = exports.PutRecordOutput || (exports.PutRecordOutput = {}));
var PutRecordsInput;
(function (PutRecordsInput) {
    PutRecordsInput.isa = (o) => smithy_client_1.isa(o, "PutRecordsInput");
})(PutRecordsInput = exports.PutRecordsInput || (exports.PutRecordsInput = {}));
var PutRecordsOutput;
(function (PutRecordsOutput) {
    PutRecordsOutput.isa = (o) => smithy_client_1.isa(o, "PutRecordsOutput");
})(PutRecordsOutput = exports.PutRecordsOutput || (exports.PutRecordsOutput = {}));
var PutRecordsRequestEntry;
(function (PutRecordsRequestEntry) {
    PutRecordsRequestEntry.isa = (o) => smithy_client_1.isa(o, "PutRecordsRequestEntry");
})(PutRecordsRequestEntry = exports.PutRecordsRequestEntry || (exports.PutRecordsRequestEntry = {}));
var PutRecordsResultEntry;
(function (PutRecordsResultEntry) {
    PutRecordsResultEntry.isa = (o) => smithy_client_1.isa(o, "PutRecordsResultEntry");
})(PutRecordsResultEntry = exports.PutRecordsResultEntry || (exports.PutRecordsResultEntry = {}));
var _Record;
(function (_Record) {
    _Record.isa = (o) => smithy_client_1.isa(o, "Record");
})(_Record = exports._Record || (exports._Record = {}));
var RegisterStreamConsumerInput;
(function (RegisterStreamConsumerInput) {
    RegisterStreamConsumerInput.isa = (o) => smithy_client_1.isa(o, "RegisterStreamConsumerInput");
})(RegisterStreamConsumerInput = exports.RegisterStreamConsumerInput || (exports.RegisterStreamConsumerInput = {}));
var RegisterStreamConsumerOutput;
(function (RegisterStreamConsumerOutput) {
    RegisterStreamConsumerOutput.isa = (o) => smithy_client_1.isa(o, "RegisterStreamConsumerOutput");
})(RegisterStreamConsumerOutput = exports.RegisterStreamConsumerOutput || (exports.RegisterStreamConsumerOutput = {}));
var RemoveTagsFromStreamInput;
(function (RemoveTagsFromStreamInput) {
    RemoveTagsFromStreamInput.isa = (o) => smithy_client_1.isa(o, "RemoveTagsFromStreamInput");
})(RemoveTagsFromStreamInput = exports.RemoveTagsFromStreamInput || (exports.RemoveTagsFromStreamInput = {}));
var ResourceInUseException;
(function (ResourceInUseException) {
    ResourceInUseException.isa = (o) => smithy_client_1.isa(o, "ResourceInUseException");
})(ResourceInUseException = exports.ResourceInUseException || (exports.ResourceInUseException = {}));
var ResourceNotFoundException;
(function (ResourceNotFoundException) {
    ResourceNotFoundException.isa = (o) => smithy_client_1.isa(o, "ResourceNotFoundException");
})(ResourceNotFoundException = exports.ResourceNotFoundException || (exports.ResourceNotFoundException = {}));
var ScalingType;
(function (ScalingType) {
    ScalingType["UNIFORM_SCALING"] = "UNIFORM_SCALING";
})(ScalingType = exports.ScalingType || (exports.ScalingType = {}));
var SequenceNumberRange;
(function (SequenceNumberRange) {
    SequenceNumberRange.isa = (o) => smithy_client_1.isa(o, "SequenceNumberRange");
})(SequenceNumberRange = exports.SequenceNumberRange || (exports.SequenceNumberRange = {}));
var Shard;
(function (Shard) {
    Shard.isa = (o) => smithy_client_1.isa(o, "Shard");
})(Shard = exports.Shard || (exports.Shard = {}));
var ShardIteratorType;
(function (ShardIteratorType) {
    ShardIteratorType["AFTER_SEQUENCE_NUMBER"] = "AFTER_SEQUENCE_NUMBER";
    ShardIteratorType["AT_SEQUENCE_NUMBER"] = "AT_SEQUENCE_NUMBER";
    ShardIteratorType["AT_TIMESTAMP"] = "AT_TIMESTAMP";
    ShardIteratorType["LATEST"] = "LATEST";
    ShardIteratorType["TRIM_HORIZON"] = "TRIM_HORIZON";
})(ShardIteratorType = exports.ShardIteratorType || (exports.ShardIteratorType = {}));
var SplitShardInput;
(function (SplitShardInput) {
    SplitShardInput.isa = (o) => smithy_client_1.isa(o, "SplitShardInput");
})(SplitShardInput = exports.SplitShardInput || (exports.SplitShardInput = {}));
var StartStreamEncryptionInput;
(function (StartStreamEncryptionInput) {
    StartStreamEncryptionInput.isa = (o) => smithy_client_1.isa(o, "StartStreamEncryptionInput");
})(StartStreamEncryptionInput = exports.StartStreamEncryptionInput || (exports.StartStreamEncryptionInput = {}));
var StartingPosition;
(function (StartingPosition) {
    StartingPosition.isa = (o) => smithy_client_1.isa(o, "StartingPosition");
})(StartingPosition = exports.StartingPosition || (exports.StartingPosition = {}));
var StopStreamEncryptionInput;
(function (StopStreamEncryptionInput) {
    StopStreamEncryptionInput.isa = (o) => smithy_client_1.isa(o, "StopStreamEncryptionInput");
})(StopStreamEncryptionInput = exports.StopStreamEncryptionInput || (exports.StopStreamEncryptionInput = {}));
var StreamDescription;
(function (StreamDescription) {
    StreamDescription.isa = (o) => smithy_client_1.isa(o, "StreamDescription");
})(StreamDescription = exports.StreamDescription || (exports.StreamDescription = {}));
var StreamDescriptionSummary;
(function (StreamDescriptionSummary) {
    StreamDescriptionSummary.isa = (o) => smithy_client_1.isa(o, "StreamDescriptionSummary");
})(StreamDescriptionSummary = exports.StreamDescriptionSummary || (exports.StreamDescriptionSummary = {}));
var StreamStatus;
(function (StreamStatus) {
    StreamStatus["ACTIVE"] = "ACTIVE";
    StreamStatus["CREATING"] = "CREATING";
    StreamStatus["DELETING"] = "DELETING";
    StreamStatus["UPDATING"] = "UPDATING";
})(StreamStatus = exports.StreamStatus || (exports.StreamStatus = {}));
var SubscribeToShardEvent;
(function (SubscribeToShardEvent) {
    SubscribeToShardEvent.isa = (o) => smithy_client_1.isa(o, "SubscribeToShardEvent");
})(SubscribeToShardEvent = exports.SubscribeToShardEvent || (exports.SubscribeToShardEvent = {}));
var SubscribeToShardEventStream;
(function (SubscribeToShardEventStream) {
    SubscribeToShardEventStream.visit = (value, visitor) => {
        if (value.InternalFailureException !== undefined)
            return visitor.InternalFailureException(value.InternalFailureException);
        if (value.KMSAccessDeniedException !== undefined)
            return visitor.KMSAccessDeniedException(value.KMSAccessDeniedException);
        if (value.KMSDisabledException !== undefined)
            return visitor.KMSDisabledException(value.KMSDisabledException);
        if (value.KMSInvalidStateException !== undefined)
            return visitor.KMSInvalidStateException(value.KMSInvalidStateException);
        if (value.KMSNotFoundException !== undefined)
            return visitor.KMSNotFoundException(value.KMSNotFoundException);
        if (value.KMSOptInRequired !== undefined)
            return visitor.KMSOptInRequired(value.KMSOptInRequired);
        if (value.KMSThrottlingException !== undefined)
            return visitor.KMSThrottlingException(value.KMSThrottlingException);
        if (value.ResourceInUseException !== undefined)
            return visitor.ResourceInUseException(value.ResourceInUseException);
        if (value.ResourceNotFoundException !== undefined)
            return visitor.ResourceNotFoundException(value.ResourceNotFoundException);
        if (value.SubscribeToShardEvent !== undefined)
            return visitor.SubscribeToShardEvent(value.SubscribeToShardEvent);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(SubscribeToShardEventStream = exports.SubscribeToShardEventStream || (exports.SubscribeToShardEventStream = {}));
var SubscribeToShardInput;
(function (SubscribeToShardInput) {
    SubscribeToShardInput.isa = (o) => smithy_client_1.isa(o, "SubscribeToShardInput");
})(SubscribeToShardInput = exports.SubscribeToShardInput || (exports.SubscribeToShardInput = {}));
var SubscribeToShardOutput;
(function (SubscribeToShardOutput) {
    SubscribeToShardOutput.isa = (o) => smithy_client_1.isa(o, "SubscribeToShardOutput");
})(SubscribeToShardOutput = exports.SubscribeToShardOutput || (exports.SubscribeToShardOutput = {}));
var Tag;
(function (Tag) {
    Tag.isa = (o) => smithy_client_1.isa(o, "Tag");
})(Tag = exports.Tag || (exports.Tag = {}));
var UpdateShardCountInput;
(function (UpdateShardCountInput) {
    UpdateShardCountInput.isa = (o) => smithy_client_1.isa(o, "UpdateShardCountInput");
})(UpdateShardCountInput = exports.UpdateShardCountInput || (exports.UpdateShardCountInput = {}));
var UpdateShardCountOutput;
(function (UpdateShardCountOutput) {
    UpdateShardCountOutput.isa = (o) => smithy_client_1.isa(o, "UpdateShardCountOutput");
})(UpdateShardCountOutput = exports.UpdateShardCountOutput || (exports.UpdateShardCountOutput = {}));
