// @aws-sdk/client-lex-runtime-service/dist/es/models/index.js
import { isa as __isa } from "@aws-sdk/smithy-client";
export var BadRequestException;
(function (BadRequestException) {
    BadRequestException.isa = function (o) {
        return __isa(o, "BadRequestException");
    };
})(BadRequestException || (BadRequestException = {}));
export var ConflictException;
(function (ConflictException) {
    ConflictException.isa = function (o) {
        return __isa(o, "ConflictException");
    };
})(ConflictException || (ConflictException = {}));
export var InternalFailureException;
(function (InternalFailureException) {
    InternalFailureException.isa = function (o) {
        return __isa(o, "InternalFailureException");
    };
})(InternalFailureException || (InternalFailureException = {}));
export var LimitExceededException;
(function (LimitExceededException) {
    LimitExceededException.isa = function (o) {
        return __isa(o, "LimitExceededException");
    };
})(LimitExceededException || (LimitExceededException = {}));
export var NotAcceptableException;
(function (NotAcceptableException) {
    NotAcceptableException.isa = function (o) {
        return __isa(o, "NotAcceptableException");
    };
})(NotAcceptableException || (NotAcceptableException = {}));
export var NotFoundException;
(function (NotFoundException) {
    NotFoundException.isa = function (o) {
        return __isa(o, "NotFoundException");
    };
})(NotFoundException || (NotFoundException = {}));
export var RequestTimeoutException;
(function (RequestTimeoutException) {
    RequestTimeoutException.isa = function (o) {
        return __isa(o, "RequestTimeoutException");
    };
})(RequestTimeoutException || (RequestTimeoutException = {}));
export var UnsupportedMediaTypeException;
(function (UnsupportedMediaTypeException) {
    UnsupportedMediaTypeException.isa = function (o) {
        return __isa(o, "UnsupportedMediaTypeException");
    };
})(UnsupportedMediaTypeException || (UnsupportedMediaTypeException = {}));
export var BadGatewayException;
(function (BadGatewayException) {
    BadGatewayException.isa = function (o) {
        return __isa(o, "BadGatewayException");
    };
})(BadGatewayException || (BadGatewayException = {}));
export var ConfirmationStatus;
(function (ConfirmationStatus) {
    ConfirmationStatus["CONFIRMED"] = "Confirmed";
    ConfirmationStatus["DENIED"] = "Denied";
    ConfirmationStatus["NONE"] = "None";
})(ConfirmationStatus || (ConfirmationStatus = {}));
export var DeleteSessionRequest;
(function (DeleteSessionRequest) {
    DeleteSessionRequest.isa = function (o) {
        return __isa(o, "DeleteSessionRequest");
    };
})(DeleteSessionRequest || (DeleteSessionRequest = {}));
export var DeleteSessionResponse;
(function (DeleteSessionResponse) {
    DeleteSessionResponse.isa = function (o) {
        return __isa(o, "DeleteSessionResponse");
    };
})(DeleteSessionResponse || (DeleteSessionResponse = {}));
export var DependencyFailedException;
(function (DependencyFailedException) {
    DependencyFailedException.isa = function (o) {
        return __isa(o, "DependencyFailedException");
    };
})(DependencyFailedException || (DependencyFailedException = {}));
export var DialogAction;
(function (DialogAction) {
    DialogAction.isa = function (o) { return __isa(o, "DialogAction"); };
})(DialogAction || (DialogAction = {}));
export var DialogActionType;
(function (DialogActionType) {
    DialogActionType["CLOSE"] = "Close";
    DialogActionType["CONFIRM_INTENT"] = "ConfirmIntent";
    DialogActionType["DELEGATE"] = "Delegate";
    DialogActionType["ELICIT_INTENT"] = "ElicitIntent";
    DialogActionType["ELICIT_SLOT"] = "ElicitSlot";
})(DialogActionType || (DialogActionType = {}));
export var DialogState;
(function (DialogState) {
    DialogState["CONFIRM_INTENT"] = "ConfirmIntent";
    DialogState["ELICIT_INTENT"] = "ElicitIntent";
    DialogState["ELICIT_SLOT"] = "ElicitSlot";
    DialogState["FAILED"] = "Failed";
    DialogState["FULFILLED"] = "Fulfilled";
    DialogState["READY_FOR_FULFILLMENT"] = "ReadyForFulfillment";
})(DialogState || (DialogState = {}));
export var FulfillmentState;
(function (FulfillmentState) {
    FulfillmentState["FAILED"] = "Failed";
    FulfillmentState["FULFILLED"] = "Fulfilled";
    FulfillmentState["READY_FOR_FULFILLMENT"] = "ReadyForFulfillment";
})(FulfillmentState || (FulfillmentState = {}));
export var GetSessionRequest;
(function (GetSessionRequest) {
    GetSessionRequest.isa = function (o) {
        return __isa(o, "GetSessionRequest");
    };
})(GetSessionRequest || (GetSessionRequest = {}));
export var GetSessionResponse;
(function (GetSessionResponse) {
    GetSessionResponse.isa = function (o) {
        return __isa(o, "GetSessionResponse");
    };
})(GetSessionResponse || (GetSessionResponse = {}));
export var IntentSummary;
(function (IntentSummary) {
    IntentSummary.isa = function (o) { return __isa(o, "IntentSummary"); };
})(IntentSummary || (IntentSummary = {}));
export var LoopDetectedException;
(function (LoopDetectedException) {
    LoopDetectedException.isa = function (o) {
        return __isa(o, "LoopDetectedException");
    };
})(LoopDetectedException || (LoopDetectedException = {}));
export var MessageFormatType;
(function (MessageFormatType) {
    MessageFormatType["COMPOSITE"] = "Composite";
    MessageFormatType["CUSTOM_PAYLOAD"] = "CustomPayload";
    MessageFormatType["PLAIN_TEXT"] = "PlainText";
    MessageFormatType["SSML"] = "SSML";
})(MessageFormatType || (MessageFormatType = {}));
export var PostContentRequest;
(function (PostContentRequest) {
    PostContentRequest.isa = function (o) {
        return __isa(o, "PostContentRequest");
    };
})(PostContentRequest || (PostContentRequest = {}));
export var PostContentResponse;
(function (PostContentResponse) {
    PostContentResponse.isa = function (o) {
        return __isa(o, "PostContentResponse");
    };
})(PostContentResponse || (PostContentResponse = {}));
export var PostTextRequest;
(function (PostTextRequest) {
    PostTextRequest.isa = function (o) {
        return __isa(o, "PostTextRequest");
    };
})(PostTextRequest || (PostTextRequest = {}));
export var PostTextResponse;
(function (PostTextResponse) {
    PostTextResponse.isa = function (o) {
        return __isa(o, "PostTextResponse");
    };
})(PostTextResponse || (PostTextResponse = {}));
export var PutSessionRequest;
(function (PutSessionRequest) {
    PutSessionRequest.isa = function (o) {
        return __isa(o, "PutSessionRequest");
    };
})(PutSessionRequest || (PutSessionRequest = {}));
export var PutSessionResponse;
(function (PutSessionResponse) {
    PutSessionResponse.isa = function (o) {
        return __isa(o, "PutSessionResponse");
    };
})(PutSessionResponse || (PutSessionResponse = {}));
export var SentimentResponse;
(function (SentimentResponse) {
    SentimentResponse.isa = function (o) {
        return __isa(o, "SentimentResponse");
    };
})(SentimentResponse || (SentimentResponse = {}));
export var Button;
(function (Button) {
    Button.isa = function (o) { return __isa(o, "Button"); };
})(Button || (Button = {}));
export var ContentType;
(function (ContentType) {
    ContentType["GENERIC"] = "application/vnd.amazonaws.card.generic";
})(ContentType || (ContentType = {}));
export var GenericAttachment;
(function (GenericAttachment) {
    GenericAttachment.isa = function (o) {
        return __isa(o, "GenericAttachment");
    };
})(GenericAttachment || (GenericAttachment = {}));
export var ResponseCard;
(function (ResponseCard) {
    ResponseCard.isa = function (o) { return __isa(o, "ResponseCard"); };
})(ResponseCard || (ResponseCard = {}));
