// @aws-amplify/ui-components/dist/esm/amplify-authenticator.entry.js
import { r as registerInstance, h, H as Host } from './index-82dae9d1.js';
import { Logger, Hub } from '@aws-amplify/core';
import { A as AuthState } from './auth-types-78df304e.js';
import './Translations-baba3f2c.js';
import { T as TOAST_AUTH_ERROR_EVENT, U as UI_AUTH_CHANNEL, A as AUTH_CHANNEL, R as REDIRECTED_FROM_HOSTED_UI, N as NO_AUTH_MODULE_FOUND, b as AUTHENTICATOR_AUTHSTATE } from './constants-3e1b5fcc.js';
import { appendToCognitoUserAgent, Auth } from '@aws-amplify/auth';
import { d as dispatchAuthStateChangeEvent, o as onAuthUIStateChange } from './helpers-9dae6f2a.js';

const amplifyAuthenticatorCss = ":host{--background-color:var(--amplify-background-color)}";

const logger = new Logger('Authenticator');
class AmplifyAuthenticator {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        /** Initial starting state of the Authenticator component. E.g. If `signup` is passed the default component is set to AmplifySignUp */
        this.initialAuthState = AuthState.SignIn;
        /** Callback for Authenticator state machine changes */
        this.handleAuthStateChange = () => { };
        this.authState = AuthState.Loading;
        this.toastMessage = '';
        this.handleExternalAuthEvent = ({ payload }) => {
            switch (payload.event) {
                case 'cognitoHostedUI':
                    return dispatchAuthStateChangeEvent(AuthState.SignedIn, payload.data);
                case 'cognitoHostedUI_failure':
                case 'parsingUrl_failure':
                case 'signOut':
                case 'customGreetingSignOut':
                    return dispatchAuthStateChangeEvent(this.initialAuthState);
            }
        };
        this.handleToastEvent = ({ payload }) => {
            switch (payload.event) {
                case TOAST_AUTH_ERROR_EVENT:
                    if (payload.message)
                        this.toastMessage = payload.message;
                    break;
            }
        };
    }
    async componentWillLoad() {
        onAuthUIStateChange((authState, authData) => {
            this.onAuthStateChange(authState, authData);
            this.toastMessage = '';
        });
        Hub.listen(UI_AUTH_CHANNEL, this.handleToastEvent);
        Hub.listen(AUTH_CHANNEL, this.handleExternalAuthEvent);
        appendToCognitoUserAgent('amplify-authenticator');
        const byHostedUI = localStorage.getItem(REDIRECTED_FROM_HOSTED_UI);
        localStorage.removeItem(REDIRECTED_FROM_HOSTED_UI);
        if (byHostedUI !== 'true')
            await this.checkUser();
    }
    async checkUser() {
        if (!Auth || typeof Auth.currentAuthenticatedUser !== 'function') {
            throw new Error(NO_AUTH_MODULE_FOUND);
        }
        try {
            const user = await Auth.currentAuthenticatedUser();
            dispatchAuthStateChangeEvent(AuthState.SignedIn, user);
        }
        catch (error) {
            let cachedAuthState = null;
            try {
                cachedAuthState = localStorage.getItem(AUTHENTICATOR_AUTHSTATE);
            }
            catch (error) {
                logger.debug('Failed to get the auth state from local storage', error);
            }
            try {
                if (cachedAuthState === AuthState.SignedIn) {
                    await Auth.signOut();
                }
                dispatchAuthStateChangeEvent(this.initialAuthState);
            }
            catch (error) {
                logger.debug('Failed to sign out', error);
            }
        }
    }
    async onAuthStateChange(nextAuthState, data) {
        if (nextAuthState === undefined)
            return logger.error('nextAuthState cannot be undefined');
        logger.info('Inside onAuthStateChange Method current authState:', this.authState);
        if (nextAuthState === AuthState.SignedOut) {
            this.authState = this.initialAuthState;
        }
        else {
            this.authState = nextAuthState;
        }
        this.authData = data;
        if (this.authData)
            logger.log('Auth Data was set:', this.authData);
        if (this.authState === nextAuthState) {
            this.handleAuthStateChange(this.authState, this.authData);
            logger.info(`authState has been updated to ${this.authState}`);
        }
    }
    renderAuthComponent(authState) {
        switch (authState) {
            case AuthState.SignIn:
                return (h("slot", { name: "sign-in" }, h("amplify-sign-in", { federated: this.federated, usernameAlias: this.usernameAlias })));
            case AuthState.ConfirmSignIn:
                return (h("slot", { name: "confirm-sign-in" }, h("amplify-confirm-sign-in", { user: this.authData })));
            case AuthState.SignUp:
                return (h("slot", { name: "sign-up" }, h("amplify-sign-up", { usernameAlias: this.usernameAlias })));
            case AuthState.ConfirmSignUp:
                return (h("slot", { name: "confirm-sign-up" }, h("amplify-confirm-sign-up", { user: this.authData, usernameAlias: this.usernameAlias })));
            case AuthState.ForgotPassword:
                return (h("slot", { name: "forgot-password" }, h("amplify-forgot-password", { usernameAlias: this.usernameAlias })));
            case AuthState.ResetPassword:
                return (h("slot", { name: "require-new-password" }, h("amplify-require-new-password", { user: this.authData })));
            case AuthState.VerifyContact:
                return (h("slot", { name: "verify-contact" }, h("amplify-verify-contact", { user: this.authData })));
            case AuthState.TOTPSetup:
                return (h("slot", { name: "totp-setup" }, h("amplify-totp-setup", { user: this.authData })));
            case AuthState.Loading:
                return (h("slot", { name: "loading" }, h("div", null, "Loading...")));
            case AuthState.SignedIn:
                return [h("slot", { name: "greetings" }), h("slot", null)];
            default:
                throw new Error(`Unhandled auth state: ${authState}`);
        }
    }
    componentWillUnload() {
        Hub.remove(AUTH_CHANNEL, this.handleExternalAuthEvent);
        Hub.remove(UI_AUTH_CHANNEL, this.handleToastEvent);
        return onAuthUIStateChange;
    }
    render() {
        return (h(Host, null, this.toastMessage ? (h("amplify-toast", { message: this.toastMessage, handleClose: () => {
                this.toastMessage = '';
            }, "data-test": "authenticator-error" })) : null, this.renderAuthComponent(this.authState)));
    }
}
AmplifyAuthenticator.style = amplifyAuthenticatorCss;

export { AmplifyAuthenticator as amplify_authenticator };
