// @aws-sdk/client-pinpoint/dist/cjs/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const smithy_client_1 = require("@aws-sdk/smithy-client");
var ADMChannelRequest;
(function (ADMChannelRequest) {
    ADMChannelRequest.isa = (o) => smithy_client_1.isa(o, "ADMChannelRequest");
})(ADMChannelRequest = exports.ADMChannelRequest || (exports.ADMChannelRequest = {}));
var ADMChannelResponse;
(function (ADMChannelResponse) {
    ADMChannelResponse.isa = (o) => smithy_client_1.isa(o, "ADMChannelResponse");
})(ADMChannelResponse = exports.ADMChannelResponse || (exports.ADMChannelResponse = {}));
var ADMMessage;
(function (ADMMessage) {
    ADMMessage.isa = (o) => smithy_client_1.isa(o, "ADMMessage");
})(ADMMessage = exports.ADMMessage || (exports.ADMMessage = {}));
var APNSChannelRequest;
(function (APNSChannelRequest) {
    APNSChannelRequest.isa = (o) => smithy_client_1.isa(o, "APNSChannelRequest");
})(APNSChannelRequest = exports.APNSChannelRequest || (exports.APNSChannelRequest = {}));
var APNSChannelResponse;
(function (APNSChannelResponse) {
    APNSChannelResponse.isa = (o) => smithy_client_1.isa(o, "APNSChannelResponse");
})(APNSChannelResponse = exports.APNSChannelResponse || (exports.APNSChannelResponse = {}));
var APNSMessage;
(function (APNSMessage) {
    APNSMessage.isa = (o) => smithy_client_1.isa(o, "APNSMessage");
})(APNSMessage = exports.APNSMessage || (exports.APNSMessage = {}));
var APNSPushNotificationTemplate;
(function (APNSPushNotificationTemplate) {
    APNSPushNotificationTemplate.isa = (o) => smithy_client_1.isa(o, "APNSPushNotificationTemplate");
})(APNSPushNotificationTemplate = exports.APNSPushNotificationTemplate || (exports.APNSPushNotificationTemplate = {}));
var APNSSandboxChannelRequest;
(function (APNSSandboxChannelRequest) {
    APNSSandboxChannelRequest.isa = (o) => smithy_client_1.isa(o, "APNSSandboxChannelRequest");
})(APNSSandboxChannelRequest = exports.APNSSandboxChannelRequest || (exports.APNSSandboxChannelRequest = {}));
var APNSSandboxChannelResponse;
(function (APNSSandboxChannelResponse) {
    APNSSandboxChannelResponse.isa = (o) => smithy_client_1.isa(o, "APNSSandboxChannelResponse");
})(APNSSandboxChannelResponse = exports.APNSSandboxChannelResponse || (exports.APNSSandboxChannelResponse = {}));
var APNSVoipChannelRequest;
(function (APNSVoipChannelRequest) {
    APNSVoipChannelRequest.isa = (o) => smithy_client_1.isa(o, "APNSVoipChannelRequest");
})(APNSVoipChannelRequest = exports.APNSVoipChannelRequest || (exports.APNSVoipChannelRequest = {}));
var APNSVoipChannelResponse;
(function (APNSVoipChannelResponse) {
    APNSVoipChannelResponse.isa = (o) => smithy_client_1.isa(o, "APNSVoipChannelResponse");
})(APNSVoipChannelResponse = exports.APNSVoipChannelResponse || (exports.APNSVoipChannelResponse = {}));
var APNSVoipSandboxChannelRequest;
(function (APNSVoipSandboxChannelRequest) {
    APNSVoipSandboxChannelRequest.isa = (o) => smithy_client_1.isa(o, "APNSVoipSandboxChannelRequest");
})(APNSVoipSandboxChannelRequest = exports.APNSVoipSandboxChannelRequest || (exports.APNSVoipSandboxChannelRequest = {}));
var APNSVoipSandboxChannelResponse;
(function (APNSVoipSandboxChannelResponse) {
    APNSVoipSandboxChannelResponse.isa = (o) => smithy_client_1.isa(o, "APNSVoipSandboxChannelResponse");
})(APNSVoipSandboxChannelResponse = exports.APNSVoipSandboxChannelResponse || (exports.APNSVoipSandboxChannelResponse = {}));
var Action;
(function (Action) {
    Action["DEEP_LINK"] = "DEEP_LINK";
    Action["OPEN_APP"] = "OPEN_APP";
    Action["URL"] = "URL";
})(Action = exports.Action || (exports.Action = {}));
var ActivitiesResponse;
(function (ActivitiesResponse) {
    ActivitiesResponse.isa = (o) => smithy_client_1.isa(o, "ActivitiesResponse");
})(ActivitiesResponse = exports.ActivitiesResponse || (exports.ActivitiesResponse = {}));
var Activity;
(function (Activity) {
    Activity.isa = (o) => smithy_client_1.isa(o, "Activity");
})(Activity = exports.Activity || (exports.Activity = {}));
var ActivityResponse;
(function (ActivityResponse) {
    ActivityResponse.isa = (o) => smithy_client_1.isa(o, "ActivityResponse");
})(ActivityResponse = exports.ActivityResponse || (exports.ActivityResponse = {}));
var AddressConfiguration;
(function (AddressConfiguration) {
    AddressConfiguration.isa = (o) => smithy_client_1.isa(o, "AddressConfiguration");
})(AddressConfiguration = exports.AddressConfiguration || (exports.AddressConfiguration = {}));
var AndroidPushNotificationTemplate;
(function (AndroidPushNotificationTemplate) {
    AndroidPushNotificationTemplate.isa = (o) => smithy_client_1.isa(o, "AndroidPushNotificationTemplate");
})(AndroidPushNotificationTemplate = exports.AndroidPushNotificationTemplate || (exports.AndroidPushNotificationTemplate = {}));
var ApplicationDateRangeKpiResponse;
(function (ApplicationDateRangeKpiResponse) {
    ApplicationDateRangeKpiResponse.isa = (o) => smithy_client_1.isa(o, "ApplicationDateRangeKpiResponse");
})(ApplicationDateRangeKpiResponse = exports.ApplicationDateRangeKpiResponse || (exports.ApplicationDateRangeKpiResponse = {}));
var ApplicationResponse;
(function (ApplicationResponse) {
    ApplicationResponse.isa = (o) => smithy_client_1.isa(o, "ApplicationResponse");
})(ApplicationResponse = exports.ApplicationResponse || (exports.ApplicationResponse = {}));
var ApplicationSettingsResource;
(function (ApplicationSettingsResource) {
    ApplicationSettingsResource.isa = (o) => smithy_client_1.isa(o, "ApplicationSettingsResource");
})(ApplicationSettingsResource = exports.ApplicationSettingsResource || (exports.ApplicationSettingsResource = {}));
var ApplicationsResponse;
(function (ApplicationsResponse) {
    ApplicationsResponse.isa = (o) => smithy_client_1.isa(o, "ApplicationsResponse");
})(ApplicationsResponse = exports.ApplicationsResponse || (exports.ApplicationsResponse = {}));
var AttributeDimension;
(function (AttributeDimension) {
    AttributeDimension.isa = (o) => smithy_client_1.isa(o, "AttributeDimension");
})(AttributeDimension = exports.AttributeDimension || (exports.AttributeDimension = {}));
var AttributeType;
(function (AttributeType) {
    AttributeType["EXCLUSIVE"] = "EXCLUSIVE";
    AttributeType["INCLUSIVE"] = "INCLUSIVE";
})(AttributeType = exports.AttributeType || (exports.AttributeType = {}));
var AttributesResource;
(function (AttributesResource) {
    AttributesResource.isa = (o) => smithy_client_1.isa(o, "AttributesResource");
})(AttributesResource = exports.AttributesResource || (exports.AttributesResource = {}));
var BadRequestException;
(function (BadRequestException) {
    BadRequestException.isa = (o) => smithy_client_1.isa(o, "BadRequestException");
})(BadRequestException = exports.BadRequestException || (exports.BadRequestException = {}));
var BaiduChannelRequest;
(function (BaiduChannelRequest) {
    BaiduChannelRequest.isa = (o) => smithy_client_1.isa(o, "BaiduChannelRequest");
})(BaiduChannelRequest = exports.BaiduChannelRequest || (exports.BaiduChannelRequest = {}));
var BaiduChannelResponse;
(function (BaiduChannelResponse) {
    BaiduChannelResponse.isa = (o) => smithy_client_1.isa(o, "BaiduChannelResponse");
})(BaiduChannelResponse = exports.BaiduChannelResponse || (exports.BaiduChannelResponse = {}));
var BaiduMessage;
(function (BaiduMessage) {
    BaiduMessage.isa = (o) => smithy_client_1.isa(o, "BaiduMessage");
})(BaiduMessage = exports.BaiduMessage || (exports.BaiduMessage = {}));
var BaseKpiResult;
(function (BaseKpiResult) {
    BaseKpiResult.isa = (o) => smithy_client_1.isa(o, "BaseKpiResult");
})(BaseKpiResult = exports.BaseKpiResult || (exports.BaseKpiResult = {}));
var CampaignDateRangeKpiResponse;
(function (CampaignDateRangeKpiResponse) {
    CampaignDateRangeKpiResponse.isa = (o) => smithy_client_1.isa(o, "CampaignDateRangeKpiResponse");
})(CampaignDateRangeKpiResponse = exports.CampaignDateRangeKpiResponse || (exports.CampaignDateRangeKpiResponse = {}));
var CampaignEmailMessage;
(function (CampaignEmailMessage) {
    CampaignEmailMessage.isa = (o) => smithy_client_1.isa(o, "CampaignEmailMessage");
})(CampaignEmailMessage = exports.CampaignEmailMessage || (exports.CampaignEmailMessage = {}));
var CampaignEventFilter;
(function (CampaignEventFilter) {
    CampaignEventFilter.isa = (o) => smithy_client_1.isa(o, "CampaignEventFilter");
})(CampaignEventFilter = exports.CampaignEventFilter || (exports.CampaignEventFilter = {}));
var CampaignHook;
(function (CampaignHook) {
    CampaignHook.isa = (o) => smithy_client_1.isa(o, "CampaignHook");
})(CampaignHook = exports.CampaignHook || (exports.CampaignHook = {}));
var CampaignLimits;
(function (CampaignLimits) {
    CampaignLimits.isa = (o) => smithy_client_1.isa(o, "CampaignLimits");
})(CampaignLimits = exports.CampaignLimits || (exports.CampaignLimits = {}));
var CampaignResponse;
(function (CampaignResponse) {
    CampaignResponse.isa = (o) => smithy_client_1.isa(o, "CampaignResponse");
})(CampaignResponse = exports.CampaignResponse || (exports.CampaignResponse = {}));
var CampaignSmsMessage;
(function (CampaignSmsMessage) {
    CampaignSmsMessage.isa = (o) => smithy_client_1.isa(o, "CampaignSmsMessage");
})(CampaignSmsMessage = exports.CampaignSmsMessage || (exports.CampaignSmsMessage = {}));
var CampaignState;
(function (CampaignState) {
    CampaignState.isa = (o) => smithy_client_1.isa(o, "CampaignState");
})(CampaignState = exports.CampaignState || (exports.CampaignState = {}));
var CampaignStatus;
(function (CampaignStatus) {
    CampaignStatus["COMPLETED"] = "COMPLETED";
    CampaignStatus["DELETED"] = "DELETED";
    CampaignStatus["EXECUTING"] = "EXECUTING";
    CampaignStatus["PAUSED"] = "PAUSED";
    CampaignStatus["PENDING_NEXT_RUN"] = "PENDING_NEXT_RUN";
    CampaignStatus["SCHEDULED"] = "SCHEDULED";
})(CampaignStatus = exports.CampaignStatus || (exports.CampaignStatus = {}));
var CampaignsResponse;
(function (CampaignsResponse) {
    CampaignsResponse.isa = (o) => smithy_client_1.isa(o, "CampaignsResponse");
})(CampaignsResponse = exports.CampaignsResponse || (exports.CampaignsResponse = {}));
var ChannelResponse;
(function (ChannelResponse) {
    ChannelResponse.isa = (o) => smithy_client_1.isa(o, "ChannelResponse");
})(ChannelResponse = exports.ChannelResponse || (exports.ChannelResponse = {}));
var ChannelType;
(function (ChannelType) {
    ChannelType["ADM"] = "ADM";
    ChannelType["APNS"] = "APNS";
    ChannelType["APNS_SANDBOX"] = "APNS_SANDBOX";
    ChannelType["APNS_VOIP"] = "APNS_VOIP";
    ChannelType["APNS_VOIP_SANDBOX"] = "APNS_VOIP_SANDBOX";
    ChannelType["BAIDU"] = "BAIDU";
    ChannelType["CUSTOM"] = "CUSTOM";
    ChannelType["EMAIL"] = "EMAIL";
    ChannelType["GCM"] = "GCM";
    ChannelType["SMS"] = "SMS";
    ChannelType["VOICE"] = "VOICE";
})(ChannelType = exports.ChannelType || (exports.ChannelType = {}));
var ChannelsResponse;
(function (ChannelsResponse) {
    ChannelsResponse.isa = (o) => smithy_client_1.isa(o, "ChannelsResponse");
})(ChannelsResponse = exports.ChannelsResponse || (exports.ChannelsResponse = {}));
var Condition;
(function (Condition) {
    Condition.isa = (o) => smithy_client_1.isa(o, "Condition");
})(Condition = exports.Condition || (exports.Condition = {}));
var ConditionalSplitActivity;
(function (ConditionalSplitActivity) {
    ConditionalSplitActivity.isa = (o) => smithy_client_1.isa(o, "ConditionalSplitActivity");
})(ConditionalSplitActivity = exports.ConditionalSplitActivity || (exports.ConditionalSplitActivity = {}));
var CreateAppRequest;
(function (CreateAppRequest) {
    CreateAppRequest.isa = (o) => smithy_client_1.isa(o, "CreateAppRequest");
})(CreateAppRequest = exports.CreateAppRequest || (exports.CreateAppRequest = {}));
var CreateAppResponse;
(function (CreateAppResponse) {
    CreateAppResponse.isa = (o) => smithy_client_1.isa(o, "CreateAppResponse");
})(CreateAppResponse = exports.CreateAppResponse || (exports.CreateAppResponse = {}));
var CreateApplicationRequest;
(function (CreateApplicationRequest) {
    CreateApplicationRequest.isa = (o) => smithy_client_1.isa(o, "CreateApplicationRequest");
})(CreateApplicationRequest = exports.CreateApplicationRequest || (exports.CreateApplicationRequest = {}));
var CreateCampaignRequest;
(function (CreateCampaignRequest) {
    CreateCampaignRequest.isa = (o) => smithy_client_1.isa(o, "CreateCampaignRequest");
})(CreateCampaignRequest = exports.CreateCampaignRequest || (exports.CreateCampaignRequest = {}));
var CreateCampaignResponse;
(function (CreateCampaignResponse) {
    CreateCampaignResponse.isa = (o) => smithy_client_1.isa(o, "CreateCampaignResponse");
})(CreateCampaignResponse = exports.CreateCampaignResponse || (exports.CreateCampaignResponse = {}));
var CreateEmailTemplateRequest;
(function (CreateEmailTemplateRequest) {
    CreateEmailTemplateRequest.isa = (o) => smithy_client_1.isa(o, "CreateEmailTemplateRequest");
})(CreateEmailTemplateRequest = exports.CreateEmailTemplateRequest || (exports.CreateEmailTemplateRequest = {}));
var CreateEmailTemplateResponse;
(function (CreateEmailTemplateResponse) {
    CreateEmailTemplateResponse.isa = (o) => smithy_client_1.isa(o, "CreateEmailTemplateResponse");
})(CreateEmailTemplateResponse = exports.CreateEmailTemplateResponse || (exports.CreateEmailTemplateResponse = {}));
var CreateExportJobRequest;
(function (CreateExportJobRequest) {
    CreateExportJobRequest.isa = (o) => smithy_client_1.isa(o, "CreateExportJobRequest");
})(CreateExportJobRequest = exports.CreateExportJobRequest || (exports.CreateExportJobRequest = {}));
var CreateExportJobResponse;
(function (CreateExportJobResponse) {
    CreateExportJobResponse.isa = (o) => smithy_client_1.isa(o, "CreateExportJobResponse");
})(CreateExportJobResponse = exports.CreateExportJobResponse || (exports.CreateExportJobResponse = {}));
var CreateImportJobRequest;
(function (CreateImportJobRequest) {
    CreateImportJobRequest.isa = (o) => smithy_client_1.isa(o, "CreateImportJobRequest");
})(CreateImportJobRequest = exports.CreateImportJobRequest || (exports.CreateImportJobRequest = {}));
var CreateImportJobResponse;
(function (CreateImportJobResponse) {
    CreateImportJobResponse.isa = (o) => smithy_client_1.isa(o, "CreateImportJobResponse");
})(CreateImportJobResponse = exports.CreateImportJobResponse || (exports.CreateImportJobResponse = {}));
var CreateJourneyRequest;
(function (CreateJourneyRequest) {
    CreateJourneyRequest.isa = (o) => smithy_client_1.isa(o, "CreateJourneyRequest");
})(CreateJourneyRequest = exports.CreateJourneyRequest || (exports.CreateJourneyRequest = {}));
var CreateJourneyResponse;
(function (CreateJourneyResponse) {
    CreateJourneyResponse.isa = (o) => smithy_client_1.isa(o, "CreateJourneyResponse");
})(CreateJourneyResponse = exports.CreateJourneyResponse || (exports.CreateJourneyResponse = {}));
var CreatePushTemplateRequest;
(function (CreatePushTemplateRequest) {
    CreatePushTemplateRequest.isa = (o) => smithy_client_1.isa(o, "CreatePushTemplateRequest");
})(CreatePushTemplateRequest = exports.CreatePushTemplateRequest || (exports.CreatePushTemplateRequest = {}));
var CreatePushTemplateResponse;
(function (CreatePushTemplateResponse) {
    CreatePushTemplateResponse.isa = (o) => smithy_client_1.isa(o, "CreatePushTemplateResponse");
})(CreatePushTemplateResponse = exports.CreatePushTemplateResponse || (exports.CreatePushTemplateResponse = {}));
var CreateSegmentRequest;
(function (CreateSegmentRequest) {
    CreateSegmentRequest.isa = (o) => smithy_client_1.isa(o, "CreateSegmentRequest");
})(CreateSegmentRequest = exports.CreateSegmentRequest || (exports.CreateSegmentRequest = {}));
var CreateSegmentResponse;
(function (CreateSegmentResponse) {
    CreateSegmentResponse.isa = (o) => smithy_client_1.isa(o, "CreateSegmentResponse");
})(CreateSegmentResponse = exports.CreateSegmentResponse || (exports.CreateSegmentResponse = {}));
var CreateSmsTemplateRequest;
(function (CreateSmsTemplateRequest) {
    CreateSmsTemplateRequest.isa = (o) => smithy_client_1.isa(o, "CreateSmsTemplateRequest");
})(CreateSmsTemplateRequest = exports.CreateSmsTemplateRequest || (exports.CreateSmsTemplateRequest = {}));
var CreateSmsTemplateResponse;
(function (CreateSmsTemplateResponse) {
    CreateSmsTemplateResponse.isa = (o) => smithy_client_1.isa(o, "CreateSmsTemplateResponse");
})(CreateSmsTemplateResponse = exports.CreateSmsTemplateResponse || (exports.CreateSmsTemplateResponse = {}));
var CreateTemplateMessageBody;
(function (CreateTemplateMessageBody) {
    CreateTemplateMessageBody.isa = (o) => smithy_client_1.isa(o, "CreateTemplateMessageBody");
})(CreateTemplateMessageBody = exports.CreateTemplateMessageBody || (exports.CreateTemplateMessageBody = {}));
var CreateVoiceTemplateRequest;
(function (CreateVoiceTemplateRequest) {
    CreateVoiceTemplateRequest.isa = (o) => smithy_client_1.isa(o, "CreateVoiceTemplateRequest");
})(CreateVoiceTemplateRequest = exports.CreateVoiceTemplateRequest || (exports.CreateVoiceTemplateRequest = {}));
var CreateVoiceTemplateResponse;
(function (CreateVoiceTemplateResponse) {
    CreateVoiceTemplateResponse.isa = (o) => smithy_client_1.isa(o, "CreateVoiceTemplateResponse");
})(CreateVoiceTemplateResponse = exports.CreateVoiceTemplateResponse || (exports.CreateVoiceTemplateResponse = {}));
var DefaultMessage;
(function (DefaultMessage) {
    DefaultMessage.isa = (o) => smithy_client_1.isa(o, "DefaultMessage");
})(DefaultMessage = exports.DefaultMessage || (exports.DefaultMessage = {}));
var DefaultPushNotificationMessage;
(function (DefaultPushNotificationMessage) {
    DefaultPushNotificationMessage.isa = (o) => smithy_client_1.isa(o, "DefaultPushNotificationMessage");
})(DefaultPushNotificationMessage = exports.DefaultPushNotificationMessage || (exports.DefaultPushNotificationMessage = {}));
var DefaultPushNotificationTemplate;
(function (DefaultPushNotificationTemplate) {
    DefaultPushNotificationTemplate.isa = (o) => smithy_client_1.isa(o, "DefaultPushNotificationTemplate");
})(DefaultPushNotificationTemplate = exports.DefaultPushNotificationTemplate || (exports.DefaultPushNotificationTemplate = {}));
var DeleteAdmChannelRequest;
(function (DeleteAdmChannelRequest) {
    DeleteAdmChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteAdmChannelRequest");
})(DeleteAdmChannelRequest = exports.DeleteAdmChannelRequest || (exports.DeleteAdmChannelRequest = {}));
var DeleteAdmChannelResponse;
(function (DeleteAdmChannelResponse) {
    DeleteAdmChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteAdmChannelResponse");
})(DeleteAdmChannelResponse = exports.DeleteAdmChannelResponse || (exports.DeleteAdmChannelResponse = {}));
var DeleteApnsChannelRequest;
(function (DeleteApnsChannelRequest) {
    DeleteApnsChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteApnsChannelRequest");
})(DeleteApnsChannelRequest = exports.DeleteApnsChannelRequest || (exports.DeleteApnsChannelRequest = {}));
var DeleteApnsChannelResponse;
(function (DeleteApnsChannelResponse) {
    DeleteApnsChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteApnsChannelResponse");
})(DeleteApnsChannelResponse = exports.DeleteApnsChannelResponse || (exports.DeleteApnsChannelResponse = {}));
var DeleteApnsSandboxChannelRequest;
(function (DeleteApnsSandboxChannelRequest) {
    DeleteApnsSandboxChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteApnsSandboxChannelRequest");
})(DeleteApnsSandboxChannelRequest = exports.DeleteApnsSandboxChannelRequest || (exports.DeleteApnsSandboxChannelRequest = {}));
var DeleteApnsSandboxChannelResponse;
(function (DeleteApnsSandboxChannelResponse) {
    DeleteApnsSandboxChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteApnsSandboxChannelResponse");
})(DeleteApnsSandboxChannelResponse = exports.DeleteApnsSandboxChannelResponse || (exports.DeleteApnsSandboxChannelResponse = {}));
var DeleteApnsVoipChannelRequest;
(function (DeleteApnsVoipChannelRequest) {
    DeleteApnsVoipChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteApnsVoipChannelRequest");
})(DeleteApnsVoipChannelRequest = exports.DeleteApnsVoipChannelRequest || (exports.DeleteApnsVoipChannelRequest = {}));
var DeleteApnsVoipChannelResponse;
(function (DeleteApnsVoipChannelResponse) {
    DeleteApnsVoipChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteApnsVoipChannelResponse");
})(DeleteApnsVoipChannelResponse = exports.DeleteApnsVoipChannelResponse || (exports.DeleteApnsVoipChannelResponse = {}));
var DeleteApnsVoipSandboxChannelRequest;
(function (DeleteApnsVoipSandboxChannelRequest) {
    DeleteApnsVoipSandboxChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteApnsVoipSandboxChannelRequest");
})(DeleteApnsVoipSandboxChannelRequest = exports.DeleteApnsVoipSandboxChannelRequest || (exports.DeleteApnsVoipSandboxChannelRequest = {}));
var DeleteApnsVoipSandboxChannelResponse;
(function (DeleteApnsVoipSandboxChannelResponse) {
    DeleteApnsVoipSandboxChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteApnsVoipSandboxChannelResponse");
})(DeleteApnsVoipSandboxChannelResponse = exports.DeleteApnsVoipSandboxChannelResponse || (exports.DeleteApnsVoipSandboxChannelResponse = {}));
var DeleteAppRequest;
(function (DeleteAppRequest) {
    DeleteAppRequest.isa = (o) => smithy_client_1.isa(o, "DeleteAppRequest");
})(DeleteAppRequest = exports.DeleteAppRequest || (exports.DeleteAppRequest = {}));
var DeleteAppResponse;
(function (DeleteAppResponse) {
    DeleteAppResponse.isa = (o) => smithy_client_1.isa(o, "DeleteAppResponse");
})(DeleteAppResponse = exports.DeleteAppResponse || (exports.DeleteAppResponse = {}));
var DeleteBaiduChannelRequest;
(function (DeleteBaiduChannelRequest) {
    DeleteBaiduChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteBaiduChannelRequest");
})(DeleteBaiduChannelRequest = exports.DeleteBaiduChannelRequest || (exports.DeleteBaiduChannelRequest = {}));
var DeleteBaiduChannelResponse;
(function (DeleteBaiduChannelResponse) {
    DeleteBaiduChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteBaiduChannelResponse");
})(DeleteBaiduChannelResponse = exports.DeleteBaiduChannelResponse || (exports.DeleteBaiduChannelResponse = {}));
var DeleteCampaignRequest;
(function (DeleteCampaignRequest) {
    DeleteCampaignRequest.isa = (o) => smithy_client_1.isa(o, "DeleteCampaignRequest");
})(DeleteCampaignRequest = exports.DeleteCampaignRequest || (exports.DeleteCampaignRequest = {}));
var DeleteCampaignResponse;
(function (DeleteCampaignResponse) {
    DeleteCampaignResponse.isa = (o) => smithy_client_1.isa(o, "DeleteCampaignResponse");
})(DeleteCampaignResponse = exports.DeleteCampaignResponse || (exports.DeleteCampaignResponse = {}));
var DeleteEmailChannelRequest;
(function (DeleteEmailChannelRequest) {
    DeleteEmailChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteEmailChannelRequest");
})(DeleteEmailChannelRequest = exports.DeleteEmailChannelRequest || (exports.DeleteEmailChannelRequest = {}));
var DeleteEmailChannelResponse;
(function (DeleteEmailChannelResponse) {
    DeleteEmailChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteEmailChannelResponse");
})(DeleteEmailChannelResponse = exports.DeleteEmailChannelResponse || (exports.DeleteEmailChannelResponse = {}));
var DeleteEmailTemplateRequest;
(function (DeleteEmailTemplateRequest) {
    DeleteEmailTemplateRequest.isa = (o) => smithy_client_1.isa(o, "DeleteEmailTemplateRequest");
})(DeleteEmailTemplateRequest = exports.DeleteEmailTemplateRequest || (exports.DeleteEmailTemplateRequest = {}));
var DeleteEmailTemplateResponse;
(function (DeleteEmailTemplateResponse) {
    DeleteEmailTemplateResponse.isa = (o) => smithy_client_1.isa(o, "DeleteEmailTemplateResponse");
})(DeleteEmailTemplateResponse = exports.DeleteEmailTemplateResponse || (exports.DeleteEmailTemplateResponse = {}));
var DeleteEndpointRequest;
(function (DeleteEndpointRequest) {
    DeleteEndpointRequest.isa = (o) => smithy_client_1.isa(o, "DeleteEndpointRequest");
})(DeleteEndpointRequest = exports.DeleteEndpointRequest || (exports.DeleteEndpointRequest = {}));
var DeleteEndpointResponse;
(function (DeleteEndpointResponse) {
    DeleteEndpointResponse.isa = (o) => smithy_client_1.isa(o, "DeleteEndpointResponse");
})(DeleteEndpointResponse = exports.DeleteEndpointResponse || (exports.DeleteEndpointResponse = {}));
var DeleteEventStreamRequest;
(function (DeleteEventStreamRequest) {
    DeleteEventStreamRequest.isa = (o) => smithy_client_1.isa(o, "DeleteEventStreamRequest");
})(DeleteEventStreamRequest = exports.DeleteEventStreamRequest || (exports.DeleteEventStreamRequest = {}));
var DeleteEventStreamResponse;
(function (DeleteEventStreamResponse) {
    DeleteEventStreamResponse.isa = (o) => smithy_client_1.isa(o, "DeleteEventStreamResponse");
})(DeleteEventStreamResponse = exports.DeleteEventStreamResponse || (exports.DeleteEventStreamResponse = {}));
var DeleteGcmChannelRequest;
(function (DeleteGcmChannelRequest) {
    DeleteGcmChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteGcmChannelRequest");
})(DeleteGcmChannelRequest = exports.DeleteGcmChannelRequest || (exports.DeleteGcmChannelRequest = {}));
var DeleteGcmChannelResponse;
(function (DeleteGcmChannelResponse) {
    DeleteGcmChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteGcmChannelResponse");
})(DeleteGcmChannelResponse = exports.DeleteGcmChannelResponse || (exports.DeleteGcmChannelResponse = {}));
var DeleteJourneyRequest;
(function (DeleteJourneyRequest) {
    DeleteJourneyRequest.isa = (o) => smithy_client_1.isa(o, "DeleteJourneyRequest");
})(DeleteJourneyRequest = exports.DeleteJourneyRequest || (exports.DeleteJourneyRequest = {}));
var DeleteJourneyResponse;
(function (DeleteJourneyResponse) {
    DeleteJourneyResponse.isa = (o) => smithy_client_1.isa(o, "DeleteJourneyResponse");
})(DeleteJourneyResponse = exports.DeleteJourneyResponse || (exports.DeleteJourneyResponse = {}));
var DeletePushTemplateRequest;
(function (DeletePushTemplateRequest) {
    DeletePushTemplateRequest.isa = (o) => smithy_client_1.isa(o, "DeletePushTemplateRequest");
})(DeletePushTemplateRequest = exports.DeletePushTemplateRequest || (exports.DeletePushTemplateRequest = {}));
var DeletePushTemplateResponse;
(function (DeletePushTemplateResponse) {
    DeletePushTemplateResponse.isa = (o) => smithy_client_1.isa(o, "DeletePushTemplateResponse");
})(DeletePushTemplateResponse = exports.DeletePushTemplateResponse || (exports.DeletePushTemplateResponse = {}));
var DeleteSegmentRequest;
(function (DeleteSegmentRequest) {
    DeleteSegmentRequest.isa = (o) => smithy_client_1.isa(o, "DeleteSegmentRequest");
})(DeleteSegmentRequest = exports.DeleteSegmentRequest || (exports.DeleteSegmentRequest = {}));
var DeleteSegmentResponse;
(function (DeleteSegmentResponse) {
    DeleteSegmentResponse.isa = (o) => smithy_client_1.isa(o, "DeleteSegmentResponse");
})(DeleteSegmentResponse = exports.DeleteSegmentResponse || (exports.DeleteSegmentResponse = {}));
var DeleteSmsChannelRequest;
(function (DeleteSmsChannelRequest) {
    DeleteSmsChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteSmsChannelRequest");
})(DeleteSmsChannelRequest = exports.DeleteSmsChannelRequest || (exports.DeleteSmsChannelRequest = {}));
var DeleteSmsChannelResponse;
(function (DeleteSmsChannelResponse) {
    DeleteSmsChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteSmsChannelResponse");
})(DeleteSmsChannelResponse = exports.DeleteSmsChannelResponse || (exports.DeleteSmsChannelResponse = {}));
var DeleteSmsTemplateRequest;
(function (DeleteSmsTemplateRequest) {
    DeleteSmsTemplateRequest.isa = (o) => smithy_client_1.isa(o, "DeleteSmsTemplateRequest");
})(DeleteSmsTemplateRequest = exports.DeleteSmsTemplateRequest || (exports.DeleteSmsTemplateRequest = {}));
var DeleteSmsTemplateResponse;
(function (DeleteSmsTemplateResponse) {
    DeleteSmsTemplateResponse.isa = (o) => smithy_client_1.isa(o, "DeleteSmsTemplateResponse");
})(DeleteSmsTemplateResponse = exports.DeleteSmsTemplateResponse || (exports.DeleteSmsTemplateResponse = {}));
var DeleteUserEndpointsRequest;
(function (DeleteUserEndpointsRequest) {
    DeleteUserEndpointsRequest.isa = (o) => smithy_client_1.isa(o, "DeleteUserEndpointsRequest");
})(DeleteUserEndpointsRequest = exports.DeleteUserEndpointsRequest || (exports.DeleteUserEndpointsRequest = {}));
var DeleteUserEndpointsResponse;
(function (DeleteUserEndpointsResponse) {
    DeleteUserEndpointsResponse.isa = (o) => smithy_client_1.isa(o, "DeleteUserEndpointsResponse");
})(DeleteUserEndpointsResponse = exports.DeleteUserEndpointsResponse || (exports.DeleteUserEndpointsResponse = {}));
var DeleteVoiceChannelRequest;
(function (DeleteVoiceChannelRequest) {
    DeleteVoiceChannelRequest.isa = (o) => smithy_client_1.isa(o, "DeleteVoiceChannelRequest");
})(DeleteVoiceChannelRequest = exports.DeleteVoiceChannelRequest || (exports.DeleteVoiceChannelRequest = {}));
var DeleteVoiceChannelResponse;
(function (DeleteVoiceChannelResponse) {
    DeleteVoiceChannelResponse.isa = (o) => smithy_client_1.isa(o, "DeleteVoiceChannelResponse");
})(DeleteVoiceChannelResponse = exports.DeleteVoiceChannelResponse || (exports.DeleteVoiceChannelResponse = {}));
var DeleteVoiceTemplateRequest;
(function (DeleteVoiceTemplateRequest) {
    DeleteVoiceTemplateRequest.isa = (o) => smithy_client_1.isa(o, "DeleteVoiceTemplateRequest");
})(DeleteVoiceTemplateRequest = exports.DeleteVoiceTemplateRequest || (exports.DeleteVoiceTemplateRequest = {}));
var DeleteVoiceTemplateResponse;
(function (DeleteVoiceTemplateResponse) {
    DeleteVoiceTemplateResponse.isa = (o) => smithy_client_1.isa(o, "DeleteVoiceTemplateResponse");
})(DeleteVoiceTemplateResponse = exports.DeleteVoiceTemplateResponse || (exports.DeleteVoiceTemplateResponse = {}));
var DeliveryStatus;
(function (DeliveryStatus) {
    DeliveryStatus["DUPLICATE"] = "DUPLICATE";
    DeliveryStatus["OPT_OUT"] = "OPT_OUT";
    DeliveryStatus["PERMANENT_FAILURE"] = "PERMANENT_FAILURE";
    DeliveryStatus["SUCCESSFUL"] = "SUCCESSFUL";
    DeliveryStatus["TEMPORARY_FAILURE"] = "TEMPORARY_FAILURE";
    DeliveryStatus["THROTTLED"] = "THROTTLED";
    DeliveryStatus["UNKNOWN_FAILURE"] = "UNKNOWN_FAILURE";
})(DeliveryStatus = exports.DeliveryStatus || (exports.DeliveryStatus = {}));
var DimensionType;
(function (DimensionType) {
    DimensionType["EXCLUSIVE"] = "EXCLUSIVE";
    DimensionType["INCLUSIVE"] = "INCLUSIVE";
})(DimensionType = exports.DimensionType || (exports.DimensionType = {}));
var DirectMessageConfiguration;
(function (DirectMessageConfiguration) {
    DirectMessageConfiguration.isa = (o) => smithy_client_1.isa(o, "DirectMessageConfiguration");
})(DirectMessageConfiguration = exports.DirectMessageConfiguration || (exports.DirectMessageConfiguration = {}));
var Duration;
(function (Duration) {
    Duration["DAY_14"] = "DAY_14";
    Duration["DAY_30"] = "DAY_30";
    Duration["DAY_7"] = "DAY_7";
    Duration["HR_24"] = "HR_24";
})(Duration = exports.Duration || (exports.Duration = {}));
var EmailChannelRequest;
(function (EmailChannelRequest) {
    EmailChannelRequest.isa = (o) => smithy_client_1.isa(o, "EmailChannelRequest");
})(EmailChannelRequest = exports.EmailChannelRequest || (exports.EmailChannelRequest = {}));
var EmailChannelResponse;
(function (EmailChannelResponse) {
    EmailChannelResponse.isa = (o) => smithy_client_1.isa(o, "EmailChannelResponse");
})(EmailChannelResponse = exports.EmailChannelResponse || (exports.EmailChannelResponse = {}));
var EmailMessage;
(function (EmailMessage) {
    EmailMessage.isa = (o) => smithy_client_1.isa(o, "EmailMessage");
})(EmailMessage = exports.EmailMessage || (exports.EmailMessage = {}));
var EmailMessageActivity;
(function (EmailMessageActivity) {
    EmailMessageActivity.isa = (o) => smithy_client_1.isa(o, "EmailMessageActivity");
})(EmailMessageActivity = exports.EmailMessageActivity || (exports.EmailMessageActivity = {}));
var EmailTemplateRequest;
(function (EmailTemplateRequest) {
    EmailTemplateRequest.isa = (o) => smithy_client_1.isa(o, "EmailTemplateRequest");
})(EmailTemplateRequest = exports.EmailTemplateRequest || (exports.EmailTemplateRequest = {}));
var EmailTemplateResponse;
(function (EmailTemplateResponse) {
    EmailTemplateResponse.isa = (o) => smithy_client_1.isa(o, "EmailTemplateResponse");
})(EmailTemplateResponse = exports.EmailTemplateResponse || (exports.EmailTemplateResponse = {}));
var EndpointBatchItem;
(function (EndpointBatchItem) {
    EndpointBatchItem.isa = (o) => smithy_client_1.isa(o, "EndpointBatchItem");
})(EndpointBatchItem = exports.EndpointBatchItem || (exports.EndpointBatchItem = {}));
var EndpointBatchRequest;
(function (EndpointBatchRequest) {
    EndpointBatchRequest.isa = (o) => smithy_client_1.isa(o, "EndpointBatchRequest");
})(EndpointBatchRequest = exports.EndpointBatchRequest || (exports.EndpointBatchRequest = {}));
var EndpointDemographic;
(function (EndpointDemographic) {
    EndpointDemographic.isa = (o) => smithy_client_1.isa(o, "EndpointDemographic");
})(EndpointDemographic = exports.EndpointDemographic || (exports.EndpointDemographic = {}));
var EndpointItemResponse;
(function (EndpointItemResponse) {
    EndpointItemResponse.isa = (o) => smithy_client_1.isa(o, "EndpointItemResponse");
})(EndpointItemResponse = exports.EndpointItemResponse || (exports.EndpointItemResponse = {}));
var EndpointLocation;
(function (EndpointLocation) {
    EndpointLocation.isa = (o) => smithy_client_1.isa(o, "EndpointLocation");
})(EndpointLocation = exports.EndpointLocation || (exports.EndpointLocation = {}));
var EndpointMessageResult;
(function (EndpointMessageResult) {
    EndpointMessageResult.isa = (o) => smithy_client_1.isa(o, "EndpointMessageResult");
})(EndpointMessageResult = exports.EndpointMessageResult || (exports.EndpointMessageResult = {}));
var EndpointRequest;
(function (EndpointRequest) {
    EndpointRequest.isa = (o) => smithy_client_1.isa(o, "EndpointRequest");
})(EndpointRequest = exports.EndpointRequest || (exports.EndpointRequest = {}));
var EndpointResponse;
(function (EndpointResponse) {
    EndpointResponse.isa = (o) => smithy_client_1.isa(o, "EndpointResponse");
})(EndpointResponse = exports.EndpointResponse || (exports.EndpointResponse = {}));
var EndpointSendConfiguration;
(function (EndpointSendConfiguration) {
    EndpointSendConfiguration.isa = (o) => smithy_client_1.isa(o, "EndpointSendConfiguration");
})(EndpointSendConfiguration = exports.EndpointSendConfiguration || (exports.EndpointSendConfiguration = {}));
var EndpointUser;
(function (EndpointUser) {
    EndpointUser.isa = (o) => smithy_client_1.isa(o, "EndpointUser");
})(EndpointUser = exports.EndpointUser || (exports.EndpointUser = {}));
var EndpointsResponse;
(function (EndpointsResponse) {
    EndpointsResponse.isa = (o) => smithy_client_1.isa(o, "EndpointsResponse");
})(EndpointsResponse = exports.EndpointsResponse || (exports.EndpointsResponse = {}));
var Event;
(function (Event) {
    Event.isa = (o) => smithy_client_1.isa(o, "Event");
})(Event = exports.Event || (exports.Event = {}));
var EventCondition;
(function (EventCondition) {
    EventCondition.isa = (o) => smithy_client_1.isa(o, "EventCondition");
})(EventCondition = exports.EventCondition || (exports.EventCondition = {}));
var EventDimensions;
(function (EventDimensions) {
    EventDimensions.isa = (o) => smithy_client_1.isa(o, "EventDimensions");
})(EventDimensions = exports.EventDimensions || (exports.EventDimensions = {}));
var EventItemResponse;
(function (EventItemResponse) {
    EventItemResponse.isa = (o) => smithy_client_1.isa(o, "EventItemResponse");
})(EventItemResponse = exports.EventItemResponse || (exports.EventItemResponse = {}));
var EventStream;
(function (EventStream) {
    EventStream.isa = (o) => smithy_client_1.isa(o, "EventStream");
})(EventStream = exports.EventStream || (exports.EventStream = {}));
var EventsBatch;
(function (EventsBatch) {
    EventsBatch.isa = (o) => smithy_client_1.isa(o, "EventsBatch");
})(EventsBatch = exports.EventsBatch || (exports.EventsBatch = {}));
var EventsRequest;
(function (EventsRequest) {
    EventsRequest.isa = (o) => smithy_client_1.isa(o, "EventsRequest");
})(EventsRequest = exports.EventsRequest || (exports.EventsRequest = {}));
var EventsResponse;
(function (EventsResponse) {
    EventsResponse.isa = (o) => smithy_client_1.isa(o, "EventsResponse");
})(EventsResponse = exports.EventsResponse || (exports.EventsResponse = {}));
var ExportJobRequest;
(function (ExportJobRequest) {
    ExportJobRequest.isa = (o) => smithy_client_1.isa(o, "ExportJobRequest");
})(ExportJobRequest = exports.ExportJobRequest || (exports.ExportJobRequest = {}));
var ExportJobResource;
(function (ExportJobResource) {
    ExportJobResource.isa = (o) => smithy_client_1.isa(o, "ExportJobResource");
})(ExportJobResource = exports.ExportJobResource || (exports.ExportJobResource = {}));
var ExportJobResponse;
(function (ExportJobResponse) {
    ExportJobResponse.isa = (o) => smithy_client_1.isa(o, "ExportJobResponse");
})(ExportJobResponse = exports.ExportJobResponse || (exports.ExportJobResponse = {}));
var ExportJobsResponse;
(function (ExportJobsResponse) {
    ExportJobsResponse.isa = (o) => smithy_client_1.isa(o, "ExportJobsResponse");
})(ExportJobsResponse = exports.ExportJobsResponse || (exports.ExportJobsResponse = {}));
var FilterType;
(function (FilterType) {
    FilterType["ENDPOINT"] = "ENDPOINT";
    FilterType["SYSTEM"] = "SYSTEM";
})(FilterType = exports.FilterType || (exports.FilterType = {}));
var ForbiddenException;
(function (ForbiddenException) {
    ForbiddenException.isa = (o) => smithy_client_1.isa(o, "ForbiddenException");
})(ForbiddenException = exports.ForbiddenException || (exports.ForbiddenException = {}));
var Format;
(function (Format) {
    Format["CSV"] = "CSV";
    Format["JSON"] = "JSON";
})(Format = exports.Format || (exports.Format = {}));
var Frequency;
(function (Frequency) {
    Frequency["DAILY"] = "DAILY";
    Frequency["EVENT"] = "EVENT";
    Frequency["HOURLY"] = "HOURLY";
    Frequency["MONTHLY"] = "MONTHLY";
    Frequency["ONCE"] = "ONCE";
    Frequency["WEEKLY"] = "WEEKLY";
})(Frequency = exports.Frequency || (exports.Frequency = {}));
var GCMChannelRequest;
(function (GCMChannelRequest) {
    GCMChannelRequest.isa = (o) => smithy_client_1.isa(o, "GCMChannelRequest");
})(GCMChannelRequest = exports.GCMChannelRequest || (exports.GCMChannelRequest = {}));
var GCMChannelResponse;
(function (GCMChannelResponse) {
    GCMChannelResponse.isa = (o) => smithy_client_1.isa(o, "GCMChannelResponse");
})(GCMChannelResponse = exports.GCMChannelResponse || (exports.GCMChannelResponse = {}));
var GCMMessage;
(function (GCMMessage) {
    GCMMessage.isa = (o) => smithy_client_1.isa(o, "GCMMessage");
})(GCMMessage = exports.GCMMessage || (exports.GCMMessage = {}));
var GPSCoordinates;
(function (GPSCoordinates) {
    GPSCoordinates.isa = (o) => smithy_client_1.isa(o, "GPSCoordinates");
})(GPSCoordinates = exports.GPSCoordinates || (exports.GPSCoordinates = {}));
var GPSPointDimension;
(function (GPSPointDimension) {
    GPSPointDimension.isa = (o) => smithy_client_1.isa(o, "GPSPointDimension");
})(GPSPointDimension = exports.GPSPointDimension || (exports.GPSPointDimension = {}));
var GetAdmChannelRequest;
(function (GetAdmChannelRequest) {
    GetAdmChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetAdmChannelRequest");
})(GetAdmChannelRequest = exports.GetAdmChannelRequest || (exports.GetAdmChannelRequest = {}));
var GetAdmChannelResponse;
(function (GetAdmChannelResponse) {
    GetAdmChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetAdmChannelResponse");
})(GetAdmChannelResponse = exports.GetAdmChannelResponse || (exports.GetAdmChannelResponse = {}));
var GetApnsChannelRequest;
(function (GetApnsChannelRequest) {
    GetApnsChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetApnsChannelRequest");
})(GetApnsChannelRequest = exports.GetApnsChannelRequest || (exports.GetApnsChannelRequest = {}));
var GetApnsChannelResponse;
(function (GetApnsChannelResponse) {
    GetApnsChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetApnsChannelResponse");
})(GetApnsChannelResponse = exports.GetApnsChannelResponse || (exports.GetApnsChannelResponse = {}));
var GetApnsSandboxChannelRequest;
(function (GetApnsSandboxChannelRequest) {
    GetApnsSandboxChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetApnsSandboxChannelRequest");
})(GetApnsSandboxChannelRequest = exports.GetApnsSandboxChannelRequest || (exports.GetApnsSandboxChannelRequest = {}));
var GetApnsSandboxChannelResponse;
(function (GetApnsSandboxChannelResponse) {
    GetApnsSandboxChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetApnsSandboxChannelResponse");
})(GetApnsSandboxChannelResponse = exports.GetApnsSandboxChannelResponse || (exports.GetApnsSandboxChannelResponse = {}));
var GetApnsVoipChannelRequest;
(function (GetApnsVoipChannelRequest) {
    GetApnsVoipChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetApnsVoipChannelRequest");
})(GetApnsVoipChannelRequest = exports.GetApnsVoipChannelRequest || (exports.GetApnsVoipChannelRequest = {}));
var GetApnsVoipChannelResponse;
(function (GetApnsVoipChannelResponse) {
    GetApnsVoipChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetApnsVoipChannelResponse");
})(GetApnsVoipChannelResponse = exports.GetApnsVoipChannelResponse || (exports.GetApnsVoipChannelResponse = {}));
var GetApnsVoipSandboxChannelRequest;
(function (GetApnsVoipSandboxChannelRequest) {
    GetApnsVoipSandboxChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetApnsVoipSandboxChannelRequest");
})(GetApnsVoipSandboxChannelRequest = exports.GetApnsVoipSandboxChannelRequest || (exports.GetApnsVoipSandboxChannelRequest = {}));
var GetApnsVoipSandboxChannelResponse;
(function (GetApnsVoipSandboxChannelResponse) {
    GetApnsVoipSandboxChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetApnsVoipSandboxChannelResponse");
})(GetApnsVoipSandboxChannelResponse = exports.GetApnsVoipSandboxChannelResponse || (exports.GetApnsVoipSandboxChannelResponse = {}));
var GetAppRequest;
(function (GetAppRequest) {
    GetAppRequest.isa = (o) => smithy_client_1.isa(o, "GetAppRequest");
})(GetAppRequest = exports.GetAppRequest || (exports.GetAppRequest = {}));
var GetAppResponse;
(function (GetAppResponse) {
    GetAppResponse.isa = (o) => smithy_client_1.isa(o, "GetAppResponse");
})(GetAppResponse = exports.GetAppResponse || (exports.GetAppResponse = {}));
var GetApplicationDateRangeKpiRequest;
(function (GetApplicationDateRangeKpiRequest) {
    GetApplicationDateRangeKpiRequest.isa = (o) => smithy_client_1.isa(o, "GetApplicationDateRangeKpiRequest");
})(GetApplicationDateRangeKpiRequest = exports.GetApplicationDateRangeKpiRequest || (exports.GetApplicationDateRangeKpiRequest = {}));
var GetApplicationDateRangeKpiResponse;
(function (GetApplicationDateRangeKpiResponse) {
    GetApplicationDateRangeKpiResponse.isa = (o) => smithy_client_1.isa(o, "GetApplicationDateRangeKpiResponse");
})(GetApplicationDateRangeKpiResponse = exports.GetApplicationDateRangeKpiResponse || (exports.GetApplicationDateRangeKpiResponse = {}));
var GetApplicationSettingsRequest;
(function (GetApplicationSettingsRequest) {
    GetApplicationSettingsRequest.isa = (o) => smithy_client_1.isa(o, "GetApplicationSettingsRequest");
})(GetApplicationSettingsRequest = exports.GetApplicationSettingsRequest || (exports.GetApplicationSettingsRequest = {}));
var GetApplicationSettingsResponse;
(function (GetApplicationSettingsResponse) {
    GetApplicationSettingsResponse.isa = (o) => smithy_client_1.isa(o, "GetApplicationSettingsResponse");
})(GetApplicationSettingsResponse = exports.GetApplicationSettingsResponse || (exports.GetApplicationSettingsResponse = {}));
var GetAppsRequest;
(function (GetAppsRequest) {
    GetAppsRequest.isa = (o) => smithy_client_1.isa(o, "GetAppsRequest");
})(GetAppsRequest = exports.GetAppsRequest || (exports.GetAppsRequest = {}));
var GetAppsResponse;
(function (GetAppsResponse) {
    GetAppsResponse.isa = (o) => smithy_client_1.isa(o, "GetAppsResponse");
})(GetAppsResponse = exports.GetAppsResponse || (exports.GetAppsResponse = {}));
var GetBaiduChannelRequest;
(function (GetBaiduChannelRequest) {
    GetBaiduChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetBaiduChannelRequest");
})(GetBaiduChannelRequest = exports.GetBaiduChannelRequest || (exports.GetBaiduChannelRequest = {}));
var GetBaiduChannelResponse;
(function (GetBaiduChannelResponse) {
    GetBaiduChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetBaiduChannelResponse");
})(GetBaiduChannelResponse = exports.GetBaiduChannelResponse || (exports.GetBaiduChannelResponse = {}));
var GetCampaignActivitiesRequest;
(function (GetCampaignActivitiesRequest) {
    GetCampaignActivitiesRequest.isa = (o) => smithy_client_1.isa(o, "GetCampaignActivitiesRequest");
})(GetCampaignActivitiesRequest = exports.GetCampaignActivitiesRequest || (exports.GetCampaignActivitiesRequest = {}));
var GetCampaignActivitiesResponse;
(function (GetCampaignActivitiesResponse) {
    GetCampaignActivitiesResponse.isa = (o) => smithy_client_1.isa(o, "GetCampaignActivitiesResponse");
})(GetCampaignActivitiesResponse = exports.GetCampaignActivitiesResponse || (exports.GetCampaignActivitiesResponse = {}));
var GetCampaignDateRangeKpiRequest;
(function (GetCampaignDateRangeKpiRequest) {
    GetCampaignDateRangeKpiRequest.isa = (o) => smithy_client_1.isa(o, "GetCampaignDateRangeKpiRequest");
})(GetCampaignDateRangeKpiRequest = exports.GetCampaignDateRangeKpiRequest || (exports.GetCampaignDateRangeKpiRequest = {}));
var GetCampaignDateRangeKpiResponse;
(function (GetCampaignDateRangeKpiResponse) {
    GetCampaignDateRangeKpiResponse.isa = (o) => smithy_client_1.isa(o, "GetCampaignDateRangeKpiResponse");
})(GetCampaignDateRangeKpiResponse = exports.GetCampaignDateRangeKpiResponse || (exports.GetCampaignDateRangeKpiResponse = {}));
var GetCampaignRequest;
(function (GetCampaignRequest) {
    GetCampaignRequest.isa = (o) => smithy_client_1.isa(o, "GetCampaignRequest");
})(GetCampaignRequest = exports.GetCampaignRequest || (exports.GetCampaignRequest = {}));
var GetCampaignResponse;
(function (GetCampaignResponse) {
    GetCampaignResponse.isa = (o) => smithy_client_1.isa(o, "GetCampaignResponse");
})(GetCampaignResponse = exports.GetCampaignResponse || (exports.GetCampaignResponse = {}));
var GetCampaignVersionRequest;
(function (GetCampaignVersionRequest) {
    GetCampaignVersionRequest.isa = (o) => smithy_client_1.isa(o, "GetCampaignVersionRequest");
})(GetCampaignVersionRequest = exports.GetCampaignVersionRequest || (exports.GetCampaignVersionRequest = {}));
var GetCampaignVersionResponse;
(function (GetCampaignVersionResponse) {
    GetCampaignVersionResponse.isa = (o) => smithy_client_1.isa(o, "GetCampaignVersionResponse");
})(GetCampaignVersionResponse = exports.GetCampaignVersionResponse || (exports.GetCampaignVersionResponse = {}));
var GetCampaignVersionsRequest;
(function (GetCampaignVersionsRequest) {
    GetCampaignVersionsRequest.isa = (o) => smithy_client_1.isa(o, "GetCampaignVersionsRequest");
})(GetCampaignVersionsRequest = exports.GetCampaignVersionsRequest || (exports.GetCampaignVersionsRequest = {}));
var GetCampaignVersionsResponse;
(function (GetCampaignVersionsResponse) {
    GetCampaignVersionsResponse.isa = (o) => smithy_client_1.isa(o, "GetCampaignVersionsResponse");
})(GetCampaignVersionsResponse = exports.GetCampaignVersionsResponse || (exports.GetCampaignVersionsResponse = {}));
var GetCampaignsRequest;
(function (GetCampaignsRequest) {
    GetCampaignsRequest.isa = (o) => smithy_client_1.isa(o, "GetCampaignsRequest");
})(GetCampaignsRequest = exports.GetCampaignsRequest || (exports.GetCampaignsRequest = {}));
var GetCampaignsResponse;
(function (GetCampaignsResponse) {
    GetCampaignsResponse.isa = (o) => smithy_client_1.isa(o, "GetCampaignsResponse");
})(GetCampaignsResponse = exports.GetCampaignsResponse || (exports.GetCampaignsResponse = {}));
var GetChannelsRequest;
(function (GetChannelsRequest) {
    GetChannelsRequest.isa = (o) => smithy_client_1.isa(o, "GetChannelsRequest");
})(GetChannelsRequest = exports.GetChannelsRequest || (exports.GetChannelsRequest = {}));
var GetChannelsResponse;
(function (GetChannelsResponse) {
    GetChannelsResponse.isa = (o) => smithy_client_1.isa(o, "GetChannelsResponse");
})(GetChannelsResponse = exports.GetChannelsResponse || (exports.GetChannelsResponse = {}));
var GetEmailChannelRequest;
(function (GetEmailChannelRequest) {
    GetEmailChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetEmailChannelRequest");
})(GetEmailChannelRequest = exports.GetEmailChannelRequest || (exports.GetEmailChannelRequest = {}));
var GetEmailChannelResponse;
(function (GetEmailChannelResponse) {
    GetEmailChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetEmailChannelResponse");
})(GetEmailChannelResponse = exports.GetEmailChannelResponse || (exports.GetEmailChannelResponse = {}));
var GetEmailTemplateRequest;
(function (GetEmailTemplateRequest) {
    GetEmailTemplateRequest.isa = (o) => smithy_client_1.isa(o, "GetEmailTemplateRequest");
})(GetEmailTemplateRequest = exports.GetEmailTemplateRequest || (exports.GetEmailTemplateRequest = {}));
var GetEmailTemplateResponse;
(function (GetEmailTemplateResponse) {
    GetEmailTemplateResponse.isa = (o) => smithy_client_1.isa(o, "GetEmailTemplateResponse");
})(GetEmailTemplateResponse = exports.GetEmailTemplateResponse || (exports.GetEmailTemplateResponse = {}));
var GetEndpointRequest;
(function (GetEndpointRequest) {
    GetEndpointRequest.isa = (o) => smithy_client_1.isa(o, "GetEndpointRequest");
})(GetEndpointRequest = exports.GetEndpointRequest || (exports.GetEndpointRequest = {}));
var GetEndpointResponse;
(function (GetEndpointResponse) {
    GetEndpointResponse.isa = (o) => smithy_client_1.isa(o, "GetEndpointResponse");
})(GetEndpointResponse = exports.GetEndpointResponse || (exports.GetEndpointResponse = {}));
var GetEventStreamRequest;
(function (GetEventStreamRequest) {
    GetEventStreamRequest.isa = (o) => smithy_client_1.isa(o, "GetEventStreamRequest");
})(GetEventStreamRequest = exports.GetEventStreamRequest || (exports.GetEventStreamRequest = {}));
var GetEventStreamResponse;
(function (GetEventStreamResponse) {
    GetEventStreamResponse.isa = (o) => smithy_client_1.isa(o, "GetEventStreamResponse");
})(GetEventStreamResponse = exports.GetEventStreamResponse || (exports.GetEventStreamResponse = {}));
var GetExportJobRequest;
(function (GetExportJobRequest) {
    GetExportJobRequest.isa = (o) => smithy_client_1.isa(o, "GetExportJobRequest");
})(GetExportJobRequest = exports.GetExportJobRequest || (exports.GetExportJobRequest = {}));
var GetExportJobResponse;
(function (GetExportJobResponse) {
    GetExportJobResponse.isa = (o) => smithy_client_1.isa(o, "GetExportJobResponse");
})(GetExportJobResponse = exports.GetExportJobResponse || (exports.GetExportJobResponse = {}));
var GetExportJobsRequest;
(function (GetExportJobsRequest) {
    GetExportJobsRequest.isa = (o) => smithy_client_1.isa(o, "GetExportJobsRequest");
})(GetExportJobsRequest = exports.GetExportJobsRequest || (exports.GetExportJobsRequest = {}));
var GetExportJobsResponse;
(function (GetExportJobsResponse) {
    GetExportJobsResponse.isa = (o) => smithy_client_1.isa(o, "GetExportJobsResponse");
})(GetExportJobsResponse = exports.GetExportJobsResponse || (exports.GetExportJobsResponse = {}));
var GetGcmChannelRequest;
(function (GetGcmChannelRequest) {
    GetGcmChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetGcmChannelRequest");
})(GetGcmChannelRequest = exports.GetGcmChannelRequest || (exports.GetGcmChannelRequest = {}));
var GetGcmChannelResponse;
(function (GetGcmChannelResponse) {
    GetGcmChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetGcmChannelResponse");
})(GetGcmChannelResponse = exports.GetGcmChannelResponse || (exports.GetGcmChannelResponse = {}));
var GetImportJobRequest;
(function (GetImportJobRequest) {
    GetImportJobRequest.isa = (o) => smithy_client_1.isa(o, "GetImportJobRequest");
})(GetImportJobRequest = exports.GetImportJobRequest || (exports.GetImportJobRequest = {}));
var GetImportJobResponse;
(function (GetImportJobResponse) {
    GetImportJobResponse.isa = (o) => smithy_client_1.isa(o, "GetImportJobResponse");
})(GetImportJobResponse = exports.GetImportJobResponse || (exports.GetImportJobResponse = {}));
var GetImportJobsRequest;
(function (GetImportJobsRequest) {
    GetImportJobsRequest.isa = (o) => smithy_client_1.isa(o, "GetImportJobsRequest");
})(GetImportJobsRequest = exports.GetImportJobsRequest || (exports.GetImportJobsRequest = {}));
var GetImportJobsResponse;
(function (GetImportJobsResponse) {
    GetImportJobsResponse.isa = (o) => smithy_client_1.isa(o, "GetImportJobsResponse");
})(GetImportJobsResponse = exports.GetImportJobsResponse || (exports.GetImportJobsResponse = {}));
var GetJourneyDateRangeKpiRequest;
(function (GetJourneyDateRangeKpiRequest) {
    GetJourneyDateRangeKpiRequest.isa = (o) => smithy_client_1.isa(o, "GetJourneyDateRangeKpiRequest");
})(GetJourneyDateRangeKpiRequest = exports.GetJourneyDateRangeKpiRequest || (exports.GetJourneyDateRangeKpiRequest = {}));
var GetJourneyDateRangeKpiResponse;
(function (GetJourneyDateRangeKpiResponse) {
    GetJourneyDateRangeKpiResponse.isa = (o) => smithy_client_1.isa(o, "GetJourneyDateRangeKpiResponse");
})(GetJourneyDateRangeKpiResponse = exports.GetJourneyDateRangeKpiResponse || (exports.GetJourneyDateRangeKpiResponse = {}));
var GetJourneyExecutionActivityMetricsRequest;
(function (GetJourneyExecutionActivityMetricsRequest) {
    GetJourneyExecutionActivityMetricsRequest.isa = (o) => smithy_client_1.isa(o, "GetJourneyExecutionActivityMetricsRequest");
})(GetJourneyExecutionActivityMetricsRequest = exports.GetJourneyExecutionActivityMetricsRequest || (exports.GetJourneyExecutionActivityMetricsRequest = {}));
var GetJourneyExecutionActivityMetricsResponse;
(function (GetJourneyExecutionActivityMetricsResponse) {
    GetJourneyExecutionActivityMetricsResponse.isa = (o) => smithy_client_1.isa(o, "GetJourneyExecutionActivityMetricsResponse");
})(GetJourneyExecutionActivityMetricsResponse = exports.GetJourneyExecutionActivityMetricsResponse || (exports.GetJourneyExecutionActivityMetricsResponse = {}));
var GetJourneyExecutionMetricsRequest;
(function (GetJourneyExecutionMetricsRequest) {
    GetJourneyExecutionMetricsRequest.isa = (o) => smithy_client_1.isa(o, "GetJourneyExecutionMetricsRequest");
})(GetJourneyExecutionMetricsRequest = exports.GetJourneyExecutionMetricsRequest || (exports.GetJourneyExecutionMetricsRequest = {}));
var GetJourneyExecutionMetricsResponse;
(function (GetJourneyExecutionMetricsResponse) {
    GetJourneyExecutionMetricsResponse.isa = (o) => smithy_client_1.isa(o, "GetJourneyExecutionMetricsResponse");
})(GetJourneyExecutionMetricsResponse = exports.GetJourneyExecutionMetricsResponse || (exports.GetJourneyExecutionMetricsResponse = {}));
var GetJourneyRequest;
(function (GetJourneyRequest) {
    GetJourneyRequest.isa = (o) => smithy_client_1.isa(o, "GetJourneyRequest");
})(GetJourneyRequest = exports.GetJourneyRequest || (exports.GetJourneyRequest = {}));
var GetJourneyResponse;
(function (GetJourneyResponse) {
    GetJourneyResponse.isa = (o) => smithy_client_1.isa(o, "GetJourneyResponse");
})(GetJourneyResponse = exports.GetJourneyResponse || (exports.GetJourneyResponse = {}));
var GetPushTemplateRequest;
(function (GetPushTemplateRequest) {
    GetPushTemplateRequest.isa = (o) => smithy_client_1.isa(o, "GetPushTemplateRequest");
})(GetPushTemplateRequest = exports.GetPushTemplateRequest || (exports.GetPushTemplateRequest = {}));
var GetPushTemplateResponse;
(function (GetPushTemplateResponse) {
    GetPushTemplateResponse.isa = (o) => smithy_client_1.isa(o, "GetPushTemplateResponse");
})(GetPushTemplateResponse = exports.GetPushTemplateResponse || (exports.GetPushTemplateResponse = {}));
var GetSegmentExportJobsRequest;
(function (GetSegmentExportJobsRequest) {
    GetSegmentExportJobsRequest.isa = (o) => smithy_client_1.isa(o, "GetSegmentExportJobsRequest");
})(GetSegmentExportJobsRequest = exports.GetSegmentExportJobsRequest || (exports.GetSegmentExportJobsRequest = {}));
var GetSegmentExportJobsResponse;
(function (GetSegmentExportJobsResponse) {
    GetSegmentExportJobsResponse.isa = (o) => smithy_client_1.isa(o, "GetSegmentExportJobsResponse");
})(GetSegmentExportJobsResponse = exports.GetSegmentExportJobsResponse || (exports.GetSegmentExportJobsResponse = {}));
var GetSegmentImportJobsRequest;
(function (GetSegmentImportJobsRequest) {
    GetSegmentImportJobsRequest.isa = (o) => smithy_client_1.isa(o, "GetSegmentImportJobsRequest");
})(GetSegmentImportJobsRequest = exports.GetSegmentImportJobsRequest || (exports.GetSegmentImportJobsRequest = {}));
var GetSegmentImportJobsResponse;
(function (GetSegmentImportJobsResponse) {
    GetSegmentImportJobsResponse.isa = (o) => smithy_client_1.isa(o, "GetSegmentImportJobsResponse");
})(GetSegmentImportJobsResponse = exports.GetSegmentImportJobsResponse || (exports.GetSegmentImportJobsResponse = {}));
var GetSegmentRequest;
(function (GetSegmentRequest) {
    GetSegmentRequest.isa = (o) => smithy_client_1.isa(o, "GetSegmentRequest");
})(GetSegmentRequest = exports.GetSegmentRequest || (exports.GetSegmentRequest = {}));
var GetSegmentResponse;
(function (GetSegmentResponse) {
    GetSegmentResponse.isa = (o) => smithy_client_1.isa(o, "GetSegmentResponse");
})(GetSegmentResponse = exports.GetSegmentResponse || (exports.GetSegmentResponse = {}));
var GetSegmentVersionRequest;
(function (GetSegmentVersionRequest) {
    GetSegmentVersionRequest.isa = (o) => smithy_client_1.isa(o, "GetSegmentVersionRequest");
})(GetSegmentVersionRequest = exports.GetSegmentVersionRequest || (exports.GetSegmentVersionRequest = {}));
var GetSegmentVersionResponse;
(function (GetSegmentVersionResponse) {
    GetSegmentVersionResponse.isa = (o) => smithy_client_1.isa(o, "GetSegmentVersionResponse");
})(GetSegmentVersionResponse = exports.GetSegmentVersionResponse || (exports.GetSegmentVersionResponse = {}));
var GetSegmentVersionsRequest;
(function (GetSegmentVersionsRequest) {
    GetSegmentVersionsRequest.isa = (o) => smithy_client_1.isa(o, "GetSegmentVersionsRequest");
})(GetSegmentVersionsRequest = exports.GetSegmentVersionsRequest || (exports.GetSegmentVersionsRequest = {}));
var GetSegmentVersionsResponse;
(function (GetSegmentVersionsResponse) {
    GetSegmentVersionsResponse.isa = (o) => smithy_client_1.isa(o, "GetSegmentVersionsResponse");
})(GetSegmentVersionsResponse = exports.GetSegmentVersionsResponse || (exports.GetSegmentVersionsResponse = {}));
var GetSegmentsRequest;
(function (GetSegmentsRequest) {
    GetSegmentsRequest.isa = (o) => smithy_client_1.isa(o, "GetSegmentsRequest");
})(GetSegmentsRequest = exports.GetSegmentsRequest || (exports.GetSegmentsRequest = {}));
var GetSegmentsResponse;
(function (GetSegmentsResponse) {
    GetSegmentsResponse.isa = (o) => smithy_client_1.isa(o, "GetSegmentsResponse");
})(GetSegmentsResponse = exports.GetSegmentsResponse || (exports.GetSegmentsResponse = {}));
var GetSmsChannelRequest;
(function (GetSmsChannelRequest) {
    GetSmsChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetSmsChannelRequest");
})(GetSmsChannelRequest = exports.GetSmsChannelRequest || (exports.GetSmsChannelRequest = {}));
var GetSmsChannelResponse;
(function (GetSmsChannelResponse) {
    GetSmsChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetSmsChannelResponse");
})(GetSmsChannelResponse = exports.GetSmsChannelResponse || (exports.GetSmsChannelResponse = {}));
var GetSmsTemplateRequest;
(function (GetSmsTemplateRequest) {
    GetSmsTemplateRequest.isa = (o) => smithy_client_1.isa(o, "GetSmsTemplateRequest");
})(GetSmsTemplateRequest = exports.GetSmsTemplateRequest || (exports.GetSmsTemplateRequest = {}));
var GetSmsTemplateResponse;
(function (GetSmsTemplateResponse) {
    GetSmsTemplateResponse.isa = (o) => smithy_client_1.isa(o, "GetSmsTemplateResponse");
})(GetSmsTemplateResponse = exports.GetSmsTemplateResponse || (exports.GetSmsTemplateResponse = {}));
var GetUserEndpointsRequest;
(function (GetUserEndpointsRequest) {
    GetUserEndpointsRequest.isa = (o) => smithy_client_1.isa(o, "GetUserEndpointsRequest");
})(GetUserEndpointsRequest = exports.GetUserEndpointsRequest || (exports.GetUserEndpointsRequest = {}));
var GetUserEndpointsResponse;
(function (GetUserEndpointsResponse) {
    GetUserEndpointsResponse.isa = (o) => smithy_client_1.isa(o, "GetUserEndpointsResponse");
})(GetUserEndpointsResponse = exports.GetUserEndpointsResponse || (exports.GetUserEndpointsResponse = {}));
var GetVoiceChannelRequest;
(function (GetVoiceChannelRequest) {
    GetVoiceChannelRequest.isa = (o) => smithy_client_1.isa(o, "GetVoiceChannelRequest");
})(GetVoiceChannelRequest = exports.GetVoiceChannelRequest || (exports.GetVoiceChannelRequest = {}));
var GetVoiceChannelResponse;
(function (GetVoiceChannelResponse) {
    GetVoiceChannelResponse.isa = (o) => smithy_client_1.isa(o, "GetVoiceChannelResponse");
})(GetVoiceChannelResponse = exports.GetVoiceChannelResponse || (exports.GetVoiceChannelResponse = {}));
var GetVoiceTemplateRequest;
(function (GetVoiceTemplateRequest) {
    GetVoiceTemplateRequest.isa = (o) => smithy_client_1.isa(o, "GetVoiceTemplateRequest");
})(GetVoiceTemplateRequest = exports.GetVoiceTemplateRequest || (exports.GetVoiceTemplateRequest = {}));
var GetVoiceTemplateResponse;
(function (GetVoiceTemplateResponse) {
    GetVoiceTemplateResponse.isa = (o) => smithy_client_1.isa(o, "GetVoiceTemplateResponse");
})(GetVoiceTemplateResponse = exports.GetVoiceTemplateResponse || (exports.GetVoiceTemplateResponse = {}));
var HoldoutActivity;
(function (HoldoutActivity) {
    HoldoutActivity.isa = (o) => smithy_client_1.isa(o, "HoldoutActivity");
})(HoldoutActivity = exports.HoldoutActivity || (exports.HoldoutActivity = {}));
var ImportJobRequest;
(function (ImportJobRequest) {
    ImportJobRequest.isa = (o) => smithy_client_1.isa(o, "ImportJobRequest");
})(ImportJobRequest = exports.ImportJobRequest || (exports.ImportJobRequest = {}));
var ImportJobResource;
(function (ImportJobResource) {
    ImportJobResource.isa = (o) => smithy_client_1.isa(o, "ImportJobResource");
})(ImportJobResource = exports.ImportJobResource || (exports.ImportJobResource = {}));
var ImportJobResponse;
(function (ImportJobResponse) {
    ImportJobResponse.isa = (o) => smithy_client_1.isa(o, "ImportJobResponse");
})(ImportJobResponse = exports.ImportJobResponse || (exports.ImportJobResponse = {}));
var ImportJobsResponse;
(function (ImportJobsResponse) {
    ImportJobsResponse.isa = (o) => smithy_client_1.isa(o, "ImportJobsResponse");
})(ImportJobsResponse = exports.ImportJobsResponse || (exports.ImportJobsResponse = {}));
var Include;
(function (Include) {
    Include["ALL"] = "ALL";
    Include["ANY"] = "ANY";
    Include["NONE"] = "NONE";
})(Include = exports.Include || (exports.Include = {}));
var InternalServerErrorException;
(function (InternalServerErrorException) {
    InternalServerErrorException.isa = (o) => smithy_client_1.isa(o, "InternalServerErrorException");
})(InternalServerErrorException = exports.InternalServerErrorException || (exports.InternalServerErrorException = {}));
var ItemResponse;
(function (ItemResponse) {
    ItemResponse.isa = (o) => smithy_client_1.isa(o, "ItemResponse");
})(ItemResponse = exports.ItemResponse || (exports.ItemResponse = {}));
var JobStatus;
(function (JobStatus) {
    JobStatus["COMPLETED"] = "COMPLETED";
    JobStatus["COMPLETING"] = "COMPLETING";
    JobStatus["CREATED"] = "CREATED";
    JobStatus["FAILED"] = "FAILED";
    JobStatus["FAILING"] = "FAILING";
    JobStatus["INITIALIZING"] = "INITIALIZING";
    JobStatus["PENDING_JOB"] = "PENDING_JOB";
    JobStatus["PREPARING_FOR_INITIALIZATION"] = "PREPARING_FOR_INITIALIZATION";
    JobStatus["PROCESSING"] = "PROCESSING";
})(JobStatus = exports.JobStatus || (exports.JobStatus = {}));
var JourneyDateRangeKpiResponse;
(function (JourneyDateRangeKpiResponse) {
    JourneyDateRangeKpiResponse.isa = (o) => smithy_client_1.isa(o, "JourneyDateRangeKpiResponse");
})(JourneyDateRangeKpiResponse = exports.JourneyDateRangeKpiResponse || (exports.JourneyDateRangeKpiResponse = {}));
var JourneyEmailMessage;
(function (JourneyEmailMessage) {
    JourneyEmailMessage.isa = (o) => smithy_client_1.isa(o, "JourneyEmailMessage");
})(JourneyEmailMessage = exports.JourneyEmailMessage || (exports.JourneyEmailMessage = {}));
var JourneyExecutionActivityMetricsResponse;
(function (JourneyExecutionActivityMetricsResponse) {
    JourneyExecutionActivityMetricsResponse.isa = (o) => smithy_client_1.isa(o, "JourneyExecutionActivityMetricsResponse");
})(JourneyExecutionActivityMetricsResponse = exports.JourneyExecutionActivityMetricsResponse || (exports.JourneyExecutionActivityMetricsResponse = {}));
var JourneyExecutionMetricsResponse;
(function (JourneyExecutionMetricsResponse) {
    JourneyExecutionMetricsResponse.isa = (o) => smithy_client_1.isa(o, "JourneyExecutionMetricsResponse");
})(JourneyExecutionMetricsResponse = exports.JourneyExecutionMetricsResponse || (exports.JourneyExecutionMetricsResponse = {}));
var JourneyLimits;
(function (JourneyLimits) {
    JourneyLimits.isa = (o) => smithy_client_1.isa(o, "JourneyLimits");
})(JourneyLimits = exports.JourneyLimits || (exports.JourneyLimits = {}));
var JourneyResponse;
(function (JourneyResponse) {
    JourneyResponse.isa = (o) => smithy_client_1.isa(o, "JourneyResponse");
})(JourneyResponse = exports.JourneyResponse || (exports.JourneyResponse = {}));
var JourneySchedule;
(function (JourneySchedule) {
    JourneySchedule.isa = (o) => smithy_client_1.isa(o, "JourneySchedule");
})(JourneySchedule = exports.JourneySchedule || (exports.JourneySchedule = {}));
var JourneyStateRequest;
(function (JourneyStateRequest) {
    JourneyStateRequest.isa = (o) => smithy_client_1.isa(o, "JourneyStateRequest");
})(JourneyStateRequest = exports.JourneyStateRequest || (exports.JourneyStateRequest = {}));
var JourneysResponse;
(function (JourneysResponse) {
    JourneysResponse.isa = (o) => smithy_client_1.isa(o, "JourneysResponse");
})(JourneysResponse = exports.JourneysResponse || (exports.JourneysResponse = {}));
var ListJourneysRequest;
(function (ListJourneysRequest) {
    ListJourneysRequest.isa = (o) => smithy_client_1.isa(o, "ListJourneysRequest");
})(ListJourneysRequest = exports.ListJourneysRequest || (exports.ListJourneysRequest = {}));
var ListJourneysResponse;
(function (ListJourneysResponse) {
    ListJourneysResponse.isa = (o) => smithy_client_1.isa(o, "ListJourneysResponse");
})(ListJourneysResponse = exports.ListJourneysResponse || (exports.ListJourneysResponse = {}));
var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.isa = (o) => smithy_client_1.isa(o, "ListTagsForResourceRequest");
})(ListTagsForResourceRequest = exports.ListTagsForResourceRequest || (exports.ListTagsForResourceRequest = {}));
var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.isa = (o) => smithy_client_1.isa(o, "ListTagsForResourceResponse");
})(ListTagsForResourceResponse = exports.ListTagsForResourceResponse || (exports.ListTagsForResourceResponse = {}));
var ListTemplateVersionsRequest;
(function (ListTemplateVersionsRequest) {
    ListTemplateVersionsRequest.isa = (o) => smithy_client_1.isa(o, "ListTemplateVersionsRequest");
})(ListTemplateVersionsRequest = exports.ListTemplateVersionsRequest || (exports.ListTemplateVersionsRequest = {}));
var ListTemplateVersionsResponse;
(function (ListTemplateVersionsResponse) {
    ListTemplateVersionsResponse.isa = (o) => smithy_client_1.isa(o, "ListTemplateVersionsResponse");
})(ListTemplateVersionsResponse = exports.ListTemplateVersionsResponse || (exports.ListTemplateVersionsResponse = {}));
var ListTemplatesRequest;
(function (ListTemplatesRequest) {
    ListTemplatesRequest.isa = (o) => smithy_client_1.isa(o, "ListTemplatesRequest");
})(ListTemplatesRequest = exports.ListTemplatesRequest || (exports.ListTemplatesRequest = {}));
var ListTemplatesResponse;
(function (ListTemplatesResponse) {
    ListTemplatesResponse.isa = (o) => smithy_client_1.isa(o, "ListTemplatesResponse");
})(ListTemplatesResponse = exports.ListTemplatesResponse || (exports.ListTemplatesResponse = {}));
var Message;
(function (Message) {
    Message.isa = (o) => smithy_client_1.isa(o, "Message");
})(Message = exports.Message || (exports.Message = {}));
var MessageBody;
(function (MessageBody) {
    MessageBody.isa = (o) => smithy_client_1.isa(o, "MessageBody");
})(MessageBody = exports.MessageBody || (exports.MessageBody = {}));
var MessageConfiguration;
(function (MessageConfiguration) {
    MessageConfiguration.isa = (o) => smithy_client_1.isa(o, "MessageConfiguration");
})(MessageConfiguration = exports.MessageConfiguration || (exports.MessageConfiguration = {}));
var MessageRequest;
(function (MessageRequest) {
    MessageRequest.isa = (o) => smithy_client_1.isa(o, "MessageRequest");
})(MessageRequest = exports.MessageRequest || (exports.MessageRequest = {}));
var MessageResponse;
(function (MessageResponse) {
    MessageResponse.isa = (o) => smithy_client_1.isa(o, "MessageResponse");
})(MessageResponse = exports.MessageResponse || (exports.MessageResponse = {}));
var MessageResult;
(function (MessageResult) {
    MessageResult.isa = (o) => smithy_client_1.isa(o, "MessageResult");
})(MessageResult = exports.MessageResult || (exports.MessageResult = {}));
var MessageType;
(function (MessageType) {
    MessageType["PROMOTIONAL"] = "PROMOTIONAL";
    MessageType["TRANSACTIONAL"] = "TRANSACTIONAL";
})(MessageType = exports.MessageType || (exports.MessageType = {}));
var MethodNotAllowedException;
(function (MethodNotAllowedException) {
    MethodNotAllowedException.isa = (o) => smithy_client_1.isa(o, "MethodNotAllowedException");
})(MethodNotAllowedException = exports.MethodNotAllowedException || (exports.MethodNotAllowedException = {}));
var MetricDimension;
(function (MetricDimension) {
    MetricDimension.isa = (o) => smithy_client_1.isa(o, "MetricDimension");
})(MetricDimension = exports.MetricDimension || (exports.MetricDimension = {}));
var Mode;
(function (Mode) {
    Mode["DELIVERY"] = "DELIVERY";
    Mode["FILTER"] = "FILTER";
})(Mode = exports.Mode || (exports.Mode = {}));
var MultiConditionalBranch;
(function (MultiConditionalBranch) {
    MultiConditionalBranch.isa = (o) => smithy_client_1.isa(o, "MultiConditionalBranch");
})(MultiConditionalBranch = exports.MultiConditionalBranch || (exports.MultiConditionalBranch = {}));
var MultiConditionalSplitActivity;
(function (MultiConditionalSplitActivity) {
    MultiConditionalSplitActivity.isa = (o) => smithy_client_1.isa(o, "MultiConditionalSplitActivity");
})(MultiConditionalSplitActivity = exports.MultiConditionalSplitActivity || (exports.MultiConditionalSplitActivity = {}));
var NotFoundException;
(function (NotFoundException) {
    NotFoundException.isa = (o) => smithy_client_1.isa(o, "NotFoundException");
})(NotFoundException = exports.NotFoundException || (exports.NotFoundException = {}));
var NumberValidateRequest;
(function (NumberValidateRequest) {
    NumberValidateRequest.isa = (o) => smithy_client_1.isa(o, "NumberValidateRequest");
})(NumberValidateRequest = exports.NumberValidateRequest || (exports.NumberValidateRequest = {}));
var NumberValidateResponse;
(function (NumberValidateResponse) {
    NumberValidateResponse.isa = (o) => smithy_client_1.isa(o, "NumberValidateResponse");
})(NumberValidateResponse = exports.NumberValidateResponse || (exports.NumberValidateResponse = {}));
var Operator;
(function (Operator) {
    Operator["ALL"] = "ALL";
    Operator["ANY"] = "ANY";
})(Operator = exports.Operator || (exports.Operator = {}));
var PhoneNumberValidateRequest;
(function (PhoneNumberValidateRequest) {
    PhoneNumberValidateRequest.isa = (o) => smithy_client_1.isa(o, "PhoneNumberValidateRequest");
})(PhoneNumberValidateRequest = exports.PhoneNumberValidateRequest || (exports.PhoneNumberValidateRequest = {}));
var PhoneNumberValidateResponse;
(function (PhoneNumberValidateResponse) {
    PhoneNumberValidateResponse.isa = (o) => smithy_client_1.isa(o, "PhoneNumberValidateResponse");
})(PhoneNumberValidateResponse = exports.PhoneNumberValidateResponse || (exports.PhoneNumberValidateResponse = {}));
var PublicEndpoint;
(function (PublicEndpoint) {
    PublicEndpoint.isa = (o) => smithy_client_1.isa(o, "PublicEndpoint");
})(PublicEndpoint = exports.PublicEndpoint || (exports.PublicEndpoint = {}));
var PushNotificationTemplateRequest;
(function (PushNotificationTemplateRequest) {
    PushNotificationTemplateRequest.isa = (o) => smithy_client_1.isa(o, "PushNotificationTemplateRequest");
})(PushNotificationTemplateRequest = exports.PushNotificationTemplateRequest || (exports.PushNotificationTemplateRequest = {}));
var PushNotificationTemplateResponse;
(function (PushNotificationTemplateResponse) {
    PushNotificationTemplateResponse.isa = (o) => smithy_client_1.isa(o, "PushNotificationTemplateResponse");
})(PushNotificationTemplateResponse = exports.PushNotificationTemplateResponse || (exports.PushNotificationTemplateResponse = {}));
var PutEventStreamRequest;
(function (PutEventStreamRequest) {
    PutEventStreamRequest.isa = (o) => smithy_client_1.isa(o, "PutEventStreamRequest");
})(PutEventStreamRequest = exports.PutEventStreamRequest || (exports.PutEventStreamRequest = {}));
var PutEventStreamResponse;
(function (PutEventStreamResponse) {
    PutEventStreamResponse.isa = (o) => smithy_client_1.isa(o, "PutEventStreamResponse");
})(PutEventStreamResponse = exports.PutEventStreamResponse || (exports.PutEventStreamResponse = {}));
var PutEventsRequest;
(function (PutEventsRequest) {
    PutEventsRequest.isa = (o) => smithy_client_1.isa(o, "PutEventsRequest");
})(PutEventsRequest = exports.PutEventsRequest || (exports.PutEventsRequest = {}));
var PutEventsResponse;
(function (PutEventsResponse) {
    PutEventsResponse.isa = (o) => smithy_client_1.isa(o, "PutEventsResponse");
})(PutEventsResponse = exports.PutEventsResponse || (exports.PutEventsResponse = {}));
var QuietTime;
(function (QuietTime) {
    QuietTime.isa = (o) => smithy_client_1.isa(o, "QuietTime");
})(QuietTime = exports.QuietTime || (exports.QuietTime = {}));
var RandomSplitActivity;
(function (RandomSplitActivity) {
    RandomSplitActivity.isa = (o) => smithy_client_1.isa(o, "RandomSplitActivity");
})(RandomSplitActivity = exports.RandomSplitActivity || (exports.RandomSplitActivity = {}));
var RandomSplitEntry;
(function (RandomSplitEntry) {
    RandomSplitEntry.isa = (o) => smithy_client_1.isa(o, "RandomSplitEntry");
})(RandomSplitEntry = exports.RandomSplitEntry || (exports.RandomSplitEntry = {}));
var RawEmail;
(function (RawEmail) {
    RawEmail.isa = (o) => smithy_client_1.isa(o, "RawEmail");
})(RawEmail = exports.RawEmail || (exports.RawEmail = {}));
var RecencyDimension;
(function (RecencyDimension) {
    RecencyDimension.isa = (o) => smithy_client_1.isa(o, "RecencyDimension");
})(RecencyDimension = exports.RecencyDimension || (exports.RecencyDimension = {}));
var RecencyType;
(function (RecencyType) {
    RecencyType["ACTIVE"] = "ACTIVE";
    RecencyType["INACTIVE"] = "INACTIVE";
})(RecencyType = exports.RecencyType || (exports.RecencyType = {}));
var RemoveAttributesRequest;
(function (RemoveAttributesRequest) {
    RemoveAttributesRequest.isa = (o) => smithy_client_1.isa(o, "RemoveAttributesRequest");
})(RemoveAttributesRequest = exports.RemoveAttributesRequest || (exports.RemoveAttributesRequest = {}));
var RemoveAttributesResponse;
(function (RemoveAttributesResponse) {
    RemoveAttributesResponse.isa = (o) => smithy_client_1.isa(o, "RemoveAttributesResponse");
})(RemoveAttributesResponse = exports.RemoveAttributesResponse || (exports.RemoveAttributesResponse = {}));
var ResultRow;
(function (ResultRow) {
    ResultRow.isa = (o) => smithy_client_1.isa(o, "ResultRow");
})(ResultRow = exports.ResultRow || (exports.ResultRow = {}));
var ResultRowValue;
(function (ResultRowValue) {
    ResultRowValue.isa = (o) => smithy_client_1.isa(o, "ResultRowValue");
})(ResultRowValue = exports.ResultRowValue || (exports.ResultRowValue = {}));
var SMSChannelRequest;
(function (SMSChannelRequest) {
    SMSChannelRequest.isa = (o) => smithy_client_1.isa(o, "SMSChannelRequest");
})(SMSChannelRequest = exports.SMSChannelRequest || (exports.SMSChannelRequest = {}));
var SMSChannelResponse;
(function (SMSChannelResponse) {
    SMSChannelResponse.isa = (o) => smithy_client_1.isa(o, "SMSChannelResponse");
})(SMSChannelResponse = exports.SMSChannelResponse || (exports.SMSChannelResponse = {}));
var SMSMessage;
(function (SMSMessage) {
    SMSMessage.isa = (o) => smithy_client_1.isa(o, "SMSMessage");
})(SMSMessage = exports.SMSMessage || (exports.SMSMessage = {}));
var SMSTemplateRequest;
(function (SMSTemplateRequest) {
    SMSTemplateRequest.isa = (o) => smithy_client_1.isa(o, "SMSTemplateRequest");
})(SMSTemplateRequest = exports.SMSTemplateRequest || (exports.SMSTemplateRequest = {}));
var SMSTemplateResponse;
(function (SMSTemplateResponse) {
    SMSTemplateResponse.isa = (o) => smithy_client_1.isa(o, "SMSTemplateResponse");
})(SMSTemplateResponse = exports.SMSTemplateResponse || (exports.SMSTemplateResponse = {}));
var Schedule;
(function (Schedule) {
    Schedule.isa = (o) => smithy_client_1.isa(o, "Schedule");
})(Schedule = exports.Schedule || (exports.Schedule = {}));
var SegmentBehaviors;
(function (SegmentBehaviors) {
    SegmentBehaviors.isa = (o) => smithy_client_1.isa(o, "SegmentBehaviors");
})(SegmentBehaviors = exports.SegmentBehaviors || (exports.SegmentBehaviors = {}));
var SegmentCondition;
(function (SegmentCondition) {
    SegmentCondition.isa = (o) => smithy_client_1.isa(o, "SegmentCondition");
})(SegmentCondition = exports.SegmentCondition || (exports.SegmentCondition = {}));
var SegmentDemographics;
(function (SegmentDemographics) {
    SegmentDemographics.isa = (o) => smithy_client_1.isa(o, "SegmentDemographics");
})(SegmentDemographics = exports.SegmentDemographics || (exports.SegmentDemographics = {}));
var SegmentDimensions;
(function (SegmentDimensions) {
    SegmentDimensions.isa = (o) => smithy_client_1.isa(o, "SegmentDimensions");
})(SegmentDimensions = exports.SegmentDimensions || (exports.SegmentDimensions = {}));
var SegmentGroup;
(function (SegmentGroup) {
    SegmentGroup.isa = (o) => smithy_client_1.isa(o, "SegmentGroup");
})(SegmentGroup = exports.SegmentGroup || (exports.SegmentGroup = {}));
var SegmentGroupList;
(function (SegmentGroupList) {
    SegmentGroupList.isa = (o) => smithy_client_1.isa(o, "SegmentGroupList");
})(SegmentGroupList = exports.SegmentGroupList || (exports.SegmentGroupList = {}));
var SegmentImportResource;
(function (SegmentImportResource) {
    SegmentImportResource.isa = (o) => smithy_client_1.isa(o, "SegmentImportResource");
})(SegmentImportResource = exports.SegmentImportResource || (exports.SegmentImportResource = {}));
var SegmentLocation;
(function (SegmentLocation) {
    SegmentLocation.isa = (o) => smithy_client_1.isa(o, "SegmentLocation");
})(SegmentLocation = exports.SegmentLocation || (exports.SegmentLocation = {}));
var SegmentReference;
(function (SegmentReference) {
    SegmentReference.isa = (o) => smithy_client_1.isa(o, "SegmentReference");
})(SegmentReference = exports.SegmentReference || (exports.SegmentReference = {}));
var SegmentResponse;
(function (SegmentResponse) {
    SegmentResponse.isa = (o) => smithy_client_1.isa(o, "SegmentResponse");
})(SegmentResponse = exports.SegmentResponse || (exports.SegmentResponse = {}));
var SegmentType;
(function (SegmentType) {
    SegmentType["DIMENSIONAL"] = "DIMENSIONAL";
    SegmentType["IMPORT"] = "IMPORT";
})(SegmentType = exports.SegmentType || (exports.SegmentType = {}));
var SegmentsResponse;
(function (SegmentsResponse) {
    SegmentsResponse.isa = (o) => smithy_client_1.isa(o, "SegmentsResponse");
})(SegmentsResponse = exports.SegmentsResponse || (exports.SegmentsResponse = {}));
var SendMessagesRequest;
(function (SendMessagesRequest) {
    SendMessagesRequest.isa = (o) => smithy_client_1.isa(o, "SendMessagesRequest");
})(SendMessagesRequest = exports.SendMessagesRequest || (exports.SendMessagesRequest = {}));
var SendMessagesResponse;
(function (SendMessagesResponse) {
    SendMessagesResponse.isa = (o) => smithy_client_1.isa(o, "SendMessagesResponse");
})(SendMessagesResponse = exports.SendMessagesResponse || (exports.SendMessagesResponse = {}));
var SendUsersMessageRequest;
(function (SendUsersMessageRequest) {
    SendUsersMessageRequest.isa = (o) => smithy_client_1.isa(o, "SendUsersMessageRequest");
})(SendUsersMessageRequest = exports.SendUsersMessageRequest || (exports.SendUsersMessageRequest = {}));
var SendUsersMessageResponse;
(function (SendUsersMessageResponse) {
    SendUsersMessageResponse.isa = (o) => smithy_client_1.isa(o, "SendUsersMessageResponse");
})(SendUsersMessageResponse = exports.SendUsersMessageResponse || (exports.SendUsersMessageResponse = {}));
var SendUsersMessagesRequest;
(function (SendUsersMessagesRequest) {
    SendUsersMessagesRequest.isa = (o) => smithy_client_1.isa(o, "SendUsersMessagesRequest");
})(SendUsersMessagesRequest = exports.SendUsersMessagesRequest || (exports.SendUsersMessagesRequest = {}));
var SendUsersMessagesResponse;
(function (SendUsersMessagesResponse) {
    SendUsersMessagesResponse.isa = (o) => smithy_client_1.isa(o, "SendUsersMessagesResponse");
})(SendUsersMessagesResponse = exports.SendUsersMessagesResponse || (exports.SendUsersMessagesResponse = {}));
var Session;
(function (Session) {
    Session.isa = (o) => smithy_client_1.isa(o, "Session");
})(Session = exports.Session || (exports.Session = {}));
var SetDimension;
(function (SetDimension) {
    SetDimension.isa = (o) => smithy_client_1.isa(o, "SetDimension");
})(SetDimension = exports.SetDimension || (exports.SetDimension = {}));
var SimpleCondition;
(function (SimpleCondition) {
    SimpleCondition.isa = (o) => smithy_client_1.isa(o, "SimpleCondition");
})(SimpleCondition = exports.SimpleCondition || (exports.SimpleCondition = {}));
var SimpleEmail;
(function (SimpleEmail) {
    SimpleEmail.isa = (o) => smithy_client_1.isa(o, "SimpleEmail");
})(SimpleEmail = exports.SimpleEmail || (exports.SimpleEmail = {}));
var SimpleEmailPart;
(function (SimpleEmailPart) {
    SimpleEmailPart.isa = (o) => smithy_client_1.isa(o, "SimpleEmailPart");
})(SimpleEmailPart = exports.SimpleEmailPart || (exports.SimpleEmailPart = {}));
var SourceType;
(function (SourceType) {
    SourceType["ALL"] = "ALL";
    SourceType["ANY"] = "ANY";
    SourceType["NONE"] = "NONE";
})(SourceType = exports.SourceType || (exports.SourceType = {}));
var StartCondition;
(function (StartCondition) {
    StartCondition.isa = (o) => smithy_client_1.isa(o, "StartCondition");
})(StartCondition = exports.StartCondition || (exports.StartCondition = {}));
var State;
(function (State) {
    State["ACTIVE"] = "ACTIVE";
    State["CANCELLED"] = "CANCELLED";
    State["CLOSED"] = "CLOSED";
    State["COMPLETED"] = "COMPLETED";
    State["DRAFT"] = "DRAFT";
})(State = exports.State || (exports.State = {}));
var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.isa = (o) => smithy_client_1.isa(o, "TagResourceRequest");
})(TagResourceRequest = exports.TagResourceRequest || (exports.TagResourceRequest = {}));
var TagsModel;
(function (TagsModel) {
    TagsModel.isa = (o) => smithy_client_1.isa(o, "TagsModel");
})(TagsModel = exports.TagsModel || (exports.TagsModel = {}));
var Template;
(function (Template) {
    Template.isa = (o) => smithy_client_1.isa(o, "Template");
})(Template = exports.Template || (exports.Template = {}));
var TemplateActiveVersionRequest;
(function (TemplateActiveVersionRequest) {
    TemplateActiveVersionRequest.isa = (o) => smithy_client_1.isa(o, "TemplateActiveVersionRequest");
})(TemplateActiveVersionRequest = exports.TemplateActiveVersionRequest || (exports.TemplateActiveVersionRequest = {}));
var TemplateConfiguration;
(function (TemplateConfiguration) {
    TemplateConfiguration.isa = (o) => smithy_client_1.isa(o, "TemplateConfiguration");
})(TemplateConfiguration = exports.TemplateConfiguration || (exports.TemplateConfiguration = {}));
var TemplateResponse;
(function (TemplateResponse) {
    TemplateResponse.isa = (o) => smithy_client_1.isa(o, "TemplateResponse");
})(TemplateResponse = exports.TemplateResponse || (exports.TemplateResponse = {}));
var TemplateType;
(function (TemplateType) {
    TemplateType["EMAIL"] = "EMAIL";
    TemplateType["PUSH"] = "PUSH";
    TemplateType["SMS"] = "SMS";
    TemplateType["VOICE"] = "VOICE";
})(TemplateType = exports.TemplateType || (exports.TemplateType = {}));
var TemplateVersionResponse;
(function (TemplateVersionResponse) {
    TemplateVersionResponse.isa = (o) => smithy_client_1.isa(o, "TemplateVersionResponse");
})(TemplateVersionResponse = exports.TemplateVersionResponse || (exports.TemplateVersionResponse = {}));
var TemplateVersionsResponse;
(function (TemplateVersionsResponse) {
    TemplateVersionsResponse.isa = (o) => smithy_client_1.isa(o, "TemplateVersionsResponse");
})(TemplateVersionsResponse = exports.TemplateVersionsResponse || (exports.TemplateVersionsResponse = {}));
var TemplatesResponse;
(function (TemplatesResponse) {
    TemplatesResponse.isa = (o) => smithy_client_1.isa(o, "TemplatesResponse");
})(TemplatesResponse = exports.TemplatesResponse || (exports.TemplatesResponse = {}));
var TooManyRequestsException;
(function (TooManyRequestsException) {
    TooManyRequestsException.isa = (o) => smithy_client_1.isa(o, "TooManyRequestsException");
})(TooManyRequestsException = exports.TooManyRequestsException || (exports.TooManyRequestsException = {}));
var TreatmentResource;
(function (TreatmentResource) {
    TreatmentResource.isa = (o) => smithy_client_1.isa(o, "TreatmentResource");
})(TreatmentResource = exports.TreatmentResource || (exports.TreatmentResource = {}));
var Type;
(function (Type) {
    Type["ALL"] = "ALL";
    Type["ANY"] = "ANY";
    Type["NONE"] = "NONE";
})(Type = exports.Type || (exports.Type = {}));
var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.isa = (o) => smithy_client_1.isa(o, "UntagResourceRequest");
})(UntagResourceRequest = exports.UntagResourceRequest || (exports.UntagResourceRequest = {}));
var UpdateAdmChannelRequest;
(function (UpdateAdmChannelRequest) {
    UpdateAdmChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateAdmChannelRequest");
})(UpdateAdmChannelRequest = exports.UpdateAdmChannelRequest || (exports.UpdateAdmChannelRequest = {}));
var UpdateAdmChannelResponse;
(function (UpdateAdmChannelResponse) {
    UpdateAdmChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateAdmChannelResponse");
})(UpdateAdmChannelResponse = exports.UpdateAdmChannelResponse || (exports.UpdateAdmChannelResponse = {}));
var UpdateApnsChannelRequest;
(function (UpdateApnsChannelRequest) {
    UpdateApnsChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateApnsChannelRequest");
})(UpdateApnsChannelRequest = exports.UpdateApnsChannelRequest || (exports.UpdateApnsChannelRequest = {}));
var UpdateApnsChannelResponse;
(function (UpdateApnsChannelResponse) {
    UpdateApnsChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateApnsChannelResponse");
})(UpdateApnsChannelResponse = exports.UpdateApnsChannelResponse || (exports.UpdateApnsChannelResponse = {}));
var UpdateApnsSandboxChannelRequest;
(function (UpdateApnsSandboxChannelRequest) {
    UpdateApnsSandboxChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateApnsSandboxChannelRequest");
})(UpdateApnsSandboxChannelRequest = exports.UpdateApnsSandboxChannelRequest || (exports.UpdateApnsSandboxChannelRequest = {}));
var UpdateApnsSandboxChannelResponse;
(function (UpdateApnsSandboxChannelResponse) {
    UpdateApnsSandboxChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateApnsSandboxChannelResponse");
})(UpdateApnsSandboxChannelResponse = exports.UpdateApnsSandboxChannelResponse || (exports.UpdateApnsSandboxChannelResponse = {}));
var UpdateApnsVoipChannelRequest;
(function (UpdateApnsVoipChannelRequest) {
    UpdateApnsVoipChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateApnsVoipChannelRequest");
})(UpdateApnsVoipChannelRequest = exports.UpdateApnsVoipChannelRequest || (exports.UpdateApnsVoipChannelRequest = {}));
var UpdateApnsVoipChannelResponse;
(function (UpdateApnsVoipChannelResponse) {
    UpdateApnsVoipChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateApnsVoipChannelResponse");
})(UpdateApnsVoipChannelResponse = exports.UpdateApnsVoipChannelResponse || (exports.UpdateApnsVoipChannelResponse = {}));
var UpdateApnsVoipSandboxChannelRequest;
(function (UpdateApnsVoipSandboxChannelRequest) {
    UpdateApnsVoipSandboxChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateApnsVoipSandboxChannelRequest");
})(UpdateApnsVoipSandboxChannelRequest = exports.UpdateApnsVoipSandboxChannelRequest || (exports.UpdateApnsVoipSandboxChannelRequest = {}));
var UpdateApnsVoipSandboxChannelResponse;
(function (UpdateApnsVoipSandboxChannelResponse) {
    UpdateApnsVoipSandboxChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateApnsVoipSandboxChannelResponse");
})(UpdateApnsVoipSandboxChannelResponse = exports.UpdateApnsVoipSandboxChannelResponse || (exports.UpdateApnsVoipSandboxChannelResponse = {}));
var UpdateApplicationSettingsRequest;
(function (UpdateApplicationSettingsRequest) {
    UpdateApplicationSettingsRequest.isa = (o) => smithy_client_1.isa(o, "UpdateApplicationSettingsRequest");
})(UpdateApplicationSettingsRequest = exports.UpdateApplicationSettingsRequest || (exports.UpdateApplicationSettingsRequest = {}));
var UpdateApplicationSettingsResponse;
(function (UpdateApplicationSettingsResponse) {
    UpdateApplicationSettingsResponse.isa = (o) => smithy_client_1.isa(o, "UpdateApplicationSettingsResponse");
})(UpdateApplicationSettingsResponse = exports.UpdateApplicationSettingsResponse || (exports.UpdateApplicationSettingsResponse = {}));
var UpdateAttributesRequest;
(function (UpdateAttributesRequest) {
    UpdateAttributesRequest.isa = (o) => smithy_client_1.isa(o, "UpdateAttributesRequest");
})(UpdateAttributesRequest = exports.UpdateAttributesRequest || (exports.UpdateAttributesRequest = {}));
var UpdateBaiduChannelRequest;
(function (UpdateBaiduChannelRequest) {
    UpdateBaiduChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateBaiduChannelRequest");
})(UpdateBaiduChannelRequest = exports.UpdateBaiduChannelRequest || (exports.UpdateBaiduChannelRequest = {}));
var UpdateBaiduChannelResponse;
(function (UpdateBaiduChannelResponse) {
    UpdateBaiduChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateBaiduChannelResponse");
})(UpdateBaiduChannelResponse = exports.UpdateBaiduChannelResponse || (exports.UpdateBaiduChannelResponse = {}));
var UpdateCampaignRequest;
(function (UpdateCampaignRequest) {
    UpdateCampaignRequest.isa = (o) => smithy_client_1.isa(o, "UpdateCampaignRequest");
})(UpdateCampaignRequest = exports.UpdateCampaignRequest || (exports.UpdateCampaignRequest = {}));
var UpdateCampaignResponse;
(function (UpdateCampaignResponse) {
    UpdateCampaignResponse.isa = (o) => smithy_client_1.isa(o, "UpdateCampaignResponse");
})(UpdateCampaignResponse = exports.UpdateCampaignResponse || (exports.UpdateCampaignResponse = {}));
var UpdateEmailChannelRequest;
(function (UpdateEmailChannelRequest) {
    UpdateEmailChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateEmailChannelRequest");
})(UpdateEmailChannelRequest = exports.UpdateEmailChannelRequest || (exports.UpdateEmailChannelRequest = {}));
var UpdateEmailChannelResponse;
(function (UpdateEmailChannelResponse) {
    UpdateEmailChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateEmailChannelResponse");
})(UpdateEmailChannelResponse = exports.UpdateEmailChannelResponse || (exports.UpdateEmailChannelResponse = {}));
var UpdateEmailTemplateRequest;
(function (UpdateEmailTemplateRequest) {
    UpdateEmailTemplateRequest.isa = (o) => smithy_client_1.isa(o, "UpdateEmailTemplateRequest");
})(UpdateEmailTemplateRequest = exports.UpdateEmailTemplateRequest || (exports.UpdateEmailTemplateRequest = {}));
var UpdateEmailTemplateResponse;
(function (UpdateEmailTemplateResponse) {
    UpdateEmailTemplateResponse.isa = (o) => smithy_client_1.isa(o, "UpdateEmailTemplateResponse");
})(UpdateEmailTemplateResponse = exports.UpdateEmailTemplateResponse || (exports.UpdateEmailTemplateResponse = {}));
var UpdateEndpointRequest;
(function (UpdateEndpointRequest) {
    UpdateEndpointRequest.isa = (o) => smithy_client_1.isa(o, "UpdateEndpointRequest");
})(UpdateEndpointRequest = exports.UpdateEndpointRequest || (exports.UpdateEndpointRequest = {}));
var UpdateEndpointResponse;
(function (UpdateEndpointResponse) {
    UpdateEndpointResponse.isa = (o) => smithy_client_1.isa(o, "UpdateEndpointResponse");
})(UpdateEndpointResponse = exports.UpdateEndpointResponse || (exports.UpdateEndpointResponse = {}));
var UpdateEndpointsBatchRequest;
(function (UpdateEndpointsBatchRequest) {
    UpdateEndpointsBatchRequest.isa = (o) => smithy_client_1.isa(o, "UpdateEndpointsBatchRequest");
})(UpdateEndpointsBatchRequest = exports.UpdateEndpointsBatchRequest || (exports.UpdateEndpointsBatchRequest = {}));
var UpdateEndpointsBatchResponse;
(function (UpdateEndpointsBatchResponse) {
    UpdateEndpointsBatchResponse.isa = (o) => smithy_client_1.isa(o, "UpdateEndpointsBatchResponse");
})(UpdateEndpointsBatchResponse = exports.UpdateEndpointsBatchResponse || (exports.UpdateEndpointsBatchResponse = {}));
var UpdateGcmChannelRequest;
(function (UpdateGcmChannelRequest) {
    UpdateGcmChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateGcmChannelRequest");
})(UpdateGcmChannelRequest = exports.UpdateGcmChannelRequest || (exports.UpdateGcmChannelRequest = {}));
var UpdateGcmChannelResponse;
(function (UpdateGcmChannelResponse) {
    UpdateGcmChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateGcmChannelResponse");
})(UpdateGcmChannelResponse = exports.UpdateGcmChannelResponse || (exports.UpdateGcmChannelResponse = {}));
var UpdateJourneyRequest;
(function (UpdateJourneyRequest) {
    UpdateJourneyRequest.isa = (o) => smithy_client_1.isa(o, "UpdateJourneyRequest");
})(UpdateJourneyRequest = exports.UpdateJourneyRequest || (exports.UpdateJourneyRequest = {}));
var UpdateJourneyResponse;
(function (UpdateJourneyResponse) {
    UpdateJourneyResponse.isa = (o) => smithy_client_1.isa(o, "UpdateJourneyResponse");
})(UpdateJourneyResponse = exports.UpdateJourneyResponse || (exports.UpdateJourneyResponse = {}));
var UpdateJourneyStateRequest;
(function (UpdateJourneyStateRequest) {
    UpdateJourneyStateRequest.isa = (o) => smithy_client_1.isa(o, "UpdateJourneyStateRequest");
})(UpdateJourneyStateRequest = exports.UpdateJourneyStateRequest || (exports.UpdateJourneyStateRequest = {}));
var UpdateJourneyStateResponse;
(function (UpdateJourneyStateResponse) {
    UpdateJourneyStateResponse.isa = (o) => smithy_client_1.isa(o, "UpdateJourneyStateResponse");
})(UpdateJourneyStateResponse = exports.UpdateJourneyStateResponse || (exports.UpdateJourneyStateResponse = {}));
var UpdatePushTemplateRequest;
(function (UpdatePushTemplateRequest) {
    UpdatePushTemplateRequest.isa = (o) => smithy_client_1.isa(o, "UpdatePushTemplateRequest");
})(UpdatePushTemplateRequest = exports.UpdatePushTemplateRequest || (exports.UpdatePushTemplateRequest = {}));
var UpdatePushTemplateResponse;
(function (UpdatePushTemplateResponse) {
    UpdatePushTemplateResponse.isa = (o) => smithy_client_1.isa(o, "UpdatePushTemplateResponse");
})(UpdatePushTemplateResponse = exports.UpdatePushTemplateResponse || (exports.UpdatePushTemplateResponse = {}));
var UpdateSegmentRequest;
(function (UpdateSegmentRequest) {
    UpdateSegmentRequest.isa = (o) => smithy_client_1.isa(o, "UpdateSegmentRequest");
})(UpdateSegmentRequest = exports.UpdateSegmentRequest || (exports.UpdateSegmentRequest = {}));
var UpdateSegmentResponse;
(function (UpdateSegmentResponse) {
    UpdateSegmentResponse.isa = (o) => smithy_client_1.isa(o, "UpdateSegmentResponse");
})(UpdateSegmentResponse = exports.UpdateSegmentResponse || (exports.UpdateSegmentResponse = {}));
var UpdateSmsChannelRequest;
(function (UpdateSmsChannelRequest) {
    UpdateSmsChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateSmsChannelRequest");
})(UpdateSmsChannelRequest = exports.UpdateSmsChannelRequest || (exports.UpdateSmsChannelRequest = {}));
var UpdateSmsChannelResponse;
(function (UpdateSmsChannelResponse) {
    UpdateSmsChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateSmsChannelResponse");
})(UpdateSmsChannelResponse = exports.UpdateSmsChannelResponse || (exports.UpdateSmsChannelResponse = {}));
var UpdateSmsTemplateRequest;
(function (UpdateSmsTemplateRequest) {
    UpdateSmsTemplateRequest.isa = (o) => smithy_client_1.isa(o, "UpdateSmsTemplateRequest");
})(UpdateSmsTemplateRequest = exports.UpdateSmsTemplateRequest || (exports.UpdateSmsTemplateRequest = {}));
var UpdateSmsTemplateResponse;
(function (UpdateSmsTemplateResponse) {
    UpdateSmsTemplateResponse.isa = (o) => smithy_client_1.isa(o, "UpdateSmsTemplateResponse");
})(UpdateSmsTemplateResponse = exports.UpdateSmsTemplateResponse || (exports.UpdateSmsTemplateResponse = {}));
var UpdateTemplateActiveVersionRequest;
(function (UpdateTemplateActiveVersionRequest) {
    UpdateTemplateActiveVersionRequest.isa = (o) => smithy_client_1.isa(o, "UpdateTemplateActiveVersionRequest");
})(UpdateTemplateActiveVersionRequest = exports.UpdateTemplateActiveVersionRequest || (exports.UpdateTemplateActiveVersionRequest = {}));
var UpdateTemplateActiveVersionResponse;
(function (UpdateTemplateActiveVersionResponse) {
    UpdateTemplateActiveVersionResponse.isa = (o) => smithy_client_1.isa(o, "UpdateTemplateActiveVersionResponse");
})(UpdateTemplateActiveVersionResponse = exports.UpdateTemplateActiveVersionResponse || (exports.UpdateTemplateActiveVersionResponse = {}));
var UpdateVoiceChannelRequest;
(function (UpdateVoiceChannelRequest) {
    UpdateVoiceChannelRequest.isa = (o) => smithy_client_1.isa(o, "UpdateVoiceChannelRequest");
})(UpdateVoiceChannelRequest = exports.UpdateVoiceChannelRequest || (exports.UpdateVoiceChannelRequest = {}));
var UpdateVoiceChannelResponse;
(function (UpdateVoiceChannelResponse) {
    UpdateVoiceChannelResponse.isa = (o) => smithy_client_1.isa(o, "UpdateVoiceChannelResponse");
})(UpdateVoiceChannelResponse = exports.UpdateVoiceChannelResponse || (exports.UpdateVoiceChannelResponse = {}));
var UpdateVoiceTemplateRequest;
(function (UpdateVoiceTemplateRequest) {
    UpdateVoiceTemplateRequest.isa = (o) => smithy_client_1.isa(o, "UpdateVoiceTemplateRequest");
})(UpdateVoiceTemplateRequest = exports.UpdateVoiceTemplateRequest || (exports.UpdateVoiceTemplateRequest = {}));
var UpdateVoiceTemplateResponse;
(function (UpdateVoiceTemplateResponse) {
    UpdateVoiceTemplateResponse.isa = (o) => smithy_client_1.isa(o, "UpdateVoiceTemplateResponse");
})(UpdateVoiceTemplateResponse = exports.UpdateVoiceTemplateResponse || (exports.UpdateVoiceTemplateResponse = {}));
var VoiceChannelRequest;
(function (VoiceChannelRequest) {
    VoiceChannelRequest.isa = (o) => smithy_client_1.isa(o, "VoiceChannelRequest");
})(VoiceChannelRequest = exports.VoiceChannelRequest || (exports.VoiceChannelRequest = {}));
var VoiceChannelResponse;
(function (VoiceChannelResponse) {
    VoiceChannelResponse.isa = (o) => smithy_client_1.isa(o, "VoiceChannelResponse");
})(VoiceChannelResponse = exports.VoiceChannelResponse || (exports.VoiceChannelResponse = {}));
var VoiceMessage;
(function (VoiceMessage) {
    VoiceMessage.isa = (o) => smithy_client_1.isa(o, "VoiceMessage");
})(VoiceMessage = exports.VoiceMessage || (exports.VoiceMessage = {}));
var VoiceTemplateRequest;
(function (VoiceTemplateRequest) {
    VoiceTemplateRequest.isa = (o) => smithy_client_1.isa(o, "VoiceTemplateRequest");
})(VoiceTemplateRequest = exports.VoiceTemplateRequest || (exports.VoiceTemplateRequest = {}));
var VoiceTemplateResponse;
(function (VoiceTemplateResponse) {
    VoiceTemplateResponse.isa = (o) => smithy_client_1.isa(o, "VoiceTemplateResponse");
})(VoiceTemplateResponse = exports.VoiceTemplateResponse || (exports.VoiceTemplateResponse = {}));
var WaitActivity;
(function (WaitActivity) {
    WaitActivity.isa = (o) => smithy_client_1.isa(o, "WaitActivity");
})(WaitActivity = exports.WaitActivity || (exports.WaitActivity = {}));
var WaitTime;
(function (WaitTime) {
    WaitTime.isa = (o) => smithy_client_1.isa(o, "WaitTime");
})(WaitTime = exports.WaitTime || (exports.WaitTime = {}));
var WriteApplicationSettingsRequest;
(function (WriteApplicationSettingsRequest) {
    WriteApplicationSettingsRequest.isa = (o) => smithy_client_1.isa(o, "WriteApplicationSettingsRequest");
})(WriteApplicationSettingsRequest = exports.WriteApplicationSettingsRequest || (exports.WriteApplicationSettingsRequest = {}));
var WriteCampaignRequest;
(function (WriteCampaignRequest) {
    WriteCampaignRequest.isa = (o) => smithy_client_1.isa(o, "WriteCampaignRequest");
})(WriteCampaignRequest = exports.WriteCampaignRequest || (exports.WriteCampaignRequest = {}));
var WriteEventStream;
(function (WriteEventStream) {
    WriteEventStream.isa = (o) => smithy_client_1.isa(o, "WriteEventStream");
})(WriteEventStream = exports.WriteEventStream || (exports.WriteEventStream = {}));
var WriteJourneyRequest;
(function (WriteJourneyRequest) {
    WriteJourneyRequest.isa = (o) => smithy_client_1.isa(o, "WriteJourneyRequest");
})(WriteJourneyRequest = exports.WriteJourneyRequest || (exports.WriteJourneyRequest = {}));
var WriteSegmentRequest;
(function (WriteSegmentRequest) {
    WriteSegmentRequest.isa = (o) => smithy_client_1.isa(o, "WriteSegmentRequest");
})(WriteSegmentRequest = exports.WriteSegmentRequest || (exports.WriteSegmentRequest = {}));
var WriteTreatmentResource;
(function (WriteTreatmentResource) {
    WriteTreatmentResource.isa = (o) => smithy_client_1.isa(o, "WriteTreatmentResource");
})(WriteTreatmentResource = exports.WriteTreatmentResource || (exports.WriteTreatmentResource = {}));
