// @aws-amplify/ui-components/dist/cjs/amplify-greetings.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-b93ab635.js');
require('@aws-amplify/core');
require('./auth-types-cd1f71d2.js');
require('./Translations-7087b270.js');
require('./constants-3aa9204f.js');
require('@aws-amplify/auth');
const helpers = require('./helpers-7af08dc1.js');

const amplifyGreetingsCss = ":host{--background-color:var(--amplify-white);--border-color:var(--amplify-light-grey);--font-family:var(--amplify-font-family)}.greetings{position:relative;border:1px solid transparent;background-color:var(--background-color);border-color:var(--border-color);padding:10px;font-family:var(--font-family)}amplify-sign-out{margin-left:1rem}";

class AmplifyGreetings {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        /** Username displayed in the greetings */
        this.username = null;
        /** Logo displayed inside of the header */
        this.logo = null;
        /** Auth state change handler for this component */
        this.handleAuthStateChange = helpers.dispatchAuthStateChangeEvent;
    }
    render() {
        return (index.h("header", { class: "greetings" }, this.logo && index.h("span", null, this.logo), index.h("amplify-nav", null, this.username && (index.h("slot", { name: "greetings-message" }, index.h("span", null, "Hello, ", this.username))), index.h("amplify-sign-out", { handleAuthStateChange: this.handleAuthStateChange }))));
    }
}
AmplifyGreetings.style = amplifyGreetingsCss;

exports.amplify_greetings = AmplifyGreetings;
