// @aws-amplify/ui-components/dist/collection/components/amplify-authenticator/amplify-authenticator.js
import { Component, State, Prop, h, Host } from '@stencil/core';
import { AuthState, } from '../../common/types/auth-types';
import { AUTH_CHANNEL, NO_AUTH_MODULE_FOUND, REDIRECTED_FROM_HOSTED_UI, AUTHENTICATOR_AUTHSTATE, UI_AUTH_CHANNEL, TOAST_AUTH_ERROR_EVENT, } from '../../common/constants';
import { Auth, appendToCognitoUserAgent } from '@aws-amplify/auth';
import { Hub, Logger } from '@aws-amplify/core';
import { dispatchAuthStateChangeEvent, onAuthUIStateChange } from '../../common/helpers';
const logger = new Logger('Authenticator');
export class AmplifyAuthenticator {
    constructor() {
        /** Initial starting state of the Authenticator component. E.g. If `signup` is passed the default component is set to AmplifySignUp */
        this.initialAuthState = AuthState.SignIn;
        /** Callback for Authenticator state machine changes */
        this.handleAuthStateChange = () => { };
        this.authState = AuthState.Loading;
        this.toastMessage = '';
        this.handleExternalAuthEvent = ({ payload }) => {
            switch (payload.event) {
                case 'cognitoHostedUI':
                    return dispatchAuthStateChangeEvent(AuthState.SignedIn, payload.data);
                case 'cognitoHostedUI_failure':
                case 'parsingUrl_failure':
                case 'signOut':
                case 'customGreetingSignOut':
                    return dispatchAuthStateChangeEvent(this.initialAuthState);
            }
        };
        this.handleToastEvent = ({ payload }) => {
            switch (payload.event) {
                case TOAST_AUTH_ERROR_EVENT:
                    if (payload.message)
                        this.toastMessage = payload.message;
                    break;
            }
        };
    }
    async componentWillLoad() {
        onAuthUIStateChange((authState, authData) => {
            this.onAuthStateChange(authState, authData);
            this.toastMessage = '';
        });
        Hub.listen(UI_AUTH_CHANNEL, this.handleToastEvent);
        Hub.listen(AUTH_CHANNEL, this.handleExternalAuthEvent);
        appendToCognitoUserAgent('amplify-authenticator');
        const byHostedUI = localStorage.getItem(REDIRECTED_FROM_HOSTED_UI);
        localStorage.removeItem(REDIRECTED_FROM_HOSTED_UI);
        if (byHostedUI !== 'true')
            await this.checkUser();
    }
    async checkUser() {
        if (!Auth || typeof Auth.currentAuthenticatedUser !== 'function') {
            throw new Error(NO_AUTH_MODULE_FOUND);
        }
        try {
            const user = await Auth.currentAuthenticatedUser();
            dispatchAuthStateChangeEvent(AuthState.SignedIn, user);
        }
        catch (error) {
            let cachedAuthState = null;
            try {
                cachedAuthState = localStorage.getItem(AUTHENTICATOR_AUTHSTATE);
            }
            catch (error) {
                logger.debug('Failed to get the auth state from local storage', error);
            }
            try {
                if (cachedAuthState === AuthState.SignedIn) {
                    await Auth.signOut();
                }
                dispatchAuthStateChangeEvent(this.initialAuthState);
            }
            catch (error) {
                logger.debug('Failed to sign out', error);
            }
        }
    }
    async onAuthStateChange(nextAuthState, data) {
        if (nextAuthState === undefined)
            return logger.error('nextAuthState cannot be undefined');
        logger.info('Inside onAuthStateChange Method current authState:', this.authState);
        if (nextAuthState === AuthState.SignedOut) {
            this.authState = this.initialAuthState;
        }
        else {
            this.authState = nextAuthState;
        }
        this.authData = data;
        if (this.authData)
            logger.log('Auth Data was set:', this.authData);
        if (this.authState === nextAuthState) {
            this.handleAuthStateChange(this.authState, this.authData);
            logger.info(`authState has been updated to ${this.authState}`);
        }
    }
    renderAuthComponent(authState) {
        switch (authState) {
            case AuthState.SignIn:
                return (h("slot", { name: "sign-in" },
                    h("amplify-sign-in", { federated: this.federated, usernameAlias: this.usernameAlias })));
            case AuthState.ConfirmSignIn:
                return (h("slot", { name: "confirm-sign-in" },
                    h("amplify-confirm-sign-in", { user: this.authData })));
            case AuthState.SignUp:
                return (h("slot", { name: "sign-up" },
                    h("amplify-sign-up", { usernameAlias: this.usernameAlias })));
            case AuthState.ConfirmSignUp:
                return (h("slot", { name: "confirm-sign-up" },
                    h("amplify-confirm-sign-up", { user: this.authData, usernameAlias: this.usernameAlias })));
            case AuthState.ForgotPassword:
                return (h("slot", { name: "forgot-password" },
                    h("amplify-forgot-password", { usernameAlias: this.usernameAlias })));
            case AuthState.ResetPassword:
                return (h("slot", { name: "require-new-password" },
                    h("amplify-require-new-password", { user: this.authData })));
            case AuthState.VerifyContact:
                return (h("slot", { name: "verify-contact" },
                    h("amplify-verify-contact", { user: this.authData })));
            case AuthState.TOTPSetup:
                return (h("slot", { name: "totp-setup" },
                    h("amplify-totp-setup", { user: this.authData })));
            case AuthState.Loading:
                return (h("slot", { name: "loading" },
                    h("div", null, "Loading...")));
            case AuthState.SignedIn:
                return [h("slot", { name: "greetings" }), h("slot", null)];
            default:
                throw new Error(`Unhandled auth state: ${authState}`);
        }
    }
    componentWillUnload() {
        Hub.remove(AUTH_CHANNEL, this.handleExternalAuthEvent);
        Hub.remove(UI_AUTH_CHANNEL, this.handleToastEvent);
        return onAuthUIStateChange;
    }
    render() {
        return (h(Host, null,
            this.toastMessage ? (h("amplify-toast", { message: this.toastMessage, handleClose: () => {
                    this.toastMessage = '';
                }, "data-test": "authenticator-error" })) : null,
            this.renderAuthComponent(this.authState)));
    }
    static get is() { return "amplify-authenticator"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() { return {
        "$": ["amplify-authenticator.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["amplify-authenticator.css"]
    }; }
    static get properties() { return {
        "initialAuthState": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "AuthState.SignIn | AuthState.SignUp",
                "resolved": "AuthState.SignIn | AuthState.SignUp",
                "references": {
                    "AuthState": {
                        "location": "import",
                        "path": "../../common/types/auth-types"
                    }
                }
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Initial starting state of the Authenticator component. E.g. If `signup` is passed the default component is set to AmplifySignUp"
            },
            "attribute": "initial-auth-state",
            "reflect": false,
            "defaultValue": "AuthState.SignIn"
        },
        "federated": {
            "type": "unknown",
            "mutable": false,
            "complexType": {
                "original": "FederatedConfig",
                "resolved": "FederatedConfig",
                "references": {
                    "FederatedConfig": {
                        "location": "import",
                        "path": "../../common/types/auth-types"
                    }
                }
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Federated credentials & configuration."
            }
        },
        "usernameAlias": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "UsernameAliasStrings",
                "resolved": "\"email\" | \"phone_number\" | \"username\"",
                "references": {
                    "UsernameAliasStrings": {
                        "location": "import",
                        "path": "../../common/types/auth-types"
                    }
                }
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Username Alias is used to setup authentication with `username`, `email` or `phone_number`"
            },
            "attribute": "username-alias",
            "reflect": false
        },
        "handleAuthStateChange": {
            "type": "unknown",
            "mutable": false,
            "complexType": {
                "original": "AuthStateHandler",
                "resolved": "(nextAuthState: AuthState, data?: object) => void",
                "references": {
                    "AuthStateHandler": {
                        "location": "import",
                        "path": "../../common/types/auth-types"
                    }
                }
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Callback for Authenticator state machine changes"
            },
            "defaultValue": "() => {}"
        }
    }; }
    static get states() { return {
        "authState": {},
        "authData": {},
        "toastMessage": {}
    }; }
}
